<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="es" xml:lang="es"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.42">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="OMICs Analysis">

<title>Analisis Genomas: E. coli</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="index_files/libs/clipboard/clipboard.min.js"></script>
<script src="index_files/libs/quarto-html/quarto.js"></script>
<script src="index_files/libs/quarto-html/popper.min.js"></script>
<script src="index_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="index_files/libs/quarto-html/anchor.min.js"></script>
<link href="index_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="index_files/libs/quarto-html/quarto-syntax-highlighting-2f5df379a58b258e96c21c0638c20c03.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="index_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="index_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="index_files/libs/bootstrap/bootstrap-a71a25f19e128005acb9c00d38d9c2cf.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Tabla de contenidos</h2>
   
  <ul>
  <li><a href="#introducción" id="toc-introducción" class="nav-link active" data-scroll-target="#introducción">Introducción</a></li>
  <li><a href="#anotación-de-genomas" id="toc-anotación-de-genomas" class="nav-link" data-scroll-target="#anotación-de-genomas">Anotación de genomas</a>
  <ul class="collapse">
  <li><a href="#prokka" id="toc-prokka" class="nav-link" data-scroll-target="#prokka">Prokka</a></li>
  <li><a href="#bakta" id="toc-bakta" class="nav-link" data-scroll-target="#bakta">Bakta</a></li>
  </ul></li>
  </ul>
</nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Analisis Genomas: E. coli</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Autor/a</div>
    <div class="quarto-title-meta-contents">
             <p>OMICs Analysis </p>
          </div>
  </div>
    
  
    
  </div>
  


</header>


<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/logo_OAN_for_white.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="300"></p>
</figure>
</div>
<section id="introducción" class="level1">
<h1>Introducción</h1>
<p>Vamos a trabajar con ensambles de distintas cepas de E. coli hospedero Bovino.</p>
<p><strong>NOTA</strong> ⚠️👁️‍🗨️: Para las figuras del curso se utilizaron todos los 110 genomas con hospedero Bovino. Para este tutorial generaremos un subconjunto de 15, ya incluidos genomas de referencia para patotipos EHEC y EPEC.</p>
<p>Para iniciar vamos a descargar los datos disponibles que existan usando <code>datasets</code> de NCBI, lo puedes instalar <a href="https://github.com/ncbi/datasets">aquí</a>.</p>
<p>Con el siguiente código vamos a descargar los genomas depoitados en NCBI de <em>E. coli</em></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> 1_data</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> 1_data</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="va">$HOME</span><span class="ex">/bin/datasets</span> <span class="dt">\</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>summary genome taxon <span class="st">'562'</span> <span class="dt">\</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>--assembly-level complete<span class="dt">\</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>--assembly-source refseq <span class="dt">\</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>--as-json-lines <span class="op">&gt;</span> Escherichia_coli_562.json</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="va">$HOME</span><span class="ex">/bin/dataformat</span> tsv genome <span class="dt">\</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>--inputfile Escherichia_coli_562.json <span class="op">&gt;</span> Escherichia_coli_562.tsv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Este conjunto de metadatos <code>Escherichia_coli_562.tsv</code> nos servirá para seleccionar los ensambles de genoma que necesitamos. Para ello puedes utilizar la plataforma o método que más prefieras, y quedarnos con un conjunto más pequeño de genomas para descagar. En este tutorial este será el archivo con el que trabajaremos <code>Escherichia_coli_Bovine_Host.tsv</code>, este archivo se ve de la siguiete forma (las primeras 5 filas):</p>
<div class="cell">
<div class="cell-output-display">
<div id="bxvglmpkcr" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#bxvglmpkcr table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#bxvglmpkcr thead, #bxvglmpkcr tbody, #bxvglmpkcr tfoot, #bxvglmpkcr tr, #bxvglmpkcr td, #bxvglmpkcr th {
  border-style: none;
}

#bxvglmpkcr p {
  margin: 0;
  padding: 0;
}

#bxvglmpkcr .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#bxvglmpkcr .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#bxvglmpkcr .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#bxvglmpkcr .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#bxvglmpkcr .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#bxvglmpkcr .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#bxvglmpkcr .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#bxvglmpkcr .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#bxvglmpkcr .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#bxvglmpkcr .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#bxvglmpkcr .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#bxvglmpkcr .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#bxvglmpkcr .gt_spanner_row {
  border-bottom-style: hidden;
}

#bxvglmpkcr .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#bxvglmpkcr .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#bxvglmpkcr .gt_from_md > :first-child {
  margin-top: 0;
}

#bxvglmpkcr .gt_from_md > :last-child {
  margin-bottom: 0;
}

#bxvglmpkcr .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#bxvglmpkcr .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#bxvglmpkcr .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#bxvglmpkcr .gt_row_group_first td {
  border-top-width: 2px;
}

#bxvglmpkcr .gt_row_group_first th {
  border-top-width: 2px;
}

#bxvglmpkcr .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#bxvglmpkcr .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#bxvglmpkcr .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#bxvglmpkcr .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#bxvglmpkcr .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#bxvglmpkcr .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#bxvglmpkcr .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#bxvglmpkcr .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#bxvglmpkcr .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#bxvglmpkcr .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#bxvglmpkcr .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#bxvglmpkcr .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#bxvglmpkcr .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#bxvglmpkcr .gt_left {
  text-align: left;
}

#bxvglmpkcr .gt_center {
  text-align: center;
}

#bxvglmpkcr .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#bxvglmpkcr .gt_font_normal {
  font-weight: normal;
}

#bxvglmpkcr .gt_font_bold {
  font-weight: bold;
}

#bxvglmpkcr .gt_font_italic {
  font-style: italic;
}

#bxvglmpkcr .gt_super {
  font-size: 65%;
}

#bxvglmpkcr .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#bxvglmpkcr .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#bxvglmpkcr .gt_indent_1 {
  text-indent: 5px;
}

#bxvglmpkcr .gt_indent_2 {
  text-indent: 10px;
}

#bxvglmpkcr .gt_indent_3 {
  text-indent: 15px;
}

#bxvglmpkcr .gt_indent_4 {
  text-indent: 20px;
}

#bxvglmpkcr .gt_indent_5 {
  text-indent: 25px;
}

#bxvglmpkcr .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#bxvglmpkcr div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead>
<tr class="gt_col_headings header">
<th id="Assembly.Accession" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Assembly.Accession</th>
<th id="Assembly.Level" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Assembly.Level</th>
<th id="Strain" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Strain</th>
<th id="Attribute" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Attribute</th>
<th id="Organism.Name" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Organism.Name</th>
<th id="ANI.Best.ANI.match.ANI" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">ANI.Best.ANI.match.ANI</th>
<th id="ANI.Best.ANI.match.Assembly" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">ANI.Best.ANI.match.Assembly</th>
<th id="ANI.Best.ANI.match.Assembly.Coverage" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">ANI.Best.ANI.match.Assembly.Coverage</th>
<th id="ANI.Best.ANI.match.Type.Category" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">ANI.Best.ANI.match.Type.Category</th>
<th id="ANI.Best.ANI.match.Organism" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">ANI.Best.ANI.match.Organism</th>
<th id="ANI.Best.ANI.match.Type.Assembly.Coverage" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">ANI.Best.ANI.match.Type.Assembly.Coverage</th>
<th id="ANI.Best.match.status" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">ANI.Best.match.status</th>
<th id="ANI.Category" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">ANI.Category</th>
<th id="ANI.Check.status" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">ANI.Check.status</th>
<th id="ANI.Comment" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">ANI.Comment</th>
<th id="ANI.Declared.ANI.match.ANI" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">ANI.Declared.ANI.match.ANI</th>
<th id="ANI.Declared.ANI.match.Assembly" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">ANI.Declared.ANI.match.Assembly</th>
<th id="ANI.Declared.ANI.match.Assembly.Coverage" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">ANI.Declared.ANI.match.Assembly.Coverage</th>
<th id="ANI.Declared.ANI.match.Type.Category" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">ANI.Declared.ANI.match.Type.Category</th>
<th id="ANI.Declared.ANI.match.Organism" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">ANI.Declared.ANI.match.Organism</th>
<th id="ANI.Declared.ANI.match.Type.Assembly.Coverage" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">ANI.Declared.ANI.match.Type.Assembly.Coverage</th>
<th id="ANI.Submitted.organism" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">ANI.Submitted.organism</th>
<th id="ANI.Submitted.species" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">ANI.Submitted.species</th>
<th id="CheckM.completeness" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">CheckM.completeness</th>
<th id="CheckM.completeness.percentile" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">CheckM.completeness.percentile</th>
<th id="CheckM.contamination" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">CheckM.contamination</th>
<th id="CheckM.marker.set" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">CheckM.marker.set</th>
<th id="CheckM.marker.set.rank" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">CheckM.marker.set.rank</th>
<th id="CheckM.species.tax.id" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">CheckM.species.tax.id</th>
<th id="CheckM.version" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">CheckM.version</th>
<th id="Accession" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Accession</th>
<th id="Collected.by" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Collected.by</th>
<th id="Collection.date" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Collection.date</th>
<th id="Description.Comment" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Description.Comment</th>
<th id="Description.Organism.Name" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Description.Organism.Name</th>
<th id="Description.Organism.Taxonomic.ID" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">Description.Organism.Taxonomic.ID</th>
<th id="Description.Title" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Description.Title</th>
<th id="Geographic.location" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Geographic.location</th>
<th id="Host" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Host</th>
<th id="Host.disease" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Host.disease</th>
<th id="Isolation.source" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Isolation.source</th>
<th id="Last.updated" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">Last.updated</th>
<th id="Latitude./.Longitude" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Latitude./.Longitude</th>
<th id="Serovar" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Serovar</th>
<th id="Owner.Name" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Owner.Name</th>
<th id="Assembly.BioProject.Accession" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Assembly.BioProject.Accession</th>
<th id="find" class="gt_col_heading gt_columns_bottom_border gt_center" data-quarto-table-cell-role="th" scope="col">find</th>
<th id="source" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">source</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_000007445.1</td>
<td class="gt_row gt_left" headers="Assembly.Level">Complete Genome</td>
<td class="gt_row gt_left" headers="Strain">CFT073</td>
<td class="gt_row gt_left" headers="Attribute">strain : CFT073</td>
<td class="gt_row gt_left" headers="Organism.Name">Escherichia coli CFT073</td>
<td class="gt_row gt_right" headers="ANI.Best.ANI.match.ANI">99.02</td>
<td class="gt_row gt_left" headers="ANI.Best.ANI.match.Assembly">GCA_000013265.1</td>
<td class="gt_row gt_right" headers="ANI.Best.ANI.match.Assembly.Coverage">88.89</td>
<td class="gt_row gt_left" headers="ANI.Best.ANI.match.Type.Category">claderef</td>
<td class="gt_row gt_left" headers="ANI.Best.ANI.match.Organism">Escherichia coli</td>
<td class="gt_row gt_right" headers="ANI.Best.ANI.match.Type.Assembly.Coverage">89.78</td>
<td class="gt_row gt_left" headers="ANI.Best.match.status">species_match</td>
<td class="gt_row gt_left" headers="ANI.Category">category_na</td>
<td class="gt_row gt_left" headers="ANI.Check.status">OK</td>
<td class="gt_row gt_left" headers="ANI.Comment">na</td>
<td class="gt_row gt_right" headers="ANI.Declared.ANI.match.ANI">99.02</td>
<td class="gt_row gt_left" headers="ANI.Declared.ANI.match.Assembly">GCA_000013265.1</td>
<td class="gt_row gt_right" headers="ANI.Declared.ANI.match.Assembly.Coverage">88.89</td>
<td class="gt_row gt_left" headers="ANI.Declared.ANI.match.Type.Category">claderef</td>
<td class="gt_row gt_left" headers="ANI.Declared.ANI.match.Organism">Escherichia coli UTI89</td>
<td class="gt_row gt_right" headers="ANI.Declared.ANI.match.Type.Assembly.Coverage">89.78</td>
<td class="gt_row gt_left" headers="ANI.Submitted.organism">Escherichia coli CFT073</td>
<td class="gt_row gt_left" headers="ANI.Submitted.species">Escherichia coli</td>
<td class="gt_row gt_right" headers="CheckM.completeness">97.28</td>
<td class="gt_row gt_right" headers="CheckM.completeness.percentile">5.92197</td>
<td class="gt_row gt_right" headers="CheckM.contamination">0.41</td>
<td class="gt_row gt_left" headers="CheckM.marker.set">Escherichia coli</td>
<td class="gt_row gt_left" headers="CheckM.marker.set.rank">species</td>
<td class="gt_row gt_right" headers="CheckM.species.tax.id">562</td>
<td class="gt_row gt_left" headers="CheckM.version">v1.2.2</td>
<td class="gt_row gt_left" headers="Accession">SAMN02604094</td>
<td class="gt_row gt_left" headers="Collected.by">NA</td>
<td class="gt_row gt_left" headers="Collection.date">NA</td>
<td class="gt_row gt_left" headers="Description.Comment">NA</td>
<td class="gt_row gt_left" headers="Description.Organism.Name">Escherichia coli CFT073</td>
<td class="gt_row gt_right" headers="Description.Organism.Taxonomic.ID">199310</td>
<td class="gt_row gt_left" headers="Description.Title">Sample from Escherichia coli CFT073</td>
<td class="gt_row gt_left" headers="Geographic.location">NA</td>
<td class="gt_row gt_left" headers="Host">NA</td>
<td class="gt_row gt_left" headers="Host.disease">NA</td>
<td class="gt_row gt_left" headers="Isolation.source">NA</td>
<td class="gt_row gt_right" headers="Last.updated">2015-05-18 13:20:05</td>
<td class="gt_row gt_left" headers="Latitude./.Longitude">NA</td>
<td class="gt_row gt_left" headers="Serovar">NA</td>
<td class="gt_row gt_left" headers="Owner.Name">NCBI</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJNA313</td>
<td class="gt_row gt_center" headers="find">TRUE</td>
<td class="gt_row gt_left" headers="source">reference</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_000010385.1</td>
<td class="gt_row gt_left" headers="Assembly.Level">Complete Genome</td>
<td class="gt_row gt_left" headers="Strain">SE11</td>
<td class="gt_row gt_left" headers="Attribute">sample_name : AP009240-AP009246; strain : SE11</td>
<td class="gt_row gt_left" headers="Organism.Name">Escherichia coli SE11</td>
<td class="gt_row gt_right" headers="ANI.Best.ANI.match.ANI">98.90</td>
<td class="gt_row gt_left" headers="ANI.Best.ANI.match.Assembly">GCA_000263895.1</td>
<td class="gt_row gt_right" headers="ANI.Best.ANI.match.Assembly.Coverage">86.71</td>
<td class="gt_row gt_left" headers="ANI.Best.ANI.match.Type.Category">claderef</td>
<td class="gt_row gt_left" headers="ANI.Best.ANI.match.Organism">Escherichia coli</td>
<td class="gt_row gt_right" headers="ANI.Best.ANI.match.Type.Assembly.Coverage">84.98</td>
<td class="gt_row gt_left" headers="ANI.Best.match.status">species_match</td>
<td class="gt_row gt_left" headers="ANI.Category">claderef</td>
<td class="gt_row gt_left" headers="ANI.Check.status">OK</td>
<td class="gt_row gt_left" headers="ANI.Comment">na</td>
<td class="gt_row gt_right" headers="ANI.Declared.ANI.match.ANI">98.90</td>
<td class="gt_row gt_left" headers="ANI.Declared.ANI.match.Assembly">GCA_000263895.1</td>
<td class="gt_row gt_right" headers="ANI.Declared.ANI.match.Assembly.Coverage">86.71</td>
<td class="gt_row gt_left" headers="ANI.Declared.ANI.match.Type.Category">claderef</td>
<td class="gt_row gt_left" headers="ANI.Declared.ANI.match.Organism">Escherichia coli O103:H25 str. CVM9340</td>
<td class="gt_row gt_right" headers="ANI.Declared.ANI.match.Type.Assembly.Coverage">84.98</td>
<td class="gt_row gt_left" headers="ANI.Submitted.organism">Escherichia coli SE11</td>
<td class="gt_row gt_left" headers="ANI.Submitted.species">Escherichia coli</td>
<td class="gt_row gt_right" headers="CheckM.completeness">99.52</td>
<td class="gt_row gt_right" headers="CheckM.completeness.percentile">94.82860</td>
<td class="gt_row gt_right" headers="CheckM.contamination">0.21</td>
<td class="gt_row gt_left" headers="CheckM.marker.set">Escherichia coli</td>
<td class="gt_row gt_left" headers="CheckM.marker.set.rank">species</td>
<td class="gt_row gt_right" headers="CheckM.species.tax.id">562</td>
<td class="gt_row gt_left" headers="CheckM.version">v1.2.2</td>
<td class="gt_row gt_left" headers="Accession">SAMD00061087</td>
<td class="gt_row gt_left" headers="Collected.by">NA</td>
<td class="gt_row gt_left" headers="Collection.date">NA</td>
<td class="gt_row gt_left" headers="Description.Comment">NA</td>
<td class="gt_row gt_left" headers="Description.Organism.Name">Escherichia coli SE11</td>
<td class="gt_row gt_right" headers="Description.Organism.Taxonomic.ID">409438</td>
<td class="gt_row gt_left" headers="Description.Title">Generic sample from Escherichia coli SE11</td>
<td class="gt_row gt_left" headers="Geographic.location">NA</td>
<td class="gt_row gt_left" headers="Host">NA</td>
<td class="gt_row gt_left" headers="Host.disease">NA</td>
<td class="gt_row gt_left" headers="Isolation.source">NA</td>
<td class="gt_row gt_right" headers="Last.updated">2016-09-26 20:10:59</td>
<td class="gt_row gt_left" headers="Latitude./.Longitude">NA</td>
<td class="gt_row gt_left" headers="Serovar">NA</td>
<td class="gt_row gt_left" headers="Owner.Name">DDBJ</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJNA18057</td>
<td class="gt_row gt_center" headers="find">TRUE</td>
<td class="gt_row gt_left" headers="source">reference</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_000010485.1</td>
<td class="gt_row gt_left" headers="Assembly.Level">Complete Genome</td>
<td class="gt_row gt_left" headers="Strain">SE15</td>
<td class="gt_row gt_left" headers="Attribute">sample_name : AP009378-AP009379; strain : SE15</td>
<td class="gt_row gt_left" headers="Organism.Name">Escherichia coli SE15</td>
<td class="gt_row gt_right" headers="ANI.Best.ANI.match.ANI">99.56</td>
<td class="gt_row gt_left" headers="ANI.Best.ANI.match.Assembly">GCA_024519395.1</td>
<td class="gt_row gt_right" headers="ANI.Best.ANI.match.Assembly.Coverage">95.73</td>
<td class="gt_row gt_left" headers="ANI.Best.ANI.match.Type.Category">neotype</td>
<td class="gt_row gt_left" headers="ANI.Best.ANI.match.Organism">Escherichia coli</td>
<td class="gt_row gt_right" headers="ANI.Best.ANI.match.Type.Assembly.Coverage">86.67</td>
<td class="gt_row gt_left" headers="ANI.Best.match.status">species_match</td>
<td class="gt_row gt_left" headers="ANI.Category">category_na</td>
<td class="gt_row gt_left" headers="ANI.Check.status">OK</td>
<td class="gt_row gt_left" headers="ANI.Comment">na</td>
<td class="gt_row gt_right" headers="ANI.Declared.ANI.match.ANI">99.56</td>
<td class="gt_row gt_left" headers="ANI.Declared.ANI.match.Assembly">GCA_024519395.1</td>
<td class="gt_row gt_right" headers="ANI.Declared.ANI.match.Assembly.Coverage">95.73</td>
<td class="gt_row gt_left" headers="ANI.Declared.ANI.match.Type.Category">neotype</td>
<td class="gt_row gt_left" headers="ANI.Declared.ANI.match.Organism">Escherichia coli DSM 30083 = JCM 1649 = ATCC 11775</td>
<td class="gt_row gt_right" headers="ANI.Declared.ANI.match.Type.Assembly.Coverage">86.67</td>
<td class="gt_row gt_left" headers="ANI.Submitted.organism">Escherichia coli SE15</td>
<td class="gt_row gt_left" headers="ANI.Submitted.species">Escherichia coli</td>
<td class="gt_row gt_right" headers="CheckM.completeness">98.70</td>
<td class="gt_row gt_right" headers="CheckM.completeness.percentile">30.20443</td>
<td class="gt_row gt_right" headers="CheckM.contamination">0.51</td>
<td class="gt_row gt_left" headers="CheckM.marker.set">Escherichia coli</td>
<td class="gt_row gt_left" headers="CheckM.marker.set.rank">species</td>
<td class="gt_row gt_right" headers="CheckM.species.tax.id">562</td>
<td class="gt_row gt_left" headers="CheckM.version">v1.2.2</td>
<td class="gt_row gt_left" headers="Accession">SAMD00060923</td>
<td class="gt_row gt_left" headers="Collected.by">NA</td>
<td class="gt_row gt_left" headers="Collection.date">NA</td>
<td class="gt_row gt_left" headers="Description.Comment">NA</td>
<td class="gt_row gt_left" headers="Description.Organism.Name">Escherichia coli SE15</td>
<td class="gt_row gt_right" headers="Description.Organism.Taxonomic.ID">431946</td>
<td class="gt_row gt_left" headers="Description.Title">Generic sample from Escherichia coli SE15</td>
<td class="gt_row gt_left" headers="Geographic.location">NA</td>
<td class="gt_row gt_left" headers="Host">NA</td>
<td class="gt_row gt_left" headers="Host.disease">NA</td>
<td class="gt_row gt_left" headers="Isolation.source">NA</td>
<td class="gt_row gt_right" headers="Last.updated">2016-09-26 20:10:59</td>
<td class="gt_row gt_left" headers="Latitude./.Longitude">NA</td>
<td class="gt_row gt_left" headers="Serovar">NA</td>
<td class="gt_row gt_left" headers="Owner.Name">DDBJ</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJDA19053</td>
<td class="gt_row gt_center" headers="find">TRUE</td>
<td class="gt_row gt_left" headers="source">reference</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_000010745.1</td>
<td class="gt_row gt_left" headers="Assembly.Level">Complete Genome</td>
<td class="gt_row gt_left" headers="Strain">12009</td>
<td class="gt_row gt_left" headers="Attribute">sample_name : AP010958-AP010959; serovar : O103:H2; strain : 12009</td>
<td class="gt_row gt_left" headers="Organism.Name">Escherichia coli O103:H2 str. 12009</td>
<td class="gt_row gt_right" headers="ANI.Best.ANI.match.ANI">99.03</td>
<td class="gt_row gt_left" headers="ANI.Best.ANI.match.Assembly">GCA_000010385.1</td>
<td class="gt_row gt_right" headers="ANI.Best.ANI.match.Assembly.Coverage">82.63</td>
<td class="gt_row gt_left" headers="ANI.Best.ANI.match.Type.Category">claderef</td>
<td class="gt_row gt_left" headers="ANI.Best.ANI.match.Organism">Escherichia coli</td>
<td class="gt_row gt_right" headers="ANI.Best.ANI.match.Type.Assembly.Coverage">88.55</td>
<td class="gt_row gt_left" headers="ANI.Best.match.status">species_match</td>
<td class="gt_row gt_left" headers="ANI.Category">category_na</td>
<td class="gt_row gt_left" headers="ANI.Check.status">OK</td>
<td class="gt_row gt_left" headers="ANI.Comment">na</td>
<td class="gt_row gt_right" headers="ANI.Declared.ANI.match.ANI">99.03</td>
<td class="gt_row gt_left" headers="ANI.Declared.ANI.match.Assembly">GCA_000010385.1</td>
<td class="gt_row gt_right" headers="ANI.Declared.ANI.match.Assembly.Coverage">82.63</td>
<td class="gt_row gt_left" headers="ANI.Declared.ANI.match.Type.Category">claderef</td>
<td class="gt_row gt_left" headers="ANI.Declared.ANI.match.Organism">Escherichia coli SE11</td>
<td class="gt_row gt_right" headers="ANI.Declared.ANI.match.Type.Assembly.Coverage">88.55</td>
<td class="gt_row gt_left" headers="ANI.Submitted.organism">Escherichia coli O103:H2 str. 12009</td>
<td class="gt_row gt_left" headers="ANI.Submitted.species">Escherichia coli</td>
<td class="gt_row gt_right" headers="CheckM.completeness">99.56</td>
<td class="gt_row gt_right" headers="CheckM.completeness.percentile">99.95963</td>
<td class="gt_row gt_right" headers="CheckM.contamination">0.15</td>
<td class="gt_row gt_left" headers="CheckM.marker.set">Escherichia coli</td>
<td class="gt_row gt_left" headers="CheckM.marker.set.rank">species</td>
<td class="gt_row gt_right" headers="CheckM.species.tax.id">562</td>
<td class="gt_row gt_left" headers="CheckM.version">v1.2.2</td>
<td class="gt_row gt_left" headers="Accession">SAMD00060956</td>
<td class="gt_row gt_left" headers="Collected.by">NA</td>
<td class="gt_row gt_left" headers="Collection.date">NA</td>
<td class="gt_row gt_left" headers="Description.Comment">NA</td>
<td class="gt_row gt_left" headers="Description.Organism.Name">Escherichia coli O103:H2 str. 12009</td>
<td class="gt_row gt_right" headers="Description.Organism.Taxonomic.ID">585395</td>
<td class="gt_row gt_left" headers="Description.Title">Generic sample from Escherichia coli O103:H2 str. 12009</td>
<td class="gt_row gt_left" headers="Geographic.location">NA</td>
<td class="gt_row gt_left" headers="Host">NA</td>
<td class="gt_row gt_left" headers="Host.disease">NA</td>
<td class="gt_row gt_left" headers="Isolation.source">NA</td>
<td class="gt_row gt_right" headers="Last.updated">2016-09-26 20:10:59</td>
<td class="gt_row gt_left" headers="Latitude./.Longitude">NA</td>
<td class="gt_row gt_left" headers="Serovar">O103:H2</td>
<td class="gt_row gt_left" headers="Owner.Name">DDBJ</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJDA32511</td>
<td class="gt_row gt_center" headers="find">TRUE</td>
<td class="gt_row gt_left" headers="source">reference</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_000010765.1</td>
<td class="gt_row gt_left" headers="Assembly.Level">Complete Genome</td>
<td class="gt_row gt_left" headers="Strain">11128</td>
<td class="gt_row gt_left" headers="Attribute">sample_name : AP010960-AP010965; serovar : O111:H-; strain : 11128</td>
<td class="gt_row gt_left" headers="Organism.Name">Escherichia coli O111:H- str. 11128</td>
<td class="gt_row gt_right" headers="ANI.Best.ANI.match.ANI">98.96</td>
<td class="gt_row gt_left" headers="ANI.Best.ANI.match.Assembly">GCA_000263895.1</td>
<td class="gt_row gt_right" headers="ANI.Best.ANI.match.Assembly.Coverage">81.81</td>
<td class="gt_row gt_left" headers="ANI.Best.ANI.match.Type.Category">claderef</td>
<td class="gt_row gt_left" headers="ANI.Best.ANI.match.Organism">Escherichia coli</td>
<td class="gt_row gt_right" headers="ANI.Best.ANI.match.Type.Assembly.Coverage">89.68</td>
<td class="gt_row gt_left" headers="ANI.Best.match.status">species_match</td>
<td class="gt_row gt_left" headers="ANI.Category">category_na</td>
<td class="gt_row gt_left" headers="ANI.Check.status">OK</td>
<td class="gt_row gt_left" headers="ANI.Comment">na</td>
<td class="gt_row gt_right" headers="ANI.Declared.ANI.match.ANI">98.96</td>
<td class="gt_row gt_left" headers="ANI.Declared.ANI.match.Assembly">GCA_000263895.1</td>
<td class="gt_row gt_right" headers="ANI.Declared.ANI.match.Assembly.Coverage">81.81</td>
<td class="gt_row gt_left" headers="ANI.Declared.ANI.match.Type.Category">claderef</td>
<td class="gt_row gt_left" headers="ANI.Declared.ANI.match.Organism">Escherichia coli O103:H25 str. CVM9340</td>
<td class="gt_row gt_right" headers="ANI.Declared.ANI.match.Type.Assembly.Coverage">89.68</td>
<td class="gt_row gt_left" headers="ANI.Submitted.organism">Escherichia coli O111:H_ str. 11128</td>
<td class="gt_row gt_left" headers="ANI.Submitted.species">Escherichia coli</td>
<td class="gt_row gt_right" headers="CheckM.completeness">99.08</td>
<td class="gt_row gt_right" headers="CheckM.completeness.percentile">57.10379</td>
<td class="gt_row gt_right" headers="CheckM.contamination">1.19</td>
<td class="gt_row gt_left" headers="CheckM.marker.set">Escherichia coli</td>
<td class="gt_row gt_left" headers="CheckM.marker.set.rank">species</td>
<td class="gt_row gt_right" headers="CheckM.species.tax.id">562</td>
<td class="gt_row gt_left" headers="CheckM.version">v1.2.2</td>
<td class="gt_row gt_left" headers="Accession">SAMD00060957</td>
<td class="gt_row gt_left" headers="Collected.by">NA</td>
<td class="gt_row gt_left" headers="Collection.date">NA</td>
<td class="gt_row gt_left" headers="Description.Comment">NA</td>
<td class="gt_row gt_left" headers="Description.Organism.Name">Escherichia coli O111:H- str. 11128</td>
<td class="gt_row gt_right" headers="Description.Organism.Taxonomic.ID">585396</td>
<td class="gt_row gt_left" headers="Description.Title">Generic sample from Escherichia coli O111:H- str. 11128</td>
<td class="gt_row gt_left" headers="Geographic.location">NA</td>
<td class="gt_row gt_left" headers="Host">NA</td>
<td class="gt_row gt_left" headers="Host.disease">NA</td>
<td class="gt_row gt_left" headers="Isolation.source">NA</td>
<td class="gt_row gt_right" headers="Last.updated">2016-09-26 20:10:59</td>
<td class="gt_row gt_left" headers="Latitude./.Longitude">NA</td>
<td class="gt_row gt_left" headers="Serovar">O111:H-</td>
<td class="gt_row gt_left" headers="Owner.Name">DDBJ</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJDA32513</td>
<td class="gt_row gt_center" headers="find">TRUE</td>
<td class="gt_row gt_left" headers="source">reference</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<p>Para descargar los archivos vamos a utilizar el siguiente código:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> g <span class="kw">in</span> <span class="dt">\$</span><span class="er">(</span><span class="fu">cut</span> <span class="at">-f1</span> Escherichia_coli_Bovine_Host.tsv <span class="kw">|</span> <span class="fu">awk</span> <span class="st">'NR &gt; 1'</span><span class="kw">);</span> <span class="cf">do</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> <span class="bu">[</span> <span class="ot">-f</span> <span class="dt">\$</span>g.zip <span class="bu">]</span><span class="kw">;</span> <span class="cf">then</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span> <span class="dt">\$</span>g<span class="st">'.zip exists'</span><span class="kw">;</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>            <span class="va">$HOME</span><span class="ex">/bin/datasets</span> download genome accession <span class="dt">\$</span>g <span class="dt">\</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">--include</span> genome <span class="at">--filename</span> <span class="dt">\$</span>g<span class="st">'.zip'</span><span class="kw">;</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>            <span class="fu">sleep</span> 2<span class="kw">;</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>        <span class="cf">fi</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>    <span class="cf">done</span></span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Descomprimir los archivos</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> g <span class="kw">in</span> <span class="dt">\$</span><span class="er">(</span><span class="fu">cut</span> <span class="at">-f1</span> Escherichia_coli_Bovine_Host.tsv <span class="kw">|</span> <span class="fu">awk</span> <span class="st">'NR &gt; 1'</span><span class="kw">);</span> <span class="cf">do</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>        <span class="fu">unzip</span> <span class="dt">\$</span>g.zip <span class="at">-d</span> <span class="dt">\$</span>g<span class="kw">;</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>        <span class="fu">mv</span> <span class="dt">\$</span>g/ncbi_dataset/data/\$g/<span class="pp">*</span>.fna <span class="dt">\$</span>g.fna<span class="kw">;</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>        <span class="fu">rm</span> <span class="at">-r</span> <span class="dt">\$</span>g/<span class="kw">;</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>    <span class="cf">done</span></span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Con esto ya contamos con un sub conjunto de datos listo para trabajar.</p>
</section>
<section id="anotación-de-genomas" class="level1">
<h1>Anotación de genomas</h1>
<section id="prokka" class="level2">
<h2 class="anchored" data-anchor-id="prokka">Prokka</h2>
<p><code>Prokka</code> es un anotador de genomas para organismos procariontes, lo puedes descargar de <a href="https://github.com/tseemann/prokka">aquí</a>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="va">genomes</span><span class="op">=</span><span class="va">$(</span> <span class="fu">ls</span> <span class="va">$HOME</span>/1_data/<span class="pp">*</span>.fna <span class="va">)</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> g <span class="kw">in</span> <span class="va">$genomes</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a> <span class="ex">prokka</span> <span class="dt">\</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a> <span class="va">$g</span> <span class="dt">\</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a> <span class="at">--cpus</span> 6 <span class="dt">\</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a> <span class="at">--outdir</span> prokka/<span class="va">${id</span><span class="op">%</span>.fna<span class="va">}</span> <span class="dt">\</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a> <span class="at">--genus</span> Escherichia</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a> </span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="bakta" class="level2">
<h2 class="anchored" data-anchor-id="bakta">Bakta</h2>
<p><strong>Precaución</strong> ⚠️: Necesitas 8GB de RAM, disponibilidad de tiempo y memoria en tu máquina, con más de 10 genomas podría demorar según la capacidad de nucleos que tengas. Para este tutorial solo te mostraremos como es el código pero no se utilizará este paso para los siguientes análisis.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copiado");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copiado");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>