<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="es" xml:lang="es"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.42">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="OMICs Analysis">

<title>Análisis Genomas: E. coli</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="index_files/libs/clipboard/clipboard.min.js"></script>
<script src="index_files/libs/quarto-html/quarto.js"></script>
<script src="index_files/libs/quarto-html/popper.min.js"></script>
<script src="index_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="index_files/libs/quarto-html/anchor.min.js"></script>
<link href="index_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="index_files/libs/quarto-html/quarto-syntax-highlighting-2f5df379a58b258e96c21c0638c20c03.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="index_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="index_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="index_files/libs/bootstrap/bootstrap-a312d171523933343ca77b5e293f9fd6.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<style>html{ scroll-behavior: smooth; }</style>


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Contenido</h2>
   
  <ul>
  <li><a href="#introducción" id="toc-introducción" class="nav-link active" data-scroll-target="#introducción">Introducción</a>
  <ul class="collapse">
  <li><a href="#descarga-de-archivos" id="toc-descarga-de-archivos" class="nav-link" data-scroll-target="#descarga-de-archivos">Descarga de archivos</a></li>
  </ul></li>
  <li><a href="#control-de-calidad" id="toc-control-de-calidad" class="nav-link" data-scroll-target="#control-de-calidad">Control de Calidad</a>
  <ul class="collapse">
  <li><a href="#fastani" id="toc-fastani" class="nav-link" data-scroll-target="#fastani">FastANI</a></li>
  <li><a href="#busco" id="toc-busco" class="nav-link" data-scroll-target="#busco">BUSCO</a></li>
  </ul></li>
  <li><a href="#anotación-de-genomas" id="toc-anotación-de-genomas" class="nav-link" data-scroll-target="#anotación-de-genomas">Anotación de genomas</a>
  <ul class="collapse">
  <li><a href="#prokka" id="toc-prokka" class="nav-link" data-scroll-target="#prokka">Prokka</a></li>
  <li><a href="#bakta" id="toc-bakta" class="nav-link" data-scroll-target="#bakta">Bakta</a></li>
  <li><a href="#consejos-para-decidir-cuál-usar-en-tu-pipeline" id="toc-consejos-para-decidir-cuál-usar-en-tu-pipeline" class="nav-link" data-scroll-target="#consejos-para-decidir-cuál-usar-en-tu-pipeline">Consejos para decidir cuál usar en tu pipeline</a></li>
  <li><a href="#amrfinderplus" id="toc-amrfinderplus" class="nav-link" data-scroll-target="#amrfinderplus">AMRfinderPlus</a></li>
  <li><a href="#virulence-factor-database" id="toc-virulence-factor-database" class="nav-link" data-scroll-target="#virulence-factor-database">Virulence Factor DataBase</a></li>
  </ul></li>
  <li><a href="#pangenoma" id="toc-pangenoma" class="nav-link" data-scroll-target="#pangenoma">Pangenoma</a></li>
  <li><a href="#filogénia" id="toc-filogénia" class="nav-link" data-scroll-target="#filogénia">Filogénia</a></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Análisis Genomas: E. coli</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Autor/a</div>
    <div class="quarto-title-meta-contents">
             <p>OMICs Analysis </p>
          </div>
  </div>
    
  
    
  </div>
  


</header>

<nav id="TOC-body" role="doc-toc">
    <h2 id="toc-title">Contenido</h2>
   
  <ul>
  <li><a href="#introducción" id="toc-introducción">Introducción</a>
  <ul>
  <li><a href="#descarga-de-archivos" id="toc-descarga-de-archivos">Descarga de archivos</a></li>
  </ul></li>
  <li><a href="#control-de-calidad" id="toc-control-de-calidad">Control de Calidad</a>
  <ul>
  <li><a href="#fastani" id="toc-fastani">FastANI</a></li>
  <li><a href="#busco" id="toc-busco">BUSCO</a></li>
  </ul></li>
  <li><a href="#anotación-de-genomas" id="toc-anotación-de-genomas">Anotación de genomas</a>
  <ul>
  <li><a href="#prokka" id="toc-prokka">Prokka</a></li>
  <li><a href="#bakta" id="toc-bakta">Bakta</a></li>
  <li><a href="#consejos-para-decidir-cuál-usar-en-tu-pipeline" id="toc-consejos-para-decidir-cuál-usar-en-tu-pipeline">Consejos para decidir cuál usar en tu pipeline</a></li>
  <li><a href="#amrfinderplus" id="toc-amrfinderplus">AMRfinderPlus</a></li>
  <li><a href="#virulence-factor-database" id="toc-virulence-factor-database">Virulence Factor DataBase</a></li>
  </ul></li>
  <li><a href="#pangenoma" id="toc-pangenoma">Pangenoma</a></li>
  <li><a href="#filogénia" id="toc-filogénia">Filogénia</a></li>
  </ul>
</nav>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/logo_OAN_for_white.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="300"></p>
</figure>
</div>
<section id="introducción" class="level1">
<h1>Introducción</h1>
<p>Vamos a trabajar con ensambles de distintas cepas de <em>E. coli</em> hospedero <em>Bovino</em>.</p>
<p><strong>NOTA</strong> ⚠️👁️‍🗨️: Para las figuras del curso se utilizaron todos los 110 genomas con hospedero Bovino. Para este tutorial generaremos un subconjunto de 16, ya incluidos genomas de referencia para patotipos EHEC y EPEC.</p>
<p>Para iniciar vamos a descargar los datos disponibles que existan usando <code>datasets</code> de NCBI, lo puedes instalar <a href="https://github.com/ncbi/datasets">aquí</a>.</p>
<p>Con el siguiente código vamos a descargar los genomas depositados en <strong>NCBI</strong> de <em>E. coli</em></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> 1_data</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> 1_data</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="ex">datasets</span> <span class="dt">\</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>summary genome taxon <span class="st">'562'</span> <span class="dt">\</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>--assembly-level complete<span class="dt">\</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>--assembly-source refseq <span class="dt">\</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>--as-json-lines <span class="op">&gt;</span> Escherichia_coli_562.json</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="ex">dataformat</span> tsv genome <span class="dt">\</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>--inputfile Escherichia_coli_562.json <span class="op">&gt;</span> Escherichia_coli_562.tsv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Este conjunto de metadatos <a href="https://drive.google.com/file/d/19wuoj6hUPrJPmzf6d3KLo_OFpjnNykz4/view?usp=drive_link"><code>Escherichia_coli_562.tsv</code></a> nos servirá para seleccionar los ensambles de genoma que necesitamos, este archivo cuenta con <code>58533</code> filas y <code>174</code> columnas.</p>
<p>Por ello es necesario utilizar la plataforma o método que más prefieras, y quedarnos con un conjunto más pequeño de genomas para descargar.</p>
<p>En este tutorial haremos un ejemplo de parseo de datos con <code>R</code>.</p>
<div class="cell">
<details class="code-fold">
<summary>subsetting.R</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>raw_tsv <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(<span class="st">"0_data/Escherichia_coli_562.tsv"</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(raw_tsv)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Change colnames</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(raw_tsv) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">" "</span>, <span class="st">"."</span>, <span class="fu">colnames</span>(raw_tsv))</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>cols <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Assembly.BioSample.Accession"</span>, <span class="st">"Assembly.BioSample.Attribute.Name"</span>,</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>          <span class="st">"Assembly.BioSample.Attribute.Value"</span>, <span class="st">"Assembly.BioSample.Collected.by"</span>,</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>          <span class="st">"Assembly.BioSample.Collection.date"</span>, <span class="st">"Assembly.BioSample.Description.Comment"</span>,</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>          <span class="st">"Assembly.BioSample.Description.Organism.Name"</span>, <span class="st">"Assembly.BioSample.Description.Organism.Taxonomic.ID"</span>,</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>          <span class="st">"Assembly.BioSample.Description.Title"</span>, <span class="st">"Assembly.BioSample.Geographic.location"</span>,</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>          <span class="st">"Assembly.BioSample.Host"</span>, <span class="st">"Assembly.BioSample.Host.disease"</span>,</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>          <span class="st">"Assembly.BioSample.Isolation.source"</span>, <span class="st">"Assembly.BioSample.Last.updated"</span>,</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>          <span class="st">"Assembly.BioSample.Latitude./.Longitude"</span>, <span class="st">"Assembly.BioSample.Serovar"</span>,</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>          <span class="st">"Assembly.BioSample.Owner.Name"</span>, <span class="st">"Assembly.BioSample.Strain"</span></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>          </span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a>Refseq_for_taxon<span class="ot">&lt;-</span> raw_tsv <span class="sc">|&gt;</span></span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="st">'Assembly.Accession'</span>,<span class="st">"Assembly.Level"</span>,</span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>         <span class="st">'Organism.Name'</span>, <span class="fu">starts_with</span>(<span class="st">'ANI'</span>),<span class="fu">starts_with</span>(<span class="st">'CheckM'</span>), </span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>         cols,</span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a>         <span class="st">"Assembly.BioProject.Accession"</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(ANI.Best.ANI.match.Organism <span class="sc">==</span> <span class="st">"Escherichia coli"</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"GCF"</span>,Assembly.Accession)) <span class="sc">|&gt;</span></span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>()</span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(Refseq_for_taxon) <span class="ot">&lt;-</span> stringr<span class="sc">::</span><span class="fu">str_replace</span>(<span class="fu">colnames</span>(Refseq_for_taxon), </span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a>                                                   <span class="at">pattern =</span> <span class="st">"Assembly.BioSample."</span>,</span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a>                                                   <span class="at">replacement =</span> <span class="st">""</span>)</span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the column 'Assembly.Biosample.Attribute' by combining the values and eliminating duplicates.</span></span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a>ecoli_df <span class="ot">&lt;-</span> Refseq_for_taxon <span class="sc">|&gt;</span></span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(Assembly.Accession) <span class="sc">%&gt;%</span></span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">Attribute =</span> <span class="fu">paste0</span>(</span>
<span id="cb3-43"><a href="#cb3-43" aria-hidden="true" tabindex="-1"></a>      Attribute.Name, <span class="st">" : "</span>, </span>
<span id="cb3-44"><a href="#cb3-44" aria-hidden="true" tabindex="-1"></a>      Attribute.Value, </span>
<span id="cb3-45"><a href="#cb3-45" aria-hidden="true" tabindex="-1"></a>      <span class="at">collapse =</span> <span class="st">"; "</span></span>
<span id="cb3-46"><a href="#cb3-46" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb3-47"><a href="#cb3-47" aria-hidden="true" tabindex="-1"></a>    <span class="at">.groups =</span> <span class="st">"drop"</span></span>
<span id="cb3-48"><a href="#cb3-48" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb3-49"><a href="#cb3-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(</span>
<span id="cb3-50"><a href="#cb3-50" aria-hidden="true" tabindex="-1"></a>    Refseq_for_taxon <span class="sc">%&gt;%</span> </span>
<span id="cb3-51"><a href="#cb3-51" aria-hidden="true" tabindex="-1"></a>      <span class="fu">select</span>(<span class="sc">-</span>Attribute.Name, <span class="sc">-</span>Attribute.Value) <span class="sc">%&gt;%</span> </span>
<span id="cb3-52"><a href="#cb3-52" aria-hidden="true" tabindex="-1"></a>      <span class="fu">distinct</span>(Assembly.Accession, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>),</span>
<span id="cb3-53"><a href="#cb3-53" aria-hidden="true" tabindex="-1"></a>    <span class="at">by =</span> <span class="st">"Assembly.Accession"</span></span>
<span id="cb3-54"><a href="#cb3-54" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb3-55"><a href="#cb3-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-56"><a href="#cb3-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-57"><a href="#cb3-57" aria-hidden="true" tabindex="-1"></a>ecoli_subset <span class="ot">&lt;-</span> ecoli_df <span class="sc">%&gt;%</span> </span>
<span id="cb3-58"><a href="#cb3-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">where</span>(<span class="sc">~</span> <span class="fu">n_distinct</span>(.) <span class="sc">&gt;</span> <span class="dv">1</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb3-59"><a href="#cb3-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">where</span>(<span class="sc">~</span> <span class="sc">!</span><span class="fu">all</span>(<span class="fu">is.na</span>(.)))) <span class="sc">%&gt;%</span> </span>
<span id="cb3-60"><a href="#cb3-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(</span>
<span id="cb3-61"><a href="#cb3-61" aria-hidden="true" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">is.na</span>(CheckM.completeness),</span>
<span id="cb3-62"><a href="#cb3-62" aria-hidden="true" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">is.na</span>(Collection.date),</span>
<span id="cb3-63"><a href="#cb3-63" aria-hidden="true" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">is.na</span>(Host)</span>
<span id="cb3-64"><a href="#cb3-64" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb3-65"><a href="#cb3-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-66"><a href="#cb3-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-67"><a href="#cb3-67" aria-hidden="true" tabindex="-1"></a>ecoli_subset <span class="ot">&lt;-</span> ecoli_subset <span class="sc">%&gt;%</span> </span>
<span id="cb3-68"><a href="#cb3-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">find =</span> CheckM.completeness <span class="sc">&gt;=</span> <span class="dv">90</span> <span class="sc">&amp;</span> CheckM.contamination <span class="sc">&lt;</span> <span class="dv">5</span> <span class="sc">&amp;</span> ANI.Best.ANI.match.ANI <span class="sc">&gt;=</span> <span class="dv">96</span>)</span>
<span id="cb3-69"><a href="#cb3-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-70"><a href="#cb3-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-71"><a href="#cb3-71" aria-hidden="true" tabindex="-1"></a>ecoli_subset <span class="ot">&lt;-</span> ecoli_subset <span class="sc">%&gt;%</span> </span>
<span id="cb3-72"><a href="#cb3-72" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">where</span>(<span class="sc">~</span> <span class="fu">n_distinct</span>(.) <span class="sc">&gt;</span> <span class="dv">1</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb3-73"><a href="#cb3-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">where</span>(<span class="sc">~</span> <span class="sc">!</span><span class="fu">all</span>(<span class="fu">is.na</span>(.)))) <span class="sc">%&gt;%</span> </span>
<span id="cb3-74"><a href="#cb3-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(</span>
<span id="cb3-75"><a href="#cb3-75" aria-hidden="true" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">is.na</span>(CheckM.completeness),</span>
<span id="cb3-76"><a href="#cb3-76" aria-hidden="true" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">is.na</span>(Collection.date),</span>
<span id="cb3-77"><a href="#cb3-77" aria-hidden="true" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">is.na</span>(Host)</span>
<span id="cb3-78"><a href="#cb3-78" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb3-79"><a href="#cb3-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-80"><a href="#cb3-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-81"><a href="#cb3-81" aria-hidden="true" tabindex="-1"></a>ecoli_subset<span class="sc">$</span>Host <span class="ot">&lt;-</span> <span class="fu">tolower</span>(ecoli_subset<span class="sc">$</span>Host)</span>
<span id="cb3-82"><a href="#cb3-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-83"><a href="#cb3-83" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(ecoli_subset<span class="sc">$</span>Host)</span>
<span id="cb3-84"><a href="#cb3-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-85"><a href="#cb3-85" aria-hidden="true" tabindex="-1"></a>ecoli_subset2 <span class="ot">&lt;-</span> ecoli_subset[ecoli_subset<span class="sc">$</span>Host <span class="sc">==</span> <span class="st">"bovine"</span>,]</span>
<span id="cb3-86"><a href="#cb3-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-87"><a href="#cb3-87" aria-hidden="true" tabindex="-1"></a>ecoli_subset2<span class="sc">$</span>Geographic.location <span class="ot">&lt;-</span> <span class="fu">tolower</span>(ecoli_subset2<span class="sc">$</span>Geographic.location)</span>
<span id="cb3-88"><a href="#cb3-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-89"><a href="#cb3-89" aria-hidden="true" tabindex="-1"></a>ecoli_subset2[stringr<span class="sc">::</span><span class="fu">str_detect</span>(ecoli_subset2<span class="sc">$</span>Geographic.location, <span class="at">pattern =</span> <span class="st">"china"</span>),</span>
<span id="cb3-90"><a href="#cb3-90" aria-hidden="true" tabindex="-1"></a>              <span class="st">"Geographic.location"</span>] <span class="ot">&lt;-</span> <span class="st">"china"</span></span>
<span id="cb3-91"><a href="#cb3-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-92"><a href="#cb3-92" aria-hidden="true" tabindex="-1"></a>ecoli_subset2[stringr<span class="sc">::</span><span class="fu">str_detect</span>(ecoli_subset2<span class="sc">$</span>Geographic.location, <span class="at">pattern =</span> <span class="st">"usa"</span>),</span>
<span id="cb3-93"><a href="#cb3-93" aria-hidden="true" tabindex="-1"></a>              <span class="st">"Geographic.location"</span>] <span class="ot">&lt;-</span> <span class="st">"usa"</span></span>
<span id="cb3-94"><a href="#cb3-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-95"><a href="#cb3-95" aria-hidden="true" tabindex="-1"></a>ecoli_subset2 <span class="ot">&lt;-</span> ecoli_subset2[ecoli_subset2<span class="sc">$</span>Geographic.location <span class="sc">!=</span> <span class="st">"missing"</span>,]</span>
<span id="cb3-96"><a href="#cb3-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-97"><a href="#cb3-97" aria-hidden="true" tabindex="-1"></a><span class="co"># Quiero quedarme al azar con 4 de china y 4 de usa, y conservar a francia y suiza</span></span>
<span id="cb3-98"><a href="#cb3-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-99"><a href="#cb3-99" aria-hidden="true" tabindex="-1"></a>ecoli_bovine <span class="ot">&lt;-</span> ecoli_subset2 <span class="sc">%&gt;%</span> </span>
<span id="cb3-100"><a href="#cb3-100" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Geographic.location) <span class="sc">%&gt;%</span> </span>
<span id="cb3-101"><a href="#cb3-101" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">4</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb3-102"><a href="#cb3-102" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb3-103"><a href="#cb3-103" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Assembly.Accession, Strain, Geographic.location,</span>
<span id="cb3-104"><a href="#cb3-104" aria-hidden="true" tabindex="-1"></a>         Host, Owner.Name, Assembly.BioProject.Accession)</span>
<span id="cb3-105"><a href="#cb3-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-106"><a href="#cb3-106" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_tsv</span>(ecoli_bovine, <span class="st">"0_data/Ecoli_bovineHost.tsv"</span>)</span>
<span id="cb3-107"><a href="#cb3-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-108"><a href="#cb3-108" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(ecoli_subset2<span class="sc">$</span>Geographic.location)</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>En este tutorial este será el archivo con el que trabajaremos <a href="https://drive.google.com/file/d/19wuoj6hUPrJPmzf6d3KLo_OFpjnNykz4/view?usp=drive_link"><code>Ecoli_bovineHost.tsv</code></a>:</p>
<div class="cell">
<div class="cell-output-display">
<div id="knqfgbqwjh" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#knqfgbqwjh table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#knqfgbqwjh thead, #knqfgbqwjh tbody, #knqfgbqwjh tfoot, #knqfgbqwjh tr, #knqfgbqwjh td, #knqfgbqwjh th {
  border-style: none;
}

#knqfgbqwjh p {
  margin: 0;
  padding: 0;
}

#knqfgbqwjh .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#knqfgbqwjh .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#knqfgbqwjh .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#knqfgbqwjh .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#knqfgbqwjh .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#knqfgbqwjh .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#knqfgbqwjh .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#knqfgbqwjh .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#knqfgbqwjh .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#knqfgbqwjh .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#knqfgbqwjh .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#knqfgbqwjh .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#knqfgbqwjh .gt_spanner_row {
  border-bottom-style: hidden;
}

#knqfgbqwjh .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#knqfgbqwjh .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#knqfgbqwjh .gt_from_md > :first-child {
  margin-top: 0;
}

#knqfgbqwjh .gt_from_md > :last-child {
  margin-bottom: 0;
}

#knqfgbqwjh .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#knqfgbqwjh .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#knqfgbqwjh .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#knqfgbqwjh .gt_row_group_first td {
  border-top-width: 2px;
}

#knqfgbqwjh .gt_row_group_first th {
  border-top-width: 2px;
}

#knqfgbqwjh .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#knqfgbqwjh .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#knqfgbqwjh .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#knqfgbqwjh .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#knqfgbqwjh .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#knqfgbqwjh .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#knqfgbqwjh .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#knqfgbqwjh .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#knqfgbqwjh .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#knqfgbqwjh .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#knqfgbqwjh .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#knqfgbqwjh .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#knqfgbqwjh .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#knqfgbqwjh .gt_left {
  text-align: left;
}

#knqfgbqwjh .gt_center {
  text-align: center;
}

#knqfgbqwjh .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#knqfgbqwjh .gt_font_normal {
  font-weight: normal;
}

#knqfgbqwjh .gt_font_bold {
  font-weight: bold;
}

#knqfgbqwjh .gt_font_italic {
  font-style: italic;
}

#knqfgbqwjh .gt_super {
  font-size: 65%;
}

#knqfgbqwjh .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#knqfgbqwjh .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#knqfgbqwjh .gt_indent_1 {
  text-indent: 5px;
}

#knqfgbqwjh .gt_indent_2 {
  text-indent: 10px;
}

#knqfgbqwjh .gt_indent_3 {
  text-indent: 15px;
}

#knqfgbqwjh .gt_indent_4 {
  text-indent: 20px;
}

#knqfgbqwjh .gt_indent_5 {
  text-indent: 25px;
}

#knqfgbqwjh .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#knqfgbqwjh div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead>
<tr class="gt_col_headings header">
<th id="Assembly.Accession" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Assembly.Accession</th>
<th id="Strain" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Strain</th>
<th id="Geographic.location" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Geographic.location</th>
<th id="Source" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Source</th>
<th id="Owner.Name" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Owner.Name</th>
<th id="Assembly.BioProject.Accession" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Assembly.BioProject.Accession</th>
<th id="Patotype" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Patotype</th>
<th id="Phylogroup" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Phylogroup</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_029625135.1</td>
<td class="gt_row gt_left" headers="Strain">E47</td>
<td class="gt_row gt_left" headers="Geographic.location">china</td>
<td class="gt_row gt_left" headers="Source">bovine</td>
<td class="gt_row gt_left" headers="Owner.Name">Chinese Academy of Agricultural Sciences</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJNA949005</td>
<td class="gt_row gt_left" headers="Patotype">NA</td>
<td class="gt_row gt_left" headers="Phylogroup">NA</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_025908275.1</td>
<td class="gt_row gt_left" headers="Strain">TL-14</td>
<td class="gt_row gt_left" headers="Geographic.location">china</td>
<td class="gt_row gt_left" headers="Source">bovine</td>
<td class="gt_row gt_left" headers="Owner.Name">Inner Mongolia Minzu University</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJNA890057</td>
<td class="gt_row gt_left" headers="Patotype">NA</td>
<td class="gt_row gt_left" headers="Phylogroup">NA</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_029319035.1</td>
<td class="gt_row gt_left" headers="Strain">E94</td>
<td class="gt_row gt_left" headers="Geographic.location">china</td>
<td class="gt_row gt_left" headers="Source">bovine</td>
<td class="gt_row gt_left" headers="Owner.Name">Chinese Academy of Agricultural Sciences</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJNA942474</td>
<td class="gt_row gt_left" headers="Patotype">NA</td>
<td class="gt_row gt_left" headers="Phylogroup">NA</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_029227895.1</td>
<td class="gt_row gt_left" headers="Strain">ZYB62</td>
<td class="gt_row gt_left" headers="Geographic.location">china</td>
<td class="gt_row gt_left" headers="Source">bovine</td>
<td class="gt_row gt_left" headers="Owner.Name">Lanzhou Institute of Husbandry and Pharmaceutical Sciences, Chinese Academy of Agricultural Science</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJNA727665</td>
<td class="gt_row gt_left" headers="Patotype">NA</td>
<td class="gt_row gt_left" headers="Phylogroup">NA</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_009931435.1</td>
<td class="gt_row gt_left" headers="Strain">EC42405</td>
<td class="gt_row gt_left" headers="Geographic.location">france</td>
<td class="gt_row gt_left" headers="Source">bovine</td>
<td class="gt_row gt_left" headers="Owner.Name">Anses</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJNA556131</td>
<td class="gt_row gt_left" headers="Patotype">NA</td>
<td class="gt_row gt_left" headers="Phylogroup">NA</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_004771235.1</td>
<td class="gt_row gt_left" headers="Strain">PF9285</td>
<td class="gt_row gt_left" headers="Geographic.location">switzerland</td>
<td class="gt_row gt_left" headers="Source">bovine</td>
<td class="gt_row gt_left" headers="Owner.Name">University of Bern</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJNA530748</td>
<td class="gt_row gt_left" headers="Patotype">NA</td>
<td class="gt_row gt_left" headers="Phylogroup">NA</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_049267065.1</td>
<td class="gt_row gt_left" headers="Strain">F3_13_1_3</td>
<td class="gt_row gt_left" headers="Geographic.location">usa</td>
<td class="gt_row gt_left" headers="Source">bovine</td>
<td class="gt_row gt_left" headers="Owner.Name">United State Department of Agriculture</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJNA728709</td>
<td class="gt_row gt_left" headers="Patotype">NA</td>
<td class="gt_row gt_left" headers="Phylogroup">NA</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_013168215.1</td>
<td class="gt_row gt_left" headers="Strain">7409</td>
<td class="gt_row gt_left" headers="Geographic.location">usa</td>
<td class="gt_row gt_left" headers="Source">bovine</td>
<td class="gt_row gt_left" headers="Owner.Name">United State Department of Agriculture</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJNA528413</td>
<td class="gt_row gt_left" headers="Patotype">NA</td>
<td class="gt_row gt_left" headers="Phylogroup">NA</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_013167395.1</td>
<td class="gt_row gt_left" headers="Strain">NE122</td>
<td class="gt_row gt_left" headers="Geographic.location">usa</td>
<td class="gt_row gt_left" headers="Source">bovine</td>
<td class="gt_row gt_left" headers="Owner.Name">United State Department of Agriculture</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJNA528413</td>
<td class="gt_row gt_left" headers="Patotype">NA</td>
<td class="gt_row gt_left" headers="Phylogroup">NA</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_048571305.1</td>
<td class="gt_row gt_left" headers="Strain">7_12_30A</td>
<td class="gt_row gt_left" headers="Geographic.location">usa</td>
<td class="gt_row gt_left" headers="Source">bovine</td>
<td class="gt_row gt_left" headers="Owner.Name">United State Department of Agriculture</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJNA728709</td>
<td class="gt_row gt_left" headers="Patotype">NA</td>
<td class="gt_row gt_left" headers="Phylogroup">NA</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_000025165.1</td>
<td class="gt_row gt_left" headers="Strain">CB9615</td>
<td class="gt_row gt_left" headers="Geographic.location">Germany</td>
<td class="gt_row gt_left" headers="Source">reference</td>
<td class="gt_row gt_left" headers="Owner.Name">Nankai University</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJNA224116</td>
<td class="gt_row gt_left" headers="Patotype">EPEC</td>
<td class="gt_row gt_left" headers="Phylogroup">E</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_000026545.1</td>
<td class="gt_row gt_left" headers="Strain">E2348/69</td>
<td class="gt_row gt_left" headers="Geographic.location">UK</td>
<td class="gt_row gt_left" headers="Source">reference</td>
<td class="gt_row gt_left" headers="Owner.Name">The Wellcome Trust Sanger Institute</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJNA224116</td>
<td class="gt_row gt_left" headers="Patotype">EPEC</td>
<td class="gt_row gt_left" headers="Phylogroup">B2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_000026265.1</td>
<td class="gt_row gt_left" headers="Strain">IAI1</td>
<td class="gt_row gt_left" headers="Geographic.location">-</td>
<td class="gt_row gt_left" headers="Source">reference</td>
<td class="gt_row gt_left" headers="Owner.Name">EBI</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJNA33373</td>
<td class="gt_row gt_left" headers="Patotype">Comensal</td>
<td class="gt_row gt_left" headers="Phylogroup">B1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_000005845.2</td>
<td class="gt_row gt_left" headers="Strain">K-12 MG1655</td>
<td class="gt_row gt_left" headers="Geographic.location">usa</td>
<td class="gt_row gt_left" headers="Source">reference</td>
<td class="gt_row gt_left" headers="Owner.Name">Univ. Wisconsin</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJNA225</td>
<td class="gt_row gt_left" headers="Patotype">Comensal</td>
<td class="gt_row gt_left" headers="Phylogroup">A</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_000010745.1</td>
<td class="gt_row gt_left" headers="Strain">12009</td>
<td class="gt_row gt_left" headers="Geographic.location">japan</td>
<td class="gt_row gt_left" headers="Source">reference</td>
<td class="gt_row gt_left" headers="Owner.Name">University of Tokyo</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJDA32511</td>
<td class="gt_row gt_left" headers="Patotype">EHEC</td>
<td class="gt_row gt_left" headers="Phylogroup">B1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_029962285.1</td>
<td class="gt_row gt_left" headers="Strain">EDL933</td>
<td class="gt_row gt_left" headers="Geographic.location">china</td>
<td class="gt_row gt_left" headers="Source">reference</td>
<td class="gt_row gt_left" headers="Owner.Name">East China University of Science and Technology</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJNA963085</td>
<td class="gt_row gt_left" headers="Patotype">EHEC</td>
<td class="gt_row gt_left" headers="Phylogroup">E</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<section id="descarga-de-archivos" class="level3">
<h3 class="anchored" data-anchor-id="descarga-de-archivos">Descarga de archivos</h3>
<p>Para descargar los archivos vamos a utilizar el siguiente código:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> g <span class="kw">in</span> <span class="va">$(</span><span class="fu">cut</span> <span class="at">-f1</span> Ecoli_bovineHost.tsv <span class="kw">|</span> <span class="fu">awk</span> <span class="st">'NR &gt; 1'</span><span class="va">)</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> <span class="bu">[</span> <span class="ot">-f</span> <span class="va">$g</span>.zip <span class="bu">]</span><span class="kw">;</span> <span class="cf">then</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span> <span class="va">$g</span><span class="st">'.zip exists'</span><span class="kw">;</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>            <span class="ex">datasets</span> download genome accession <span class="va">$g</span> <span class="dt">\ </span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>            <span class="ex">--include</span> genome <span class="at">--filename</span> <span class="va">$g</span><span class="st">'.zip'</span><span class="kw">;</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>            <span class="fu">sleep</span> 2<span class="kw">;</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>        <span class="cf">fi</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>    <span class="cf">done</span></span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Descomprimir los archivos</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> g <span class="kw">in</span> <span class="va">$(</span><span class="fu">cut</span> <span class="at">-f1</span> Ecoli_bovineHost.tsv <span class="kw">|</span> <span class="fu">awk</span> <span class="st">'NR &gt; 1'</span><span class="va">)</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>        <span class="fu">unzip</span> <span class="va">$g</span>.zip <span class="at">-d</span> <span class="va">$g</span><span class="kw">;</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>        <span class="fu">mv</span> <span class="va">$g</span>/ncbi_dataset/data/<span class="va">$g</span>/<span class="pp">*</span>.fna <span class="va">$g</span>.fna<span class="kw">;</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>        <span class="fu">rm</span> <span class="at">-r</span> <span class="va">$g</span>/<span class="kw">;</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span> <span class="pp">*</span>.zip</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Con esto ya contamos con un sub conjunto de datos listo para trabajar.</p>
</section>
</section>
<section id="control-de-calidad" class="level1">
<h1>Control de Calidad</h1>
<section id="fastani" class="level2">
<h2 class="anchored" data-anchor-id="fastani">FastANI</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="ex">fastANI</span> <span class="dt">\</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>--ql <span class="va">$HOME</span>/2025_Demo_ecoli/1_QC/ANI/all_genomes.list <span class="dt">\</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>--rl <span class="va">$HOME</span>/2025_Demo_ecoli/1_QC/ANI/all_genomes.list <span class="dt">\</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>-o ecoli_all_to_all.out<span class="dt">\</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>--matrix</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="busco" class="level2">
<h2 class="anchored" data-anchor-id="busco">BUSCO</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> <span class="va">$HOME</span>/bin/busco-env/bin/activate</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="va">genomes</span><span class="op">=</span><span class="va">$(</span><span class="fu">ls</span> <span class="va">$HOME</span>/2025_Demo_ecoli/0_data/<span class="pp">*</span>.fna<span class="va">)</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> g <span class="kw">in</span> <span class="va">$genomes</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>      <span class="va">id</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="va">$g</span> <span class="kw">|</span> <span class="fu">awk</span> <span class="at">-F</span><span class="st">'/'</span> <span class="st">'{print $NF}'</span><span class="va">)</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>      <span class="ex">/usr/bin/time</span> <span class="at">-v</span> <span class="dt">\</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>        busco <span class="dt">\</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">-i</span> <span class="st">"</span><span class="va">$g</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>        <span class="at">-l</span> bacteria_odb10 <span class="dt">\</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>        <span class="at">-m</span> geno <span class="dt">\</span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>        <span class="at">-o</span> 1_QC/busco/<span class="va">${id</span><span class="op">%</span>.fna<span class="va">}</span> <span class="dt">\</span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>        <span class="at">--cpu</span> 8 <span class="dt">\</span></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>        <span class="at">--force</span> </span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>  <span class="cf">done</span> </span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="anotación-de-genomas" class="level1">
<h1>Anotación de genomas</h1>
<section id="prokka" class="level2">
<h2 class="anchored" data-anchor-id="prokka">Prokka</h2>
<p><code>Prokka</code> es un anotador rápido y estándar para genomas procariontes (bacterias, arqueas y virus asociados) que identifica características genómicas (CDS, tRNAs, rRNAs, ncRNAs, etc.), las anota con nombres/IDs y genera archivos listos para análisis posteriores o envío a bases públicas.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="va">genomes</span><span class="op">=</span><span class="va">$(</span> <span class="fu">ls</span> <span class="va">$HOME</span>/2025_Demo_ecoli/1_data/<span class="pp">*</span>.fna <span class="va">)</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> g <span class="kw">in</span> <span class="va">$genomes</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a> <span class="ex">prokka</span> <span class="dt">\</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a> <span class="va">$g</span> <span class="dt">\</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a> <span class="at">--cpus</span> 6 <span class="dt">\</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a> <span class="at">--outdir</span> prokka/<span class="va">${id</span><span class="op">%</span>.fna<span class="va">}</span> <span class="dt">\</span></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a> <span class="at">--genus</span> Escherichia</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li><p><code>genomes=.../*.fna</code> — crea la lista de archivos FASTA de contigs/consensos que serán anotados.</p></li>
<li><p>El <code>for g in $genomes; do ... done</code> recorre cada ensamblaje.</p></li>
<li><p><code>prokka $g</code> — indica a Prokka el archivo de entrada (.fna/.fasta).</p></li>
<li><p><code>--cpus 6</code> — usa 6 hilos para acelerar pasos paralelizables (e.g., búsquedas).</p></li>
<li><p><code>--outdir prokka/${id%.fna}</code> — dirige la salida a una carpeta por genoma (nota: en el snippet que enviaste la variable <code>id</code> <strong>no está definida</strong> dentro del bucle; ver corrección más abajo).</p></li>
<li><p><code>--genus Escherichia</code> — parámetro opcional que ayuda a priorizar nombres/funciones (mejorará la selección de nombres genéricos). (<a href="https://github.com/tseemann/prokka?utm_source=chatgpt.com" title="Prokka - zap: Rapid prokaryotic genome annotation">GitHub</a>)</p></li>
</ul>
<p><strong>Principales archivos que genera Prokka y para qué sirven</strong></p>
<ul>
<li><p><code>&lt;sample&gt;.gff</code> — <em>archivo maestro</em> en formato GFF3 que contiene las características anotadas (coordenadas, tipos, anotaciones). Usarlo para cargar en IGV/Artemis o como entrada a pipelines que esperan GFF. (<a href="https://nf-co.re/modules/prokka?utm_source=chatgpt.com" title="modules/prokka">nf-co.re</a>, <a href="https://stab.st-andrews.ac.uk/wiki/index.php/Prokka?utm_source=chatgpt.com" title="Prokka - wiki">stab.st-andrews.ac.uk</a>)</p></li>
<li><p><code>&lt;sample&gt;.gbk</code> (o <code>.gbk.gz</code>) — archivo en formato GenBank (útil para visualización/depósito y para herramientas que leen GenBank). (<a href="https://nf-co.re/modules/prokka?utm_source=chatgpt.com" title="modules/prokka">nf-co.re</a>)</p></li>
<li><p><code>&lt;sample&gt;.fna</code> — la secuencia de nucleótidos (contigs) usada/normalizada por Prokka (puede incluir cambios en nombres de secuencia). (<a href="https://stab.st-andrews.ac.uk/wiki/index.php/Prokka?utm_source=chatgpt.com" title="Prokka - wiki">stab.st-andrews.ac.uk</a>)</p></li>
<li><p><code>&lt;sample&gt;.ffn</code> — secuencias de nucleótidos de los genes codificantes (CDS) extraídas. (<a href="https://stab.st-andrews.ac.uk/wiki/index.php/Prokka?utm_source=chatgpt.com" title="Prokka - wiki">stab.st-andrews.ac.uk</a>)</p></li>
<li><p><code>&lt;sample&gt;.faa</code> — traducciones (secuencias proteicas) de los CDS; útil para búsquedas funcionales (BLAST/DIAMOND), clustering o análisis de pangenoma. (<a href="https://stab.st-andrews.ac.uk/wiki/index.php/Prokka?utm_source=chatgpt.com" title="Prokka - wiki">stab.st-andrews.ac.uk</a>)</p></li>
<li><p><code>&lt;sample&gt;.tbl</code> — tabla sencilla con información de anotación (útil para envío a GenBank/ENA en algunos workflows). (<a href="https://hpc.nih.gov/apps/prokka.html?utm_source=chatgpt.com" title="prokka on Biowulf">HPC NIH</a>)</p></li>
<li><p><code>&lt;sample&gt;.sqn</code> (opcional) — formato Sequin para enviar a NCBI (si se generan). (<a href="https://hpc.nih.gov/apps/prokka.html?utm_source=chatgpt.com" title="prokka on Biowulf">HPC NIH</a>)</p></li>
<li><p><code>PROKKA_*.log</code> / <code>prokka.err</code> — logs que detallan programas internos y pasos realizados (útil para depuración).</p></li>
</ul>
<p><strong>Notas prácticas y recomendaciones para Prokka</strong></p>
<ul>
<li><p>Si tu entrada es multi-FASTA con múltiples contigs, Prokka generará un GenBank multi-registro con uno por contig.</p></li>
<li><p>Ajusta <code>--kingdom</code>, <code>--genus</code>, <code>--species</code>, <code>--strain</code> para mejorar la nomenclatura.</p></li>
<li><p>Prokka usa varias bases (p.&nbsp;ej. UniProt/Swiss-Prot) y heurísticas; para anotaciones más «exhaustivas» posteriores se pueden re-analizar las proteínas (<code>.faa</code>) con herramientas especializadas. (<a href="https://github.com/tseemann/prokka?utm_source=chatgpt.com" title="Prokka - zap: Rapid prokaryotic genome annotation">GitHub</a>)</p></li>
</ul>
<hr>
</section>
<section id="bakta" class="level2">
<h2 class="anchored" data-anchor-id="bakta">Bakta</h2>
<p><code>Bakta</code> es un anotador diseñado para obtener anotaciones amplias, normalizadas y «dbxref-ricas» (múltiples referencias cruzadas) en genomas bacterianos y plásmidos. Está pensado para producir salidas estandarizadas (incluyendo JSON legible por máquina) y sORFs (small ORFs) además de las características estándar (CDS, tRNA, rRNA, ncRNA, CRISPR, etc.). Usa una base de datos local y herramientas rápidas (p.&nbsp;ej. DIAMOND) para hacer búsquedas de homología. (<a href="https://bakta.readthedocs.io/?utm_source=chatgpt.com" title="Bakta Documentation: Introduction">Bakta Documentation</a>, <a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC8743544/?utm_source=chatgpt.com" title="Bakta: rapid and standardized annotation of bacterial ...">PMC</a>)</p>
<pre><code>source $HOME/bin/bakta-env/bin/activate</code></pre>
<div class="cell">
<details class="code-fold">
<summary>Muestra el código</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="va">genomes</span><span class="op">=</span><span class="va">$(</span><span class="fu">ls</span> <span class="va">$HOME</span>/2025_Demo_ecoli/1_data<span class="va">)</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="va">experiment_file</span><span class="op">=</span><span class="st">"</span><span class="va">$HOME</span><span class="st">/2025_Demo_ecoli/Experiment_Design.tsv"</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="va">genomes</span><span class="op">=</span><span class="va">$(</span><span class="fu">ls</span> <span class="va">$HOME</span>/2025_Demo_ecoli/1_data<span class="va">)</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a><span class="va">experiment_file</span><span class="op">=</span><span class="st">"</span><span class="va">$HOME</span><span class="st">/2025_Demo_ecoli/Experiment_Design.tsv"</span></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> g <span class="kw">in</span> <span class="va">$genomes</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>    <span class="va">id</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="st">"</span><span class="va">$g</span><span class="st">"</span> <span class="kw">|</span> <span class="fu">awk</span> <span class="at">-F</span><span class="st">'/'</span> <span class="st">'{print $NF}'</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">'s/.fna$//'</span><span class="va">)</span></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>    <span class="va">strain</span><span class="op">=</span><span class="va">$(</span><span class="fu">awk</span> <span class="at">-F</span><span class="st">'\t'</span> <span class="at">-v</span> id=<span class="st">"</span><span class="va">$id</span><span class="st">"</span> <span class="st">'$1 == id {print $3}'</span> <span class="st">"</span><span class="va">$experiment_file</span><span class="st">"</span><span class="va">)</span></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>    <span class="ex">bakta</span> <span class="dt">\</span></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">--db</span> <span class="va">$HOME</span>/2025_Demo_ecoli/db <span class="dt">\</span></span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">--output</span> <span class="va">$HOME</span>/2025_Demo_ecoli/2_annotation/bakta/<span class="va">${id}</span> <span class="dt">\</span></span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">--genus</span> Escherichia <span class="dt">\</span></span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">--strain</span> <span class="st">"</span><span class="va">$strain</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">--threads</span> 6 <span class="dt">\</span></span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">--force</span> <span class="va">$HOME</span>/1_data/<span class="va">${g}</span></span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<ul>
<li><p><code>source .../bin/activate</code> — activa el entorno (p.&nbsp;ej. virtualenv/conda) donde está Bakta.</p></li>
<li><p><code>--db $HOME/.../db</code> — ubicación de la base de datos compacta de Bakta (necesaria; puede ocupar decenas de GB). (<a href="https://bakta.readthedocs.io/en/latest/cli/database.html?utm_source=chatgpt.com" title="Database - Bakta Documentation - Read the Docs">Bakta Documentation</a>)</p></li>
<li><p><code>--output ...</code> — carpeta por genoma para guardar resultados.</p></li>
<li><p><code>--genus</code> / <code>--strain</code> — ayudan en la anotación y en la generación de nombres.</p></li>
<li><p><code>--threads 6</code> — número de hilos para paralelizar búsquedas.</p></li>
<li><p><code>--force</code> — sobrescribir salidas existentes.</p></li>
<li><p>Bakta internamente usa búsquedas con DIAMOND/BLAST y una base SQLite con digest y anotaciones preasignadas (esto acelera y estandariza el proceso). (<a href="https://bakta.readthedocs.io/en/latest/cli/database.html?utm_source=chatgpt.com" title="Database - Bakta Documentation - Read the Docs">Bakta Documentation</a>, <a href="https://bactopia.github.io/v3.0.0/bactopia-tools/bakta/?utm_source=chatgpt.com" title="bakta - Bactopia">Bactopia</a>)</p></li>
</ul>
<p><strong>Precauciones de recursos</strong> ⚠️<br>
</p>
<p>Bakta requiere memoria y disco para su base y para los pasos intermedios; es habitual necesitar varios GB de RAM (tu nota de 8 GB es razonable como mínimo) y tiempo cuando se procesan muchos genomas. Si tienes &gt;10 genomas, espera que el tiempo total aumente y planifica en un clúster o máquina con suficientes recursos. (<a href="https://bactopia.github.io/v3.0.0/bactopia-tools/bakta/?utm_source=chatgpt.com" title="bakta - Bactopia">Bactopia</a>, <a href="https://software.cqls.oregonstate.edu/updates/bakta-1.8.0/?utm_source=chatgpt.com" title="Bakta 1.8.0 - CQLS Software Update List">software.cqls.oregonstate.edu</a>)</p>
<p><strong>Archivos y salidas importantes que produce Bakta</strong></p>
<ul>
<li><p><code>annotation.gff</code> / <code>&lt;id&gt;.gff</code> — GFF3 con anotaciones (coordenadas y atributos). (<a href="https://bakta.readthedocs.io/?utm_source=chatgpt.com" title="Bakta Documentation: Introduction">Bakta Documentation</a>)</p></li>
<li><p><code>&lt;id&gt;.gbk</code> — GenBank (cuando se solicita o se genera). (<a href="https://bakta.readthedocs.io/?utm_source=chatgpt.com" title="Bakta Documentation: Introduction">Bakta Documentation</a>)</p></li>
<li><p><code>&lt;id&gt;.faa</code>, <code>&lt;id&gt;.ffn</code> — proteínas y secuencias nucleotídicas de CDS (similares a Prokka). (<a href="https://bakta.readthedocs.io/?utm_source=chatgpt.com" title="Bakta Documentation: Introduction">Bakta Documentation</a>)</p></li>
<li><p><code>report.json</code> — salida JSON rica en metadatos y dbxrefs (útil para pipelines automatizados). Bakta hace énfasis en producir JSON estandarizado que puede alimentar otras herramientas. (<a href="https://bakta.readthedocs.io/?utm_source=chatgpt.com" title="Bakta Documentation: Introduction">Bakta Documentation</a>, <a href="https://github.com/oschwengers/bakta/discussions/300?utm_source=chatgpt.com" title="(Re-)Generating bakta outputs from the JSON only #300">GitHub</a>)</p></li>
<li><p><code>bakta.log</code> / registros — salida estándar y errores para depuración. (<a href="https://bakta.readthedocs.io/en/latest/api/logs.html?utm_source=chatgpt.com" title="Logs - Bakta Documentation - Read the Docs">Bakta Documentation</a>)</p></li>
</ul>
<p><strong>Ventajas frente a Prokka</strong></p>
<ul>
<li><p>Bakta genera metadatos más ricos / dbxrefs y JSON preparado para downstream automático.</p></li>
<li><p>Usa una base precomprimida con mapeos a UniProt (y otras colecciones), lo que suele producir IDs más estandarizados.</p></li>
<li><p>Prokka es muy rápido y ampliamente usado; Bakta tiende a ser más exhaustivo y «estandarizado» pero con requisitos mayores de disco/DB. Para workflows, muchos usuarios ejecutan <strong>Prokka</strong> para anotaciones rápidas y <strong>Bakta</strong> cuando quieren anotaciones más normalizadas y ricas en dbxrefs. (<a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC8743544/?utm_source=chatgpt.com" title="Bakta: rapid and standardized annotation of bacterial ...">PMC</a>, <a href="https://bactopia.github.io/v3.0.0/bactopia-tools/bakta/?utm_source=chatgpt.com" title="bakta - Bactopia">Bactopia</a>)</p></li>
</ul>
</section>
<section id="consejos-para-decidir-cuál-usar-en-tu-pipeline" class="level2">
<h2 class="anchored" data-anchor-id="consejos-para-decidir-cuál-usar-en-tu-pipeline">Consejos para decidir cuál usar en tu pipeline</h2>
<ul>
<li><p>Si quieres <strong>velocidad</strong> y una anotación estándar rápida para muchas muestras: usa <strong>Prokka</strong> y luego procesa los <code>.faa</code> con herramientas funcionales (eg. EggNOG, InterProScan) según necesites. (<a href="https://github.com/tseemann/prokka?utm_source=chatgpt.com" title="Prokka - zap: Rapid prokaryotic genome annotation">GitHub</a>)</p></li>
<li><p>Si quieres <strong>anotaciones más ricas, normalizadas y con dbxrefs/JSON</strong> para downstream automático (p.&nbsp;ej. bases de datos locales, comparaciones taxonómicas o reporte estandarizado), usa <strong>Bakta</strong> (pero planifica la DB y recursos). (<a href="https://bakta.readthedocs.io/?utm_source=chatgpt.com" title="Bakta Documentation: Introduction">Bakta Documentation</a>, <a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC8743544/?utm_source=chatgpt.com" title="Bakta: rapid and standardized annotation of bacterial ...">PMC</a>)</p></li>
</ul>
</section>
<section id="amrfinderplus" class="level2">
<h2 class="anchored" data-anchor-id="amrfinderplus">AMRfinderPlus</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate amrfinder</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> faa <span class="kw">in</span> 2_annotation/prokka/<span class="pp">*</span>/<span class="pp">*</span>.faa<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>  <span class="va">sample</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="st">"</span><span class="va">${faa</span><span class="op">%%</span>.<span class="pp">*</span><span class="va">}</span><span class="st">"</span><span class="va">)</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>  <span class="va">out</span><span class="op">=</span><span class="st">"8_AMR/</span><span class="va">${sample}</span><span class="st">_amrfinder_prot.tsv"</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>  <span class="bu">echo</span> <span class="st">"Running amrfinder -p </span><span class="va">$faa</span><span class="st"> -&gt; </span><span class="va">$out</span><span class="st">"</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>  <span class="ex">amrfinder</span> <span class="at">-p</span> <span class="st">"</span><span class="va">$faa</span><span class="st">"</span> <span class="at">--plus</span> <span class="at">--threads</span> 4 <span class="at">-o</span> <span class="st">"</span><span class="va">$out</span><span class="st">"</span></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>El <code>output</code> de cada genoma se ve de la siguiente manera:</p>
<div class="cell">
<div class="cell-output-display">
<div id="xsqwunzlpr" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#xsqwunzlpr table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#xsqwunzlpr thead, #xsqwunzlpr tbody, #xsqwunzlpr tfoot, #xsqwunzlpr tr, #xsqwunzlpr td, #xsqwunzlpr th {
  border-style: none;
}

#xsqwunzlpr p {
  margin: 0;
  padding: 0;
}

#xsqwunzlpr .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#xsqwunzlpr .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#xsqwunzlpr .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#xsqwunzlpr .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#xsqwunzlpr .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#xsqwunzlpr .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#xsqwunzlpr .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#xsqwunzlpr .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#xsqwunzlpr .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#xsqwunzlpr .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#xsqwunzlpr .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#xsqwunzlpr .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#xsqwunzlpr .gt_spanner_row {
  border-bottom-style: hidden;
}

#xsqwunzlpr .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#xsqwunzlpr .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#xsqwunzlpr .gt_from_md > :first-child {
  margin-top: 0;
}

#xsqwunzlpr .gt_from_md > :last-child {
  margin-bottom: 0;
}

#xsqwunzlpr .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#xsqwunzlpr .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#xsqwunzlpr .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#xsqwunzlpr .gt_row_group_first td {
  border-top-width: 2px;
}

#xsqwunzlpr .gt_row_group_first th {
  border-top-width: 2px;
}

#xsqwunzlpr .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#xsqwunzlpr .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#xsqwunzlpr .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#xsqwunzlpr .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#xsqwunzlpr .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#xsqwunzlpr .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#xsqwunzlpr .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#xsqwunzlpr .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#xsqwunzlpr .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#xsqwunzlpr .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#xsqwunzlpr .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#xsqwunzlpr .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#xsqwunzlpr .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#xsqwunzlpr .gt_left {
  text-align: left;
}

#xsqwunzlpr .gt_center {
  text-align: center;
}

#xsqwunzlpr .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#xsqwunzlpr .gt_font_normal {
  font-weight: normal;
}

#xsqwunzlpr .gt_font_bold {
  font-weight: bold;
}

#xsqwunzlpr .gt_font_italic {
  font-style: italic;
}

#xsqwunzlpr .gt_super {
  font-size: 65%;
}

#xsqwunzlpr .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#xsqwunzlpr .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#xsqwunzlpr .gt_indent_1 {
  text-indent: 5px;
}

#xsqwunzlpr .gt_indent_2 {
  text-indent: 10px;
}

#xsqwunzlpr .gt_indent_3 {
  text-indent: 15px;
}

#xsqwunzlpr .gt_indent_4 {
  text-indent: 20px;
}

#xsqwunzlpr .gt_indent_5 {
  text-indent: 25px;
}

#xsqwunzlpr .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#xsqwunzlpr div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead>
<tr class="gt_col_headings header">
<th id="Protein-id" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Protein id</th>
<th id="Element-symbol" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Element symbol</th>
<th id="Element-name" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Element name</th>
<th id="Scope" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Scope</th>
<th id="Type" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Type</th>
<th id="Subtype" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Subtype</th>
<th id="Class" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Class</th>
<th id="Subclass" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Subclass</th>
<th id="Method" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Method</th>
<th id="Target-length" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">Target length</th>
<th id="Reference-sequence-length" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">Reference sequence length</th>
<th id="a%-Coverage-of-reference" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">% Coverage of reference</th>
<th id="a%-Identity-to-reference" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">% Identity to reference</th>
<th id="Alignment-length" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">Alignment length</th>
<th id="Closest-reference-accession" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Closest reference accession</th>
<th id="Closest-reference-name" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Closest reference name</th>
<th id="HMM-accession" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">HMM accession</th>
<th id="HMM-description" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">HMM description</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="Protein id">PICPPNBC_00538</td>
<td class="gt_row gt_left" headers="Element symbol">emrE</td>
<td class="gt_row gt_left" headers="Element name">multidrug efflux SMR transporter EmrE</td>
<td class="gt_row gt_left" headers="Scope">plus</td>
<td class="gt_row gt_left" headers="Type">STRESS</td>
<td class="gt_row gt_left" headers="Subtype">BIOCIDE</td>
<td class="gt_row gt_left" headers="Class">EFFLUX</td>
<td class="gt_row gt_left" headers="Subclass">EFFLUX</td>
<td class="gt_row gt_left" headers="Method">EXACTP</td>
<td class="gt_row gt_right" headers="Target length">110</td>
<td class="gt_row gt_right" headers="Reference sequence length">110</td>
<td class="gt_row gt_right" headers="% Coverage of reference">100</td>
<td class="gt_row gt_right" headers="% Identity to reference">100.00</td>
<td class="gt_row gt_right" headers="Alignment length">110</td>
<td class="gt_row gt_left" headers="Closest reference accession">CAA77936.1</td>
<td class="gt_row gt_left" headers="Closest reference name">multidrug efflux SMR transporter EmrE</td>
<td class="gt_row gt_left" headers="HMM accession">NF042984.1</td>
<td class="gt_row gt_left" headers="HMM description">multidrug efflux SMR transporter EmrE</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Protein id">PICPPNBC_01161</td>
<td class="gt_row gt_left" headers="Element symbol">ariR</td>
<td class="gt_row gt_left" headers="Element name">biofilm/acid-resistance regulator AriR</td>
<td class="gt_row gt_left" headers="Scope">plus</td>
<td class="gt_row gt_left" headers="Type">STRESS</td>
<td class="gt_row gt_left" headers="Subtype">ACID</td>
<td class="gt_row gt_left" headers="Class">NA</td>
<td class="gt_row gt_left" headers="Subclass">NA</td>
<td class="gt_row gt_left" headers="Method">EXACTP</td>
<td class="gt_row gt_right" headers="Target length">88</td>
<td class="gt_row gt_right" headers="Reference sequence length">88</td>
<td class="gt_row gt_right" headers="% Coverage of reference">100</td>
<td class="gt_row gt_right" headers="% Identity to reference">100.00</td>
<td class="gt_row gt_right" headers="Alignment length">88</td>
<td class="gt_row gt_left" headers="Closest reference accession">AAC74250.1</td>
<td class="gt_row gt_left" headers="Closest reference name">biofilm/acid-resistance regulator AriR</td>
<td class="gt_row gt_left" headers="HMM accession">NA</td>
<td class="gt_row gt_left" headers="HMM description">NA</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Protein id">PICPPNBC_01599</td>
<td class="gt_row gt_left" headers="Element symbol">asr</td>
<td class="gt_row gt_left" headers="Element name">acid resistance repetitive basic protein Asr</td>
<td class="gt_row gt_left" headers="Scope">plus</td>
<td class="gt_row gt_left" headers="Type">STRESS</td>
<td class="gt_row gt_left" headers="Subtype">ACID</td>
<td class="gt_row gt_left" headers="Class">NA</td>
<td class="gt_row gt_left" headers="Subclass">NA</td>
<td class="gt_row gt_left" headers="Method">EXACTP</td>
<td class="gt_row gt_right" headers="Target length">102</td>
<td class="gt_row gt_right" headers="Reference sequence length">102</td>
<td class="gt_row gt_right" headers="% Coverage of reference">100</td>
<td class="gt_row gt_right" headers="% Identity to reference">100.00</td>
<td class="gt_row gt_right" headers="Alignment length">102</td>
<td class="gt_row gt_left" headers="Closest reference accession">BAA15331.2</td>
<td class="gt_row gt_left" headers="Closest reference name">acid resistance repetitive basic protein Asr</td>
<td class="gt_row gt_left" headers="HMM accession">NF033636.1</td>
<td class="gt_row gt_left" headers="HMM description">acid resistance repetitive basic protein Asr</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Protein id">PICPPNBC_02975</td>
<td class="gt_row gt_left" headers="Element symbol">sslE</td>
<td class="gt_row gt_left" headers="Element name">lipoprotein metalloprotease SslE</td>
<td class="gt_row gt_left" headers="Scope">plus</td>
<td class="gt_row gt_left" headers="Type">VIRULENCE</td>
<td class="gt_row gt_left" headers="Subtype">VIRULENCE</td>
<td class="gt_row gt_left" headers="Class">NA</td>
<td class="gt_row gt_left" headers="Subclass">NA</td>
<td class="gt_row gt_left" headers="Method">HMM</td>
<td class="gt_row gt_right" headers="Target length">1520</td>
<td class="gt_row gt_right" headers="Reference sequence length">1520</td>
<td class="gt_row gt_right" headers="% Coverage of reference">100</td>
<td class="gt_row gt_right" headers="% Identity to reference">88.03</td>
<td class="gt_row gt_right" headers="Alignment length">1521</td>
<td class="gt_row gt_left" headers="Closest reference accession">ADE89421.1</td>
<td class="gt_row gt_left" headers="Closest reference name">lipoprotein metalloprotease SslE</td>
<td class="gt_row gt_left" headers="HMM accession">NF037973.1</td>
<td class="gt_row gt_left" headers="HMM description">lipoprotein metalloprotease SslE</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Protein id">PICPPNBC_03259</td>
<td class="gt_row gt_left" headers="Element symbol">acrF</td>
<td class="gt_row gt_left" headers="Element name">multidrug efflux RND transporter permease subunit AcrF</td>
<td class="gt_row gt_left" headers="Scope">plus</td>
<td class="gt_row gt_left" headers="Type">AMR</td>
<td class="gt_row gt_left" headers="Subtype">AMR</td>
<td class="gt_row gt_left" headers="Class">EFFLUX</td>
<td class="gt_row gt_left" headers="Subclass">EFFLUX</td>
<td class="gt_row gt_left" headers="Method">EXACTP</td>
<td class="gt_row gt_right" headers="Target length">1034</td>
<td class="gt_row gt_right" headers="Reference sequence length">1034</td>
<td class="gt_row gt_right" headers="% Coverage of reference">100</td>
<td class="gt_row gt_right" headers="% Identity to reference">100.00</td>
<td class="gt_row gt_right" headers="Alignment length">1034</td>
<td class="gt_row gt_left" headers="Closest reference accession">AAC76298.1</td>
<td class="gt_row gt_left" headers="Closest reference name">multidrug efflux RND transporter permease subunit AcrF</td>
<td class="gt_row gt_left" headers="HMM accession">NA</td>
<td class="gt_row gt_left" headers="HMM description">NA</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Protein id">PICPPNBC_03497</td>
<td class="gt_row gt_left" headers="Element symbol">arsR</td>
<td class="gt_row gt_left" headers="Element name">As(III)-sensing metalloregulatory transcriptional repressor ArsR</td>
<td class="gt_row gt_left" headers="Scope">plus</td>
<td class="gt_row gt_left" headers="Type">STRESS</td>
<td class="gt_row gt_left" headers="Subtype">METAL</td>
<td class="gt_row gt_left" headers="Class">ARSENIC</td>
<td class="gt_row gt_left" headers="Subclass">ARSENIC</td>
<td class="gt_row gt_left" headers="Method">EXACTP</td>
<td class="gt_row gt_right" headers="Target length">117</td>
<td class="gt_row gt_right" headers="Reference sequence length">117</td>
<td class="gt_row gt_right" headers="% Coverage of reference">100</td>
<td class="gt_row gt_right" headers="% Identity to reference">100.00</td>
<td class="gt_row gt_right" headers="Alignment length">117</td>
<td class="gt_row gt_left" headers="Closest reference accession">BAE77793.1</td>
<td class="gt_row gt_left" headers="Closest reference name">As(III)-sensing metalloregulatory transcriptional repressor ArsR</td>
<td class="gt_row gt_left" headers="HMM accession">NA</td>
<td class="gt_row gt_left" headers="HMM description">NA</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Protein id">PICPPNBC_03499</td>
<td class="gt_row gt_left" headers="Element symbol">arsC</td>
<td class="gt_row gt_left" headers="Element name">glutaredoxin-dependent arsenate reductase</td>
<td class="gt_row gt_left" headers="Scope">plus</td>
<td class="gt_row gt_left" headers="Type">STRESS</td>
<td class="gt_row gt_left" headers="Subtype">METAL</td>
<td class="gt_row gt_left" headers="Class">ARSENIC</td>
<td class="gt_row gt_left" headers="Subclass">ARSENATE</td>
<td class="gt_row gt_left" headers="Method">BLASTP</td>
<td class="gt_row gt_right" headers="Target length">141</td>
<td class="gt_row gt_right" headers="Reference sequence length">141</td>
<td class="gt_row gt_right" headers="% Coverage of reference">100</td>
<td class="gt_row gt_right" headers="% Identity to reference">94.33</td>
<td class="gt_row gt_right" headers="Alignment length">141</td>
<td class="gt_row gt_left" headers="Closest reference accession">AAA21096.1</td>
<td class="gt_row gt_left" headers="Closest reference name">glutaredoxin-dependent arsenate reductase</td>
<td class="gt_row gt_left" headers="HMM accession">NF007456.1</td>
<td class="gt_row gt_left" headers="HMM description">glutaredoxin-dependent arsenate reductase</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Protein id">PICPPNBC_03676</td>
<td class="gt_row gt_left" headers="Element symbol">emrD</td>
<td class="gt_row gt_left" headers="Element name">multidrug efflux MFS transporter EmrD</td>
<td class="gt_row gt_left" headers="Scope">plus</td>
<td class="gt_row gt_left" headers="Type">AMR</td>
<td class="gt_row gt_left" headers="Subtype">AMR</td>
<td class="gt_row gt_left" headers="Class">EFFLUX</td>
<td class="gt_row gt_left" headers="Subclass">EFFLUX</td>
<td class="gt_row gt_left" headers="Method">EXACTP</td>
<td class="gt_row gt_right" headers="Target length">394</td>
<td class="gt_row gt_right" headers="Reference sequence length">394</td>
<td class="gt_row gt_right" headers="% Coverage of reference">100</td>
<td class="gt_row gt_right" headers="% Identity to reference">100.00</td>
<td class="gt_row gt_right" headers="Alignment length">394</td>
<td class="gt_row gt_left" headers="Closest reference accession">BAE77620.1</td>
<td class="gt_row gt_left" headers="Closest reference name">multidrug efflux MFS transporter EmrD</td>
<td class="gt_row gt_left" headers="HMM accession">NA</td>
<td class="gt_row gt_left" headers="HMM description">NA</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Protein id">PICPPNBC_03916</td>
<td class="gt_row gt_left" headers="Element symbol">fieF</td>
<td class="gt_row gt_left" headers="Element name">CDF family cation-efflux transporter FieF</td>
<td class="gt_row gt_left" headers="Scope">plus</td>
<td class="gt_row gt_left" headers="Type">STRESS</td>
<td class="gt_row gt_left" headers="Subtype">METAL</td>
<td class="gt_row gt_left" headers="Class">NA</td>
<td class="gt_row gt_left" headers="Subclass">NA</td>
<td class="gt_row gt_left" headers="Method">EXACTP</td>
<td class="gt_row gt_right" headers="Target length">300</td>
<td class="gt_row gt_right" headers="Reference sequence length">300</td>
<td class="gt_row gt_right" headers="% Coverage of reference">100</td>
<td class="gt_row gt_right" headers="% Identity to reference">100.00</td>
<td class="gt_row gt_right" headers="Alignment length">300</td>
<td class="gt_row gt_left" headers="Closest reference accession">BAE77395.1</td>
<td class="gt_row gt_left" headers="Closest reference name">CDF family cation-efflux transporter FieF</td>
<td class="gt_row gt_left" headers="HMM accession">NA</td>
<td class="gt_row gt_left" headers="HMM description">NA</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Protein id">PICPPNBC_04160</td>
<td class="gt_row gt_left" headers="Element symbol">blaEC</td>
<td class="gt_row gt_left" headers="Element name">BlaEC family class C beta-lactamase</td>
<td class="gt_row gt_left" headers="Scope">plus</td>
<td class="gt_row gt_left" headers="Type">AMR</td>
<td class="gt_row gt_left" headers="Subtype">AMR</td>
<td class="gt_row gt_left" headers="Class">BETA-LACTAM</td>
<td class="gt_row gt_left" headers="Subclass">BETA-LACTAM</td>
<td class="gt_row gt_left" headers="Method">EXACTP</td>
<td class="gt_row gt_right" headers="Target length">377</td>
<td class="gt_row gt_right" headers="Reference sequence length">377</td>
<td class="gt_row gt_right" headers="% Coverage of reference">100</td>
<td class="gt_row gt_right" headers="% Identity to reference">100.00</td>
<td class="gt_row gt_right" headers="Alignment length">377</td>
<td class="gt_row gt_left" headers="Closest reference accession">AAC77110.1</td>
<td class="gt_row gt_left" headers="Closest reference name">BlaEC family class C beta-lactamase</td>
<td class="gt_row gt_left" headers="HMM accession">NF000185.2</td>
<td class="gt_row gt_left" headers="HMM description">BlaEC family class C beta-lactamase</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Protein id">PICPPNBC_04349</td>
<td class="gt_row gt_left" headers="Element symbol">mdtM</td>
<td class="gt_row gt_left" headers="Element name">multidrug efflux MFS transporter MdtM</td>
<td class="gt_row gt_left" headers="Scope">plus</td>
<td class="gt_row gt_left" headers="Type">AMR</td>
<td class="gt_row gt_left" headers="Subtype">AMR</td>
<td class="gt_row gt_left" headers="Class">EFFLUX</td>
<td class="gt_row gt_left" headers="Subclass">EFFLUX</td>
<td class="gt_row gt_left" headers="Method">EXACTP</td>
<td class="gt_row gt_right" headers="Target length">410</td>
<td class="gt_row gt_right" headers="Reference sequence length">410</td>
<td class="gt_row gt_right" headers="% Coverage of reference">100</td>
<td class="gt_row gt_right" headers="% Identity to reference">100.00</td>
<td class="gt_row gt_right" headers="Alignment length">410</td>
<td class="gt_row gt_left" headers="Closest reference accession">AAC77293.1</td>
<td class="gt_row gt_left" headers="Closest reference name">multidrug efflux MFS transporter MdtM</td>
<td class="gt_row gt_left" headers="HMM accession">NA</td>
<td class="gt_row gt_left" headers="HMM description">NA</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
</section>
<section id="virulence-factor-database" class="level2">
<h2 class="anchored" data-anchor-id="virulence-factor-database">Virulence Factor DataBase</h2>
<p>Primero vamos a descargar los FASTA de VFDB <a href="https://www.mgc.ac.cn/VFs/download.htm">aquí</a>. Y vamos a crear la base de datos para poder usar posteriormente blastp. <strong>NOTA</strong>: esto lo puedes aplicar a cualquier otra base de datos!!</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="ex">makeblastdb</span> <span class="dt">\</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>-in data/VFDB/VFDB_setB_pro.fas <span class="dt">\</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>-dbtype prot <span class="dt">\</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>-out data/VFDB/vfdb</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="pangenoma" class="level1">
<h1>Pangenoma</h1>
<p>Vamos a utilizar Roary, lo puedes descargar <a href="https://github.com/sanger-pathogens/Roary">aquí</a>.</p>
<p>Preparamos los archivos <code>.gff</code> obtenidos de la anotación con <code>Prokka</code> o <code>Bakta</code> para poder realizar el análisis de pangenoma.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="va">DIR_ORIGEN</span><span class="op">=</span><span class="st">"</span><span class="va">$HOME</span><span class="st">/2025_Demo_ecoli/2_annotation/prokka"</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="va">DIR_DESTINO</span><span class="op">=</span><span class="st">"</span><span class="va">$HOME</span><span class="st">/2025_Demo_ecoli/3_pangenome/data_genomes"</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> carpeta <span class="kw">in</span> <span class="st">"</span><span class="va">$DIR_ORIGEN</span><span class="st">"</span>/<span class="pp">*</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> <span class="kw">[[</span> <span class="ot">-d</span> <span class="st">"</span><span class="va">$carpeta</span><span class="st">"</span> <span class="kw">]];</span> <span class="cf">then</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>            <span class="va">nombre_carpeta</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="st">"</span><span class="va">$carpeta</span><span class="st">"</span><span class="va">)</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Buscar el primer archivo .gff o .gff3 en la carpeta </span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>            <span class="va">archivo_gff</span><span class="op">=</span><span class="va">$(</span><span class="fu">find</span> <span class="st">"</span><span class="va">$carpeta</span><span class="st">"</span> <span class="at">-maxdepth</span> 1 <span class="at">-type</span> f <span class="dt">\(</span> <span class="at">-iname</span> <span class="st">"*.gff"</span> <span class="at">-o</span> <span class="at">-iname</span> <span class="st">"*.gff3"</span> <span class="dt">\)</span> <span class="at">-print</span> <span class="at">-quit</span> <span class="kw">||</span> <span class="fu">true</span><span class="va">)</span></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> <span class="kw">[[</span> <span class="ot">-n</span> <span class="st">"</span><span class="va">$archivo_gff</span><span class="st">"</span> <span class="kw">]];</span> <span class="cf">then</span></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>                <span class="va">destino</span><span class="op">=</span><span class="st">"</span><span class="va">$DIR_DESTINO</span><span class="st">/</span><span class="va">${nombre_carpeta}</span><span class="st">.gff"</span></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>                <span class="fu">cp</span> <span class="at">-f</span> <span class="at">--</span> <span class="st">"</span><span class="va">$archivo_gff</span><span class="st">"</span> <span class="st">"</span><span class="va">$destino</span><span class="st">"</span></span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>                <span class="bu">echo</span> <span class="st">"Archivo </span><span class="va">$archivo_gff</span><span class="st"> copiado como </span><span class="va">$destino</span><span class="st">"</span></span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>            <span class="cf">else</span></span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>                <span class="bu">echo</span> <span class="st">"No se encontró archivo .gff/.gff3 en </span><span class="va">$carpeta</span><span class="st">"</span></span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>            <span class="cf">fi</span></span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>        <span class="cf">fi</span></span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a>    <span class="cf">done</span></span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="va">DIR_ORIGEN</span><span class="op">=</span><span class="st">"</span><span class="va">$HOME</span><span class="st">/2025_Demo_ecoli/2_annotation/prokka"</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="va">DIR_DESTINO</span><span class="op">=</span><span class="st">"</span><span class="va">$HOME</span><span class="st">/2025_Demo_ecoli/3_pangenome/data_genomes"</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="ex">roary</span> <span class="at">-f</span> <span class="st">"</span><span class="va">$DIR_DESTINO</span><span class="st">/pangenome_prokka_prank"</span> <span class="dt">\</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>-e <span class="dt">\</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>-p 8 <span class="dt">\</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>-v <span class="st">"</span><span class="va">$DIR_DESTINO</span><span class="st">"</span>/<span class="pp">*</span>.gff</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="filogénia" class="level1">
<h1>Filogénia</h1>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copiado");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copiado");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




<script src="index_files/libs/quarto-html/zenscroll-min.js"></script>
</body></html>