<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="es" xml:lang="es"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.42">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="OMICs Analysis">

<title>Genómica Comparativa</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="index_files/libs/clipboard/clipboard.min.js"></script>
<script src="index_files/libs/quarto-html/quarto.js"></script>
<script src="index_files/libs/quarto-html/popper.min.js"></script>
<script src="index_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="index_files/libs/quarto-html/anchor.min.js"></script>
<link href="index_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="index_files/libs/quarto-html/quarto-syntax-highlighting-2f5df379a58b258e96c21c0638c20c03.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="index_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="index_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="index_files/libs/bootstrap/bootstrap-a312d171523933343ca77b5e293f9fd6.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<style>html{ scroll-behavior: smooth; }</style>


</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Contenido</h2>
   
  <ul>
  <li><a href="#introducción" id="toc-introducción" class="nav-link active" data-scroll-target="#introducción">Introducción</a>
  <ul class="collapse">
  <li><a href="#descarga-de-archivos" id="toc-descarga-de-archivos" class="nav-link" data-scroll-target="#descarga-de-archivos">Descarga de archivos</a></li>
  </ul></li>
  <li><a href="#control-de-calidad" id="toc-control-de-calidad" class="nav-link" data-scroll-target="#control-de-calidad">Control de Calidad</a>
  <ul class="collapse">
  <li><a href="#fastani" id="toc-fastani" class="nav-link" data-scroll-target="#fastani">FastANI</a></li>
  <li><a href="#busco" id="toc-busco" class="nav-link" data-scroll-target="#busco">BUSCO</a></li>
  </ul></li>
  <li><a href="#anotación-de-genomas" id="toc-anotación-de-genomas" class="nav-link" data-scroll-target="#anotación-de-genomas">Anotación de genomas</a>
  <ul class="collapse">
  <li><a href="#prokka" id="toc-prokka" class="nav-link" data-scroll-target="#prokka">Prokka</a></li>
  <li><a href="#bakta" id="toc-bakta" class="nav-link" data-scroll-target="#bakta">Bakta</a></li>
  <li><a href="#consejos-para-decidir-cuál-usar-en-tu-pipeline" id="toc-consejos-para-decidir-cuál-usar-en-tu-pipeline" class="nav-link" data-scroll-target="#consejos-para-decidir-cuál-usar-en-tu-pipeline">Consejos para decidir cuál usar en tu pipeline</a></li>
  <li><a href="#amrfinderplus" id="toc-amrfinderplus" class="nav-link" data-scroll-target="#amrfinderplus">AMRfinderPlus</a></li>
  <li><a href="#virulence-factor-database" id="toc-virulence-factor-database" class="nav-link" data-scroll-target="#virulence-factor-database">Virulence Factor DataBase</a></li>
  </ul></li>
  <li><a href="#pangenoma" id="toc-pangenoma" class="nav-link" data-scroll-target="#pangenoma">Pangenoma</a>
  <ul class="collapse">
  <li><a href="#visualización" id="toc-visualización" class="nav-link" data-scroll-target="#visualización">Visualización</a></li>
  </ul></li>
  <li><a href="#filogénia" id="toc-filogénia" class="nav-link" data-scroll-target="#filogénia">Filogénia</a></li>
  <li><a href="#apéndice" id="toc-apéndice" class="nav-link" data-scroll-target="#apéndice">Apéndice</a>
  <ul class="collapse">
  <li><a href="#gráficos-de-r" id="toc-gráficos-de-r" class="nav-link" data-scroll-target="#gráficos-de-r">Gráficos de R</a></li>
  </ul></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Genómica Comparativa</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Autor/a</div>
    <div class="quarto-title-meta-contents">
             <p>OMICs Analysis </p>
          </div>
  </div>
    
  
    
  </div>
  


</header>

<nav id="TOC-body" role="doc-toc">
    <h2 id="toc-title">Contenido</h2>
   
  <ul>
  <li><a href="#introducción" id="toc-introducción">Introducción</a>
  <ul>
  <li><a href="#descarga-de-archivos" id="toc-descarga-de-archivos">Descarga de archivos</a></li>
  </ul></li>
  <li><a href="#control-de-calidad" id="toc-control-de-calidad">Control de Calidad</a>
  <ul>
  <li><a href="#fastani" id="toc-fastani">FastANI</a></li>
  <li><a href="#busco" id="toc-busco">BUSCO</a></li>
  </ul></li>
  <li><a href="#anotación-de-genomas" id="toc-anotación-de-genomas">Anotación de genomas</a>
  <ul>
  <li><a href="#prokka" id="toc-prokka">Prokka</a></li>
  <li><a href="#bakta" id="toc-bakta">Bakta</a></li>
  <li><a href="#consejos-para-decidir-cuál-usar-en-tu-pipeline" id="toc-consejos-para-decidir-cuál-usar-en-tu-pipeline">Consejos para decidir cuál usar en tu pipeline</a></li>
  <li><a href="#amrfinderplus" id="toc-amrfinderplus">AMRfinderPlus</a></li>
  <li><a href="#virulence-factor-database" id="toc-virulence-factor-database">Virulence Factor DataBase</a></li>
  </ul></li>
  <li><a href="#pangenoma" id="toc-pangenoma">Pangenoma</a>
  <ul>
  <li><a href="#visualización" id="toc-visualización">Visualización</a></li>
  </ul></li>
  <li><a href="#filogénia" id="toc-filogénia">Filogénia</a></li>
  <li><a href="#apéndice" id="toc-apéndice">Apéndice</a>
  <ul>
  <li><a href="#gráficos-de-r" id="toc-gráficos-de-r">Gráficos de R</a></li>
  </ul></li>
  </ul>
</nav>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/logo_OAN_for_white.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="300"></p>
</figure>
</div>
<section id="introducción" class="level1">
<h1>Introducción</h1>
<p>Vamos a trabajar con ensambles de distintas cepas de <em>E. coli</em> hospedero <em>Bovino</em>.</p>
<p><strong>NOTA</strong> ⚠️👁️‍🗨️: Para las figuras del curso se utilizaron todos los 110 genomas con hospedero Bovino. Para este tutorial generaremos un subconjunto de 16, ya incluidos genomas de referencia para patotipos EHEC y EPEC.</p>
<p>Para iniciar vamos a descargar los datos disponibles que existan usando <code>datasets</code> de NCBI, lo puedes instalar <a href="https://github.com/ncbi/datasets">aquí</a>.</p>
<p>Con el siguiente código vamos a descargar los genomas depositados en <strong>NCBI</strong> de <em>E. coli</em></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> 1_data</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> 1_data</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="ex">datasets</span> <span class="dt">\</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>summary genome taxon <span class="st">'562'</span> <span class="dt">\</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>--assembly-level complete<span class="dt">\</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>--assembly-source refseq <span class="dt">\</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>--as-json-lines <span class="op">&gt;</span> Escherichia_coli_562.json</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="ex">dataformat</span> tsv genome <span class="dt">\</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>--inputfile Escherichia_coli_562.json <span class="op">&gt;</span> Escherichia_coli_562.tsv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Este conjunto de metadatos <a href="https://drive.google.com/file/d/19wuoj6hUPrJPmzf6d3KLo_OFpjnNykz4/view?usp=drive_link"><code>Escherichia_coli_562.tsv</code></a> nos servirá para seleccionar los ensambles de genoma que necesitamos, este archivo cuenta con <code>58533</code> filas y <code>174</code> columnas.</p>
<p>Por ello es necesario utilizar la plataforma o método que más prefieras, y quedarnos con un conjunto más pequeño de genomas para descargar.</p>
<p>En este tutorial haremos un ejemplo de parseo de datos con <code>R</code>.</p>
<div class="cell">
<details class="code-fold">
<summary>subsetting.R</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>raw_tsv <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(<span class="st">"0_data/Escherichia_coli_562.tsv"</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(raw_tsv)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Change colnames</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(raw_tsv) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">" "</span>, <span class="st">"."</span>, <span class="fu">colnames</span>(raw_tsv))</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>cols <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Assembly.BioSample.Accession"</span>, <span class="st">"Assembly.BioSample.Attribute.Name"</span>,</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>          <span class="st">"Assembly.BioSample.Attribute.Value"</span>, <span class="st">"Assembly.BioSample.Collected.by"</span>,</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>          <span class="st">"Assembly.BioSample.Collection.date"</span>, <span class="st">"Assembly.BioSample.Description.Comment"</span>,</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>          <span class="st">"Assembly.BioSample.Description.Organism.Name"</span>, <span class="st">"Assembly.BioSample.Description.Organism.Taxonomic.ID"</span>,</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>          <span class="st">"Assembly.BioSample.Description.Title"</span>, <span class="st">"Assembly.BioSample.Geographic.location"</span>,</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>          <span class="st">"Assembly.BioSample.Host"</span>, <span class="st">"Assembly.BioSample.Host.disease"</span>,</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>          <span class="st">"Assembly.BioSample.Isolation.source"</span>, <span class="st">"Assembly.BioSample.Last.updated"</span>,</span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>          <span class="st">"Assembly.BioSample.Latitude./.Longitude"</span>, <span class="st">"Assembly.BioSample.Serovar"</span>,</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>          <span class="st">"Assembly.BioSample.Owner.Name"</span>, <span class="st">"Assembly.BioSample.Strain"</span></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>          </span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a>Refseq_for_taxon<span class="ot">&lt;-</span> raw_tsv <span class="sc">|&gt;</span></span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="st">'Assembly.Accession'</span>,<span class="st">"Assembly.Level"</span>,</span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>         <span class="st">'Organism.Name'</span>, <span class="fu">starts_with</span>(<span class="st">'ANI'</span>),<span class="fu">starts_with</span>(<span class="st">'CheckM'</span>), </span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>         cols,</span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a>         <span class="st">"Assembly.BioProject.Accession"</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(ANI.Best.ANI.match.Organism <span class="sc">==</span> <span class="st">"Escherichia coli"</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">"GCF"</span>,Assembly.Accession)) <span class="sc">|&gt;</span></span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">distinct</span>()</span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(Refseq_for_taxon) <span class="ot">&lt;-</span> stringr<span class="sc">::</span><span class="fu">str_replace</span>(<span class="fu">colnames</span>(Refseq_for_taxon), </span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a>                                                   <span class="at">pattern =</span> <span class="st">"Assembly.BioSample."</span>,</span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a>                                                   <span class="at">replacement =</span> <span class="st">""</span>)</span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the column 'Assembly.Biosample.Attribute' by combining the values and eliminating duplicates.</span></span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a>ecoli_df <span class="ot">&lt;-</span> Refseq_for_taxon <span class="sc">|&gt;</span></span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(Assembly.Accession) <span class="sc">%&gt;%</span></span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a>    <span class="at">Attribute =</span> <span class="fu">paste0</span>(</span>
<span id="cb3-43"><a href="#cb3-43" aria-hidden="true" tabindex="-1"></a>      Attribute.Name, <span class="st">" : "</span>, </span>
<span id="cb3-44"><a href="#cb3-44" aria-hidden="true" tabindex="-1"></a>      Attribute.Value, </span>
<span id="cb3-45"><a href="#cb3-45" aria-hidden="true" tabindex="-1"></a>      <span class="at">collapse =</span> <span class="st">"; "</span></span>
<span id="cb3-46"><a href="#cb3-46" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb3-47"><a href="#cb3-47" aria-hidden="true" tabindex="-1"></a>    <span class="at">.groups =</span> <span class="st">"drop"</span></span>
<span id="cb3-48"><a href="#cb3-48" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb3-49"><a href="#cb3-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(</span>
<span id="cb3-50"><a href="#cb3-50" aria-hidden="true" tabindex="-1"></a>    Refseq_for_taxon <span class="sc">%&gt;%</span> </span>
<span id="cb3-51"><a href="#cb3-51" aria-hidden="true" tabindex="-1"></a>      <span class="fu">select</span>(<span class="sc">-</span>Attribute.Name, <span class="sc">-</span>Attribute.Value) <span class="sc">%&gt;%</span> </span>
<span id="cb3-52"><a href="#cb3-52" aria-hidden="true" tabindex="-1"></a>      <span class="fu">distinct</span>(Assembly.Accession, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>),</span>
<span id="cb3-53"><a href="#cb3-53" aria-hidden="true" tabindex="-1"></a>    <span class="at">by =</span> <span class="st">"Assembly.Accession"</span></span>
<span id="cb3-54"><a href="#cb3-54" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb3-55"><a href="#cb3-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-56"><a href="#cb3-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-57"><a href="#cb3-57" aria-hidden="true" tabindex="-1"></a>ecoli_subset <span class="ot">&lt;-</span> ecoli_df <span class="sc">%&gt;%</span> </span>
<span id="cb3-58"><a href="#cb3-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">where</span>(<span class="sc">~</span> <span class="fu">n_distinct</span>(.) <span class="sc">&gt;</span> <span class="dv">1</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb3-59"><a href="#cb3-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">where</span>(<span class="sc">~</span> <span class="sc">!</span><span class="fu">all</span>(<span class="fu">is.na</span>(.)))) <span class="sc">%&gt;%</span> </span>
<span id="cb3-60"><a href="#cb3-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(</span>
<span id="cb3-61"><a href="#cb3-61" aria-hidden="true" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">is.na</span>(CheckM.completeness),</span>
<span id="cb3-62"><a href="#cb3-62" aria-hidden="true" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">is.na</span>(Collection.date),</span>
<span id="cb3-63"><a href="#cb3-63" aria-hidden="true" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">is.na</span>(Host)</span>
<span id="cb3-64"><a href="#cb3-64" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb3-65"><a href="#cb3-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-66"><a href="#cb3-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-67"><a href="#cb3-67" aria-hidden="true" tabindex="-1"></a>ecoli_subset <span class="ot">&lt;-</span> ecoli_subset <span class="sc">%&gt;%</span> </span>
<span id="cb3-68"><a href="#cb3-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">find =</span> CheckM.completeness <span class="sc">&gt;=</span> <span class="dv">90</span> <span class="sc">&amp;</span> CheckM.contamination <span class="sc">&lt;</span> <span class="dv">5</span> <span class="sc">&amp;</span> ANI.Best.ANI.match.ANI <span class="sc">&gt;=</span> <span class="dv">96</span>)</span>
<span id="cb3-69"><a href="#cb3-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-70"><a href="#cb3-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-71"><a href="#cb3-71" aria-hidden="true" tabindex="-1"></a>ecoli_subset <span class="ot">&lt;-</span> ecoli_subset <span class="sc">%&gt;%</span> </span>
<span id="cb3-72"><a href="#cb3-72" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">where</span>(<span class="sc">~</span> <span class="fu">n_distinct</span>(.) <span class="sc">&gt;</span> <span class="dv">1</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb3-73"><a href="#cb3-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">where</span>(<span class="sc">~</span> <span class="sc">!</span><span class="fu">all</span>(<span class="fu">is.na</span>(.)))) <span class="sc">%&gt;%</span> </span>
<span id="cb3-74"><a href="#cb3-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(</span>
<span id="cb3-75"><a href="#cb3-75" aria-hidden="true" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">is.na</span>(CheckM.completeness),</span>
<span id="cb3-76"><a href="#cb3-76" aria-hidden="true" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">is.na</span>(Collection.date),</span>
<span id="cb3-77"><a href="#cb3-77" aria-hidden="true" tabindex="-1"></a>    <span class="sc">!</span><span class="fu">is.na</span>(Host)</span>
<span id="cb3-78"><a href="#cb3-78" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb3-79"><a href="#cb3-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-80"><a href="#cb3-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-81"><a href="#cb3-81" aria-hidden="true" tabindex="-1"></a>ecoli_subset<span class="sc">$</span>Host <span class="ot">&lt;-</span> <span class="fu">tolower</span>(ecoli_subset<span class="sc">$</span>Host)</span>
<span id="cb3-82"><a href="#cb3-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-83"><a href="#cb3-83" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(ecoli_subset<span class="sc">$</span>Host)</span>
<span id="cb3-84"><a href="#cb3-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-85"><a href="#cb3-85" aria-hidden="true" tabindex="-1"></a>ecoli_subset2 <span class="ot">&lt;-</span> ecoli_subset[ecoli_subset<span class="sc">$</span>Host <span class="sc">==</span> <span class="st">"bovine"</span>,]</span>
<span id="cb3-86"><a href="#cb3-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-87"><a href="#cb3-87" aria-hidden="true" tabindex="-1"></a>ecoli_subset2<span class="sc">$</span>Geographic.location <span class="ot">&lt;-</span> <span class="fu">tolower</span>(ecoli_subset2<span class="sc">$</span>Geographic.location)</span>
<span id="cb3-88"><a href="#cb3-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-89"><a href="#cb3-89" aria-hidden="true" tabindex="-1"></a>ecoli_subset2[stringr<span class="sc">::</span><span class="fu">str_detect</span>(ecoli_subset2<span class="sc">$</span>Geographic.location, <span class="at">pattern =</span> <span class="st">"china"</span>),</span>
<span id="cb3-90"><a href="#cb3-90" aria-hidden="true" tabindex="-1"></a>              <span class="st">"Geographic.location"</span>] <span class="ot">&lt;-</span> <span class="st">"china"</span></span>
<span id="cb3-91"><a href="#cb3-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-92"><a href="#cb3-92" aria-hidden="true" tabindex="-1"></a>ecoli_subset2[stringr<span class="sc">::</span><span class="fu">str_detect</span>(ecoli_subset2<span class="sc">$</span>Geographic.location, <span class="at">pattern =</span> <span class="st">"usa"</span>),</span>
<span id="cb3-93"><a href="#cb3-93" aria-hidden="true" tabindex="-1"></a>              <span class="st">"Geographic.location"</span>] <span class="ot">&lt;-</span> <span class="st">"usa"</span></span>
<span id="cb3-94"><a href="#cb3-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-95"><a href="#cb3-95" aria-hidden="true" tabindex="-1"></a>ecoli_subset2 <span class="ot">&lt;-</span> ecoli_subset2[ecoli_subset2<span class="sc">$</span>Geographic.location <span class="sc">!=</span> <span class="st">"missing"</span>,]</span>
<span id="cb3-96"><a href="#cb3-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-97"><a href="#cb3-97" aria-hidden="true" tabindex="-1"></a><span class="co"># Quiero quedarme al azar con 4 de china y 4 de usa, y conservar a francia y suiza</span></span>
<span id="cb3-98"><a href="#cb3-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-99"><a href="#cb3-99" aria-hidden="true" tabindex="-1"></a>ecoli_bovine <span class="ot">&lt;-</span> ecoli_subset2 <span class="sc">%&gt;%</span> </span>
<span id="cb3-100"><a href="#cb3-100" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Geographic.location) <span class="sc">%&gt;%</span> </span>
<span id="cb3-101"><a href="#cb3-101" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_sample</span>(<span class="at">n =</span> <span class="dv">4</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb3-102"><a href="#cb3-102" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb3-103"><a href="#cb3-103" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Assembly.Accession, Strain, Geographic.location,</span>
<span id="cb3-104"><a href="#cb3-104" aria-hidden="true" tabindex="-1"></a>         Host, Owner.Name, Assembly.BioProject.Accession)</span>
<span id="cb3-105"><a href="#cb3-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-106"><a href="#cb3-106" aria-hidden="true" tabindex="-1"></a>readr<span class="sc">::</span><span class="fu">write_tsv</span>(ecoli_bovine, <span class="st">"0_data/Ecoli_bovineHost.tsv"</span>)</span>
<span id="cb3-107"><a href="#cb3-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-108"><a href="#cb3-108" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(ecoli_subset2<span class="sc">$</span>Geographic.location)</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>En este tutorial este será el archivo con el que trabajaremos <a href="https://drive.google.com/file/d/19wuoj6hUPrJPmzf6d3KLo_OFpjnNykz4/view?usp=drive_link"><code>Ecoli_bovineHost.tsv</code></a>:</p>
<div class="cell">
<div class="cell-output-display">
<div id="kxidmxxxvu" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#kxidmxxxvu table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#kxidmxxxvu thead, #kxidmxxxvu tbody, #kxidmxxxvu tfoot, #kxidmxxxvu tr, #kxidmxxxvu td, #kxidmxxxvu th {
  border-style: none;
}

#kxidmxxxvu p {
  margin: 0;
  padding: 0;
}

#kxidmxxxvu .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#kxidmxxxvu .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#kxidmxxxvu .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#kxidmxxxvu .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#kxidmxxxvu .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#kxidmxxxvu .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#kxidmxxxvu .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#kxidmxxxvu .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#kxidmxxxvu .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#kxidmxxxvu .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#kxidmxxxvu .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#kxidmxxxvu .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#kxidmxxxvu .gt_spanner_row {
  border-bottom-style: hidden;
}

#kxidmxxxvu .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#kxidmxxxvu .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#kxidmxxxvu .gt_from_md > :first-child {
  margin-top: 0;
}

#kxidmxxxvu .gt_from_md > :last-child {
  margin-bottom: 0;
}

#kxidmxxxvu .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#kxidmxxxvu .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#kxidmxxxvu .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#kxidmxxxvu .gt_row_group_first td {
  border-top-width: 2px;
}

#kxidmxxxvu .gt_row_group_first th {
  border-top-width: 2px;
}

#kxidmxxxvu .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#kxidmxxxvu .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#kxidmxxxvu .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#kxidmxxxvu .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#kxidmxxxvu .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#kxidmxxxvu .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#kxidmxxxvu .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#kxidmxxxvu .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#kxidmxxxvu .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#kxidmxxxvu .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#kxidmxxxvu .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#kxidmxxxvu .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#kxidmxxxvu .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#kxidmxxxvu .gt_left {
  text-align: left;
}

#kxidmxxxvu .gt_center {
  text-align: center;
}

#kxidmxxxvu .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#kxidmxxxvu .gt_font_normal {
  font-weight: normal;
}

#kxidmxxxvu .gt_font_bold {
  font-weight: bold;
}

#kxidmxxxvu .gt_font_italic {
  font-style: italic;
}

#kxidmxxxvu .gt_super {
  font-size: 65%;
}

#kxidmxxxvu .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#kxidmxxxvu .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#kxidmxxxvu .gt_indent_1 {
  text-indent: 5px;
}

#kxidmxxxvu .gt_indent_2 {
  text-indent: 10px;
}

#kxidmxxxvu .gt_indent_3 {
  text-indent: 15px;
}

#kxidmxxxvu .gt_indent_4 {
  text-indent: 20px;
}

#kxidmxxxvu .gt_indent_5 {
  text-indent: 25px;
}

#kxidmxxxvu .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#kxidmxxxvu div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead>
<tr class="gt_col_headings header">
<th id="Assembly.Accession" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Assembly.Accession</th>
<th id="Strain" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Strain</th>
<th id="Geographic.location" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Geographic.location</th>
<th id="Source" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Source</th>
<th id="Owner.Name" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Owner.Name</th>
<th id="Assembly.BioProject.Accession" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Assembly.BioProject.Accession</th>
<th id="Patotype" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Patotype</th>
<th id="Phylogroup" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Phylogroup</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_029625135.1</td>
<td class="gt_row gt_left" headers="Strain">E47</td>
<td class="gt_row gt_left" headers="Geographic.location">china</td>
<td class="gt_row gt_left" headers="Source">bovine</td>
<td class="gt_row gt_left" headers="Owner.Name">Chinese Academy of Agricultural Sciences</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJNA949005</td>
<td class="gt_row gt_left" headers="Patotype">NA</td>
<td class="gt_row gt_left" headers="Phylogroup">NA</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_025908275.1</td>
<td class="gt_row gt_left" headers="Strain">TL-14</td>
<td class="gt_row gt_left" headers="Geographic.location">china</td>
<td class="gt_row gt_left" headers="Source">bovine</td>
<td class="gt_row gt_left" headers="Owner.Name">Inner Mongolia Minzu University</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJNA890057</td>
<td class="gt_row gt_left" headers="Patotype">NA</td>
<td class="gt_row gt_left" headers="Phylogroup">NA</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_029319035.1</td>
<td class="gt_row gt_left" headers="Strain">E94</td>
<td class="gt_row gt_left" headers="Geographic.location">china</td>
<td class="gt_row gt_left" headers="Source">bovine</td>
<td class="gt_row gt_left" headers="Owner.Name">Chinese Academy of Agricultural Sciences</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJNA942474</td>
<td class="gt_row gt_left" headers="Patotype">NA</td>
<td class="gt_row gt_left" headers="Phylogroup">NA</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_029227895.1</td>
<td class="gt_row gt_left" headers="Strain">ZYB62</td>
<td class="gt_row gt_left" headers="Geographic.location">china</td>
<td class="gt_row gt_left" headers="Source">bovine</td>
<td class="gt_row gt_left" headers="Owner.Name">Lanzhou Institute of Husbandry and Pharmaceutical Sciences, Chinese Academy of Agricultural Science</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJNA727665</td>
<td class="gt_row gt_left" headers="Patotype">NA</td>
<td class="gt_row gt_left" headers="Phylogroup">NA</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_009931435.1</td>
<td class="gt_row gt_left" headers="Strain">EC42405</td>
<td class="gt_row gt_left" headers="Geographic.location">france</td>
<td class="gt_row gt_left" headers="Source">bovine</td>
<td class="gt_row gt_left" headers="Owner.Name">Anses</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJNA556131</td>
<td class="gt_row gt_left" headers="Patotype">NA</td>
<td class="gt_row gt_left" headers="Phylogroup">NA</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_004771235.1</td>
<td class="gt_row gt_left" headers="Strain">PF9285</td>
<td class="gt_row gt_left" headers="Geographic.location">switzerland</td>
<td class="gt_row gt_left" headers="Source">bovine</td>
<td class="gt_row gt_left" headers="Owner.Name">University of Bern</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJNA530748</td>
<td class="gt_row gt_left" headers="Patotype">NA</td>
<td class="gt_row gt_left" headers="Phylogroup">NA</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_049267065.1</td>
<td class="gt_row gt_left" headers="Strain">F3_13_1_3</td>
<td class="gt_row gt_left" headers="Geographic.location">usa</td>
<td class="gt_row gt_left" headers="Source">bovine</td>
<td class="gt_row gt_left" headers="Owner.Name">United State Department of Agriculture</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJNA728709</td>
<td class="gt_row gt_left" headers="Patotype">NA</td>
<td class="gt_row gt_left" headers="Phylogroup">NA</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_013168215.1</td>
<td class="gt_row gt_left" headers="Strain">7409</td>
<td class="gt_row gt_left" headers="Geographic.location">usa</td>
<td class="gt_row gt_left" headers="Source">bovine</td>
<td class="gt_row gt_left" headers="Owner.Name">United State Department of Agriculture</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJNA528413</td>
<td class="gt_row gt_left" headers="Patotype">NA</td>
<td class="gt_row gt_left" headers="Phylogroup">NA</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_013167395.1</td>
<td class="gt_row gt_left" headers="Strain">NE122</td>
<td class="gt_row gt_left" headers="Geographic.location">usa</td>
<td class="gt_row gt_left" headers="Source">bovine</td>
<td class="gt_row gt_left" headers="Owner.Name">United State Department of Agriculture</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJNA528413</td>
<td class="gt_row gt_left" headers="Patotype">NA</td>
<td class="gt_row gt_left" headers="Phylogroup">NA</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_048571305.1</td>
<td class="gt_row gt_left" headers="Strain">7_12_30A</td>
<td class="gt_row gt_left" headers="Geographic.location">usa</td>
<td class="gt_row gt_left" headers="Source">bovine</td>
<td class="gt_row gt_left" headers="Owner.Name">United State Department of Agriculture</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJNA728709</td>
<td class="gt_row gt_left" headers="Patotype">NA</td>
<td class="gt_row gt_left" headers="Phylogroup">NA</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_000025165.1</td>
<td class="gt_row gt_left" headers="Strain">CB9615</td>
<td class="gt_row gt_left" headers="Geographic.location">Germany</td>
<td class="gt_row gt_left" headers="Source">reference</td>
<td class="gt_row gt_left" headers="Owner.Name">Nankai University</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJNA224116</td>
<td class="gt_row gt_left" headers="Patotype">EPEC</td>
<td class="gt_row gt_left" headers="Phylogroup">E</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_000026545.1</td>
<td class="gt_row gt_left" headers="Strain">E2348/69</td>
<td class="gt_row gt_left" headers="Geographic.location">UK</td>
<td class="gt_row gt_left" headers="Source">reference</td>
<td class="gt_row gt_left" headers="Owner.Name">The Wellcome Trust Sanger Institute</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJNA224116</td>
<td class="gt_row gt_left" headers="Patotype">EPEC</td>
<td class="gt_row gt_left" headers="Phylogroup">B2</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_000026265.1</td>
<td class="gt_row gt_left" headers="Strain">IAI1</td>
<td class="gt_row gt_left" headers="Geographic.location">-</td>
<td class="gt_row gt_left" headers="Source">reference</td>
<td class="gt_row gt_left" headers="Owner.Name">EBI</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJNA33373</td>
<td class="gt_row gt_left" headers="Patotype">Comensal</td>
<td class="gt_row gt_left" headers="Phylogroup">B1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_000005845.2</td>
<td class="gt_row gt_left" headers="Strain">K-12 MG1655</td>
<td class="gt_row gt_left" headers="Geographic.location">usa</td>
<td class="gt_row gt_left" headers="Source">reference</td>
<td class="gt_row gt_left" headers="Owner.Name">Univ. Wisconsin</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJNA225</td>
<td class="gt_row gt_left" headers="Patotype">Comensal</td>
<td class="gt_row gt_left" headers="Phylogroup">A</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_000010745.1</td>
<td class="gt_row gt_left" headers="Strain">12009</td>
<td class="gt_row gt_left" headers="Geographic.location">japan</td>
<td class="gt_row gt_left" headers="Source">reference</td>
<td class="gt_row gt_left" headers="Owner.Name">University of Tokyo</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJDA32511</td>
<td class="gt_row gt_left" headers="Patotype">EHEC</td>
<td class="gt_row gt_left" headers="Phylogroup">B1</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Assembly.Accession">GCF_029962285.1</td>
<td class="gt_row gt_left" headers="Strain">EDL933</td>
<td class="gt_row gt_left" headers="Geographic.location">china</td>
<td class="gt_row gt_left" headers="Source">reference</td>
<td class="gt_row gt_left" headers="Owner.Name">East China University of Science and Technology</td>
<td class="gt_row gt_left" headers="Assembly.BioProject.Accession">PRJNA963085</td>
<td class="gt_row gt_left" headers="Patotype">EHEC</td>
<td class="gt_row gt_left" headers="Phylogroup">E</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<section id="descarga-de-archivos" class="level3">
<h3 class="anchored" data-anchor-id="descarga-de-archivos">Descarga de archivos</h3>
<p>Para descargar los archivos vamos a utilizar el siguiente código:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> g <span class="kw">in</span> <span class="va">$(</span><span class="fu">cut</span> <span class="at">-f1</span> Ecoli_bovineHost.tsv <span class="kw">|</span> <span class="fu">awk</span> <span class="st">'NR &gt; 1'</span><span class="va">)</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> <span class="bu">[</span> <span class="ot">-f</span> <span class="va">$g</span>.zip <span class="bu">]</span><span class="kw">;</span> <span class="cf">then</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>            <span class="bu">echo</span> <span class="va">$g</span><span class="st">'.zip exists'</span><span class="kw">;</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>        <span class="cf">else</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>            <span class="ex">datasets</span> download genome accession <span class="va">$g</span> <span class="dt">\ </span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>            <span class="ex">--include</span> genome <span class="at">--filename</span> <span class="va">$g</span><span class="st">'.zip'</span><span class="kw">;</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>            <span class="fu">sleep</span> 2<span class="kw">;</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>        <span class="cf">fi</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>    <span class="cf">done</span></span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Descomprimir los archivos</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> g <span class="kw">in</span> <span class="va">$(</span><span class="fu">cut</span> <span class="at">-f1</span> Ecoli_bovineHost.tsv <span class="kw">|</span> <span class="fu">awk</span> <span class="st">'NR &gt; 1'</span><span class="va">)</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>        <span class="fu">unzip</span> <span class="va">$g</span>.zip <span class="at">-d</span> <span class="va">$g</span><span class="kw">;</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>        <span class="fu">mv</span> <span class="va">$g</span>/ncbi_dataset/data/<span class="va">$g</span>/<span class="pp">*</span>.fna <span class="va">$g</span>.fna<span class="kw">;</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>        <span class="fu">rm</span> <span class="at">-r</span> <span class="va">$g</span>/<span class="kw">;</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span> <span class="pp">*</span>.zip</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Con esto ya contamos con un sub conjunto de datos listo para trabajar.</p>
</section>
</section>
<section id="control-de-calidad" class="level1">
<h1>Control de Calidad</h1>
<section id="fastani" class="level2">
<h2 class="anchored" data-anchor-id="fastani">FastANI</h2>
<p>Puedes descargar <code>FastANI</code> <a href="https://github.com/ParBLiSS/FastANI">aquí</a>.</p>
<p>FastANI (Fast Average Nucleotide Identity) calcula la <strong>identidad promedio de nucleótidos (ANI)</strong> entre dos genomas para determinar su similitud genética, útil para identificar especies o cepas.</p>
<p><strong>Cómo funciona</strong>:</p>
<ol type="1">
<li><p><strong>Entrada</strong>: Dos o más ensamblajes genómicos (FASTA, .fna).</p></li>
<li><p><strong>Proceso</strong>:</p>
<ul>
<li><p>Divide los genomas en fragmentos (k-mers).</p></li>
<li><p>Compara fragmentos homólogos entre genomas usando un enfoque rápido basado en alineamiento.</p></li>
<li><p>Calcula el porcentaje de ANI (por ejemplo, &gt;95% indica misma especie).</p></li>
</ul></li>
<li><p><strong>Salida</strong>: Un archivo tabular con valores de ANI entre pares de genomas (por ejemplo, .ani o .txt).</p></li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="ex">fastANI</span> <span class="dt">\</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>--ql <span class="va">$HOME</span>/2025_Demo_ecoli/1_QC/ANI/all_genomes.list <span class="dt">\</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>--rl <span class="va">$HOME</span>/2025_Demo_ecoli/1_QC/ANI/all_genomes.list <span class="dt">\</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>-o ecoli_all_to_all.out<span class="dt">\</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>--matrix</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="busco" class="level2">
<h2 class="anchored" data-anchor-id="busco">BUSCO</h2>
<p>Puedes descargar <code>BUSCO</code> <a href="https://busco.ezlab.org/">aquí</a>.</p>
<p>BUSCO (Benchmarking Universal Single-Copy Orthologs) evalúa la <strong>completitud</strong> y calidad de un ensamblaje genómico o anotaciones genéticas al buscar genes de copia única (ortólogos) que son altamente conservados en un grupo taxonómico específico (por ejemplo, bacterias, mamíferos).</p>
<p><strong>Cómo funciona</strong>:</p>
<ol type="1">
<li><p><strong>Entrada</strong>: Un ensamblaje genómico (FASTA, .fna) o archivo de proteínas anotadas (.faa).</p></li>
<li><p><strong>Base de datos</strong>: Compara contra un conjunto de genes ortólogos específicos del linaje (por ejemplo, bacteria_odb10 para bacterias).</p></li>
<li><p><strong>Proceso</strong>:</p>
<ul>
<li><p>Usa HMMER para buscar genes ortólogos en el genoma.</p></li>
<li><p>Clasifica los genes como <strong>Completos</strong> (presentes y completos), <strong>Fragmentados</strong> (parciales), <strong>Ausentes</strong> (no encontrados), o <strong>Duplicados</strong>.</p></li>
</ul></li>
<li><p><strong>Salida</strong>: Un informe con porcentajes de completitud y archivos detallados.</p></li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> <span class="va">$HOME</span>/bin/busco-env/bin/activate</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="va">genomes</span><span class="op">=</span><span class="va">$(</span><span class="fu">ls</span> <span class="va">$HOME</span>/2025_Demo_ecoli/0_data/<span class="pp">*</span>.fna<span class="va">)</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> g <span class="kw">in</span> <span class="va">$genomes</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>      <span class="va">id</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="va">$g</span> <span class="kw">|</span> <span class="fu">awk</span> <span class="at">-F</span><span class="st">'/'</span> <span class="st">'{print $NF}'</span><span class="va">)</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>      <span class="ex">/usr/bin/time</span> <span class="at">-v</span> <span class="dt">\</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>        busco <span class="dt">\</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">-i</span> <span class="st">"</span><span class="va">$g</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>        <span class="at">-l</span> bacteria_odb10 <span class="dt">\</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>        <span class="at">-m</span> geno <span class="dt">\</span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>        <span class="at">-o</span> 1_QC/busco/<span class="va">${id</span><span class="op">%</span>.fna<span class="va">}</span> <span class="dt">\</span></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>        <span class="at">--cpu</span> 8 <span class="dt">\</span></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>        <span class="at">--force</span> </span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>  <span class="cf">done</span> </span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>El <code>output</code> se ve de la siguiente manera por cada genoma:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># BUSCO version is: 5.8.0 </span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="co"># The lineage dataset is: bacteria_odb10 (Creation date: 2025-09-02, number of genomes: 4085, number of BUSCOs: 124)</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarized benchmarking in BUSCO notation for file GCF_029319035.1.fna</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="co"># BUSCO was run in mode: prok_genome_prod</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Gene predictor used: prodigal</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>    <span class="ex">*****</span> Results: <span class="pp">*****</span></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>    <span class="ex">C:100.0%[S:100.0%,D:0.0%],F:0.0%,M:0.0%,n:124</span>      </span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>    <span class="ex">124</span> Complete BUSCOs <span class="er">(</span><span class="ex">C</span><span class="kw">)</span>            </span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>    <span class="ex">124</span> Complete and single-copy BUSCOs <span class="er">(</span><span class="ex">S</span><span class="kw">)</span>    </span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>    <span class="ex">0</span>   Complete and duplicated BUSCOs <span class="er">(</span><span class="ex">D</span><span class="kw">)</span>     </span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>    <span class="ex">0</span>   Fragmented BUSCOs <span class="er">(</span><span class="ex">F</span><span class="kw">)</span>              </span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>    <span class="ex">0</span>   Missing BUSCOs <span class="er">(</span><span class="ex">M</span><span class="kw">)</span>             </span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>    <span class="ex">124</span> Total BUSCO groups searched        </span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a><span class="ex">Assembly</span> Statistics:</span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>    <span class="ex">6</span>   Number of scaffolds</span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a>    <span class="ex">6</span>   Number of contigs</span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a>    <span class="ex">5751071</span> Total length</span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a>    <span class="ex">0.000%</span>  Percent gaps</span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a>    <span class="ex">5</span> MB    Scaffold N50</span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a>    <span class="ex">5</span> MB    Contigs N50</span>
<span id="cb8-24"><a href="#cb8-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-25"><a href="#cb8-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-26"><a href="#cb8-26" aria-hidden="true" tabindex="-1"></a><span class="ex">Dependencies</span> and versions:</span>
<span id="cb8-27"><a href="#cb8-27" aria-hidden="true" tabindex="-1"></a>    <span class="ex">hmmsearch:</span> 3.4</span>
<span id="cb8-28"><a href="#cb8-28" aria-hidden="true" tabindex="-1"></a>    <span class="ex">bbtools:</span> None</span>
<span id="cb8-29"><a href="#cb8-29" aria-hidden="true" tabindex="-1"></a>    <span class="ex">prodigal:</span> 2.6.3</span>
<span id="cb8-30"><a href="#cb8-30" aria-hidden="true" tabindex="-1"></a>    <span class="ex">python:</span> sys.version_info<span class="er">(</span><span class="va">major</span><span class="op">=</span>3, <span class="va">minor</span><span class="op">=</span>12, <span class="va">micro</span><span class="op">=</span>2, <span class="va">releaselevel</span><span class="op">=</span><span class="st">'final'</span>, <span class="va">serial</span><span class="op">=</span>0<span class="kw">)</span></span>
<span id="cb8-31"><a href="#cb8-31" aria-hidden="true" tabindex="-1"></a>    <span class="ex">busco:</span> 5.8.0</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="anotación-de-genomas" class="level1">
<h1>Anotación de genomas</h1>
<section id="prokka" class="level2">
<h2 class="anchored" data-anchor-id="prokka">Prokka</h2>
<p><code>Prokka</code> es un anotador rápido y estándar para genomas procariontes (bacterias, arqueas y virus asociados) que identifica características genómicas (CDS, tRNAs, rRNAs, ncRNAs, etc.), las anota con nombres/IDs y genera archivos listos para análisis posteriores o envío a bases públicas.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="va">genomes</span><span class="op">=</span><span class="va">$(</span> <span class="fu">ls</span> <span class="va">$HOME</span>/2025_Demo_ecoli/1_data/<span class="pp">*</span>.fna <span class="va">)</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> g <span class="kw">in</span> <span class="va">$genomes</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a> <span class="ex">prokka</span> <span class="dt">\</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a> <span class="va">$g</span> <span class="dt">\</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a> <span class="at">--cpus</span> 6 <span class="dt">\</span></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a> <span class="at">--outdir</span> prokka/<span class="va">${id</span><span class="op">%</span>.fna<span class="va">}</span> <span class="dt">\</span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a> <span class="at">--genus</span> Escherichia</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li><p><code>genomes=.../*.fna</code> — crea la lista de archivos FASTA de contigs/consensos que serán anotados.</p></li>
<li><p>El <code>for g in $genomes; do ... done</code> recorre cada ensamblaje.</p></li>
<li><p><code>prokka $g</code> — indica a Prokka el archivo de entrada (.fna/.fasta).</p></li>
<li><p><code>--cpus 6</code> — usa 6 hilos para acelerar pasos paralelizables (e.g., búsquedas).</p></li>
<li><p><code>--outdir prokka/${id%.fna}</code> — dirige la salida a una carpeta por genoma (nota: en el snippet que enviaste la variable <code>id</code> <strong>no está definida</strong> dentro del bucle; ver corrección más abajo).</p></li>
<li><p><code>--genus Escherichia</code> — parámetro opcional que ayuda a priorizar nombres/funciones (mejorará la selección de nombres genéricos).</p></li>
</ul>
<p><strong>Principales archivos que genera Prokka y para qué sirven</strong></p>
<ul>
<li><p><code>&lt;sample&gt;.gff</code> — <em>archivo maestro</em> en formato GFF3 que contiene las características anotadas (coordenadas, tipos, anotaciones). Usarlo para cargar en IGV/Artemis o como entrada a pipelines que esperan GFF.</p></li>
<li><p><code>&lt;sample&gt;.gbk</code> (o <code>.gbk.gz</code>) — archivo en formato GenBank (útil para visualización/depósito y para herramientas que leen GenBank).</p></li>
<li><p><code>&lt;sample&gt;.fna</code> — la secuencia de nucleótidos (contigs) usada/normalizada por Prokka (puede incluir cambios en nombres de secuencia).</p></li>
<li><p><code>&lt;sample&gt;.ffn</code> — secuencias de nucleótidos de los genes codificantes (CDS) extraídas.</p></li>
<li><p><code>&lt;sample&gt;.faa</code> — traducciones (secuencias proteicas) de los CDS; útil para búsquedas funcionales (BLAST/DIAMOND), clustering o análisis de pangenoma.</p></li>
<li><p><code>&lt;sample&gt;.tbl</code> — tabla sencilla con información de anotación (útil para envío a GenBank/ENA en algunos workflows).</p></li>
<li><p><code>&lt;sample&gt;.sqn</code> (opcional) — formato Sequin para enviar a NCBI (si se generan).</p></li>
<li><p><code>PROKKA_*.log</code> / <code>prokka.err</code> — logs que detallan programas internos y pasos realizados (útil para depuración).</p></li>
</ul>
<p><strong>Notas prácticas y recomendaciones para Prokka</strong></p>
<ul>
<li><p>Si tu entrada es multi-FASTA con múltiples contigs, Prokka generará un GenBank multi-registro con uno por contig.</p></li>
<li><p>Ajusta <code>--kingdom</code>, <code>--genus</code>, <code>--species</code>, <code>--strain</code> para mejorar la nomenclatura.</p></li>
<li><p>Prokka usa varias bases (p.&nbsp;ej. UniProt/Swiss-Prot) y heurísticas; para anotaciones más «exhaustivas» posteriores se pueden re-analizar las proteínas (<code>.faa</code>) con herramientas especializadas. (<a href="https://github.com/tseemann/prokka?utm_source=chatgpt.com" title="Prokka - zap: Rapid prokaryotic genome annotation">GitHub</a>)</p></li>
</ul>
<hr>
</section>
<section id="bakta" class="level2">
<h2 class="anchored" data-anchor-id="bakta">Bakta</h2>
<p><code>Bakta</code> es un anotador diseñado para obtener anotaciones amplias, normalizadas y «dbxref-ricas» (múltiples referencias cruzadas) en genomas bacterianos y plásmidos. Está pensado para producir salidas estandarizadas (incluyendo JSON legible por máquina) y sORFs (small ORFs) además de las características estándar (CDS, tRNA, rRNA, ncRNA, CRISPR, etc.). Usa una base de datos local y herramientas rápidas (p.&nbsp;ej. DIAMOND) para hacer búsquedas de homología.</p>
<pre><code>source $HOME/bin/bakta-env/bin/activate</code></pre>
<div class="cell">
<details class="code-fold">
<summary>Muestra el código</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="va">genomes</span><span class="op">=</span><span class="va">$(</span><span class="fu">ls</span> <span class="va">$HOME</span>/2025_Demo_ecoli/1_data<span class="va">)</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="va">experiment_file</span><span class="op">=</span><span class="st">"</span><span class="va">$HOME</span><span class="st">/2025_Demo_ecoli/Experiment_Design.tsv"</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="va">genomes</span><span class="op">=</span><span class="va">$(</span><span class="fu">ls</span> <span class="va">$HOME</span>/2025_Demo_ecoli/1_data<span class="va">)</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a><span class="va">experiment_file</span><span class="op">=</span><span class="st">"</span><span class="va">$HOME</span><span class="st">/2025_Demo_ecoli/Experiment_Design.tsv"</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> g <span class="kw">in</span> <span class="va">$genomes</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>    <span class="va">id</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="st">"</span><span class="va">$g</span><span class="st">"</span> <span class="kw">|</span> <span class="fu">awk</span> <span class="at">-F</span><span class="st">'/'</span> <span class="st">'{print $NF}'</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">'s/.fna$//'</span><span class="va">)</span></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>    <span class="va">strain</span><span class="op">=</span><span class="va">$(</span><span class="fu">awk</span> <span class="at">-F</span><span class="st">'\t'</span> <span class="at">-v</span> id=<span class="st">"</span><span class="va">$id</span><span class="st">"</span> <span class="st">'$1 == id {print $3}'</span> <span class="st">"</span><span class="va">$experiment_file</span><span class="st">"</span><span class="va">)</span></span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>    <span class="ex">bakta</span> <span class="dt">\</span></span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">--db</span> <span class="va">$HOME</span>/2025_Demo_ecoli/db <span class="dt">\</span></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">--output</span> <span class="va">$HOME</span>/2025_Demo_ecoli/2_annotation/bakta/<span class="va">${id}</span> <span class="dt">\</span></span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">--genus</span> Escherichia <span class="dt">\</span></span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">--strain</span> <span class="st">"</span><span class="va">$strain</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">--threads</span> 6 <span class="dt">\</span></span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">--force</span> <span class="va">$HOME</span>/1_data/<span class="va">${g}</span></span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<ul>
<li><p><code>source .../bin/activate</code> — activa el entorno (p.&nbsp;ej. virtualenv/conda) donde está Bakta.</p></li>
<li><p><code>--db $HOME/.../db</code> — ubicación de la base de datos compacta de Bakta (necesaria; puede ocupar decenas de GB).</p></li>
<li><p><code>--output ...</code> — carpeta por genoma para guardar resultados.</p></li>
<li><p><code>--genus</code> / <code>--strain</code> — ayudan en la anotación y en la generación de nombres.</p></li>
<li><p><code>--threads 6</code> — número de hilos para paralelizar búsquedas.</p></li>
<li><p><code>--force</code> — sobrescribir salidas existentes.</p></li>
<li><p>Bakta internamente usa búsquedas con DIAMOND/BLAST y una base SQLite con digest y anotaciones preasignadas (esto acelera y estandariza el proceso).</p></li>
</ul>
<p><strong>Precauciones de recursos</strong> ⚠️<br>
</p>
<p>Bakta requiere memoria y disco para su base y para los pasos intermedios; es habitual necesitar varios GB de RAM y tiempo cuando se procesan muchos genomas. Si tienes &gt;10 genomas, espera que el tiempo total aumente y planifica en un clúster o máquina con suficientes recursos.</p>
<p><strong>Archivos y salidas importantes que produce Bakta</strong></p>
<ul>
<li><p><code>annotation.gff</code> / <code>&lt;id&gt;.gff</code> — GFF3 con anotaciones (coordenadas y atributos).</p></li>
<li><p><code>&lt;id&gt;.gbk</code> — GenBank (cuando se solicita o se genera).</p></li>
<li><p><code>&lt;id&gt;.faa</code>, <code>&lt;id&gt;.ffn</code> — proteínas y secuencias nucleotídicas de CDS (similares a Prokka).</p></li>
<li><p><code>report.json</code> — salida JSON rica en metadatos y dbxrefs (útil para pipelines automatizados). Bakta hace énfasis en producir JSON estandarizado que puede alimentar otras herramientas.</p></li>
<li><p><code>bakta.log</code> / registros — salida estándar y errores para depuración.</p></li>
</ul>
<p><strong>Ventajas frente a Prokka</strong></p>
<ul>
<li><p>Bakta genera metadatos más ricos / dbxrefs y JSON preparado para downstream automático.</p></li>
<li><p>Usa una base precomprimida con mapeos a UniProt (y otras colecciones), lo que suele producir IDs más estandarizados.</p></li>
<li><p>Prokka es muy rápido y ampliamente usado; Bakta tiende a ser más exhaustivo y «estandarizado» pero con requisitos mayores de disco/DB. Para workflows, muchos usuarios ejecutan <strong>Prokka</strong> para anotaciones rápidas y <strong>Bakta</strong> cuando quieren anotaciones más normalizadas y ricas en dbxrefs.</p></li>
</ul>
</section>
<section id="consejos-para-decidir-cuál-usar-en-tu-pipeline" class="level2">
<h2 class="anchored" data-anchor-id="consejos-para-decidir-cuál-usar-en-tu-pipeline">Consejos para decidir cuál usar en tu pipeline</h2>
<ul>
<li><p>Si quieres <strong>velocidad</strong> y una anotación estándar rápida para muchas muestras: usa <strong>Prokka</strong> y luego procesa los <code>.faa</code> con herramientas funcionales (eg. EggNOG, InterProScan) según necesites.</p></li>
<li><p>Si quieres <strong>anotaciones más ricas, normalizadas y con dbxrefs/JSON</strong> para downstream automático (p.&nbsp;ej. bases de datos locales, comparaciones taxonómicas o reporte estandarizado), usa <strong>Bakta</strong> (pero planifica la DB y recursos).</p></li>
</ul>
</section>
<section id="amrfinderplus" class="level2">
<h2 class="anchored" data-anchor-id="amrfinderplus">AMRfinderPlus</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate amrfinder</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> faa <span class="kw">in</span> 2_annotation/prokka/<span class="pp">*</span>/<span class="pp">*</span>.faa<span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>  <span class="va">sample</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="st">"</span><span class="va">${faa</span><span class="op">%%</span>.<span class="pp">*</span><span class="va">}</span><span class="st">"</span><span class="va">)</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>  <span class="va">out</span><span class="op">=</span><span class="st">"8_AMR/</span><span class="va">${sample}</span><span class="st">_amrfinder_prot.tsv"</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>  <span class="bu">echo</span> <span class="st">"Running amrfinder -p </span><span class="va">$faa</span><span class="st"> -&gt; </span><span class="va">$out</span><span class="st">"</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>  <span class="ex">amrfinder</span> <span class="at">-p</span> <span class="st">"</span><span class="va">$faa</span><span class="st">"</span> <span class="at">--plus</span> <span class="at">--threads</span> 4 <span class="at">-o</span> <span class="st">"</span><span class="va">$out</span><span class="st">"</span></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>El <code>output</code> de cada genoma se ve de la siguiente manera:</p>
<div class="cell">
<div class="cell-output-display">
<div id="qjfqcezdhg" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#qjfqcezdhg table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#qjfqcezdhg thead, #qjfqcezdhg tbody, #qjfqcezdhg tfoot, #qjfqcezdhg tr, #qjfqcezdhg td, #qjfqcezdhg th {
  border-style: none;
}

#qjfqcezdhg p {
  margin: 0;
  padding: 0;
}

#qjfqcezdhg .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#qjfqcezdhg .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#qjfqcezdhg .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#qjfqcezdhg .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#qjfqcezdhg .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#qjfqcezdhg .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#qjfqcezdhg .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#qjfqcezdhg .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#qjfqcezdhg .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#qjfqcezdhg .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#qjfqcezdhg .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#qjfqcezdhg .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#qjfqcezdhg .gt_spanner_row {
  border-bottom-style: hidden;
}

#qjfqcezdhg .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#qjfqcezdhg .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#qjfqcezdhg .gt_from_md > :first-child {
  margin-top: 0;
}

#qjfqcezdhg .gt_from_md > :last-child {
  margin-bottom: 0;
}

#qjfqcezdhg .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#qjfqcezdhg .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#qjfqcezdhg .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#qjfqcezdhg .gt_row_group_first td {
  border-top-width: 2px;
}

#qjfqcezdhg .gt_row_group_first th {
  border-top-width: 2px;
}

#qjfqcezdhg .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#qjfqcezdhg .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#qjfqcezdhg .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#qjfqcezdhg .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#qjfqcezdhg .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#qjfqcezdhg .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#qjfqcezdhg .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#qjfqcezdhg .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#qjfqcezdhg .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#qjfqcezdhg .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#qjfqcezdhg .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#qjfqcezdhg .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#qjfqcezdhg .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#qjfqcezdhg .gt_left {
  text-align: left;
}

#qjfqcezdhg .gt_center {
  text-align: center;
}

#qjfqcezdhg .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#qjfqcezdhg .gt_font_normal {
  font-weight: normal;
}

#qjfqcezdhg .gt_font_bold {
  font-weight: bold;
}

#qjfqcezdhg .gt_font_italic {
  font-style: italic;
}

#qjfqcezdhg .gt_super {
  font-size: 65%;
}

#qjfqcezdhg .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#qjfqcezdhg .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#qjfqcezdhg .gt_indent_1 {
  text-indent: 5px;
}

#qjfqcezdhg .gt_indent_2 {
  text-indent: 10px;
}

#qjfqcezdhg .gt_indent_3 {
  text-indent: 15px;
}

#qjfqcezdhg .gt_indent_4 {
  text-indent: 20px;
}

#qjfqcezdhg .gt_indent_5 {
  text-indent: 25px;
}

#qjfqcezdhg .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#qjfqcezdhg div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead>
<tr class="gt_col_headings header">
<th id="Protein-id" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Protein id</th>
<th id="Element-symbol" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Element symbol</th>
<th id="Element-name" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Element name</th>
<th id="Scope" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Scope</th>
<th id="Type" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Type</th>
<th id="Subtype" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Subtype</th>
<th id="Class" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Class</th>
<th id="Subclass" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Subclass</th>
<th id="Method" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Method</th>
<th id="Target-length" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">Target length</th>
<th id="Reference-sequence-length" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">Reference sequence length</th>
<th id="a%-Coverage-of-reference" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">% Coverage of reference</th>
<th id="a%-Identity-to-reference" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">% Identity to reference</th>
<th id="Alignment-length" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">Alignment length</th>
<th id="Closest-reference-accession" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Closest reference accession</th>
<th id="Closest-reference-name" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Closest reference name</th>
<th id="HMM-accession" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">HMM accession</th>
<th id="HMM-description" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">HMM description</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="Protein id">PICPPNBC_00538</td>
<td class="gt_row gt_left" headers="Element symbol">emrE</td>
<td class="gt_row gt_left" headers="Element name">multidrug efflux SMR transporter EmrE</td>
<td class="gt_row gt_left" headers="Scope">plus</td>
<td class="gt_row gt_left" headers="Type">STRESS</td>
<td class="gt_row gt_left" headers="Subtype">BIOCIDE</td>
<td class="gt_row gt_left" headers="Class">EFFLUX</td>
<td class="gt_row gt_left" headers="Subclass">EFFLUX</td>
<td class="gt_row gt_left" headers="Method">EXACTP</td>
<td class="gt_row gt_right" headers="Target length">110</td>
<td class="gt_row gt_right" headers="Reference sequence length">110</td>
<td class="gt_row gt_right" headers="% Coverage of reference">100</td>
<td class="gt_row gt_right" headers="% Identity to reference">100.00</td>
<td class="gt_row gt_right" headers="Alignment length">110</td>
<td class="gt_row gt_left" headers="Closest reference accession">CAA77936.1</td>
<td class="gt_row gt_left" headers="Closest reference name">multidrug efflux SMR transporter EmrE</td>
<td class="gt_row gt_left" headers="HMM accession">NF042984.1</td>
<td class="gt_row gt_left" headers="HMM description">multidrug efflux SMR transporter EmrE</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Protein id">PICPPNBC_01161</td>
<td class="gt_row gt_left" headers="Element symbol">ariR</td>
<td class="gt_row gt_left" headers="Element name">biofilm/acid-resistance regulator AriR</td>
<td class="gt_row gt_left" headers="Scope">plus</td>
<td class="gt_row gt_left" headers="Type">STRESS</td>
<td class="gt_row gt_left" headers="Subtype">ACID</td>
<td class="gt_row gt_left" headers="Class">NA</td>
<td class="gt_row gt_left" headers="Subclass">NA</td>
<td class="gt_row gt_left" headers="Method">EXACTP</td>
<td class="gt_row gt_right" headers="Target length">88</td>
<td class="gt_row gt_right" headers="Reference sequence length">88</td>
<td class="gt_row gt_right" headers="% Coverage of reference">100</td>
<td class="gt_row gt_right" headers="% Identity to reference">100.00</td>
<td class="gt_row gt_right" headers="Alignment length">88</td>
<td class="gt_row gt_left" headers="Closest reference accession">AAC74250.1</td>
<td class="gt_row gt_left" headers="Closest reference name">biofilm/acid-resistance regulator AriR</td>
<td class="gt_row gt_left" headers="HMM accession">NA</td>
<td class="gt_row gt_left" headers="HMM description">NA</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Protein id">PICPPNBC_01599</td>
<td class="gt_row gt_left" headers="Element symbol">asr</td>
<td class="gt_row gt_left" headers="Element name">acid resistance repetitive basic protein Asr</td>
<td class="gt_row gt_left" headers="Scope">plus</td>
<td class="gt_row gt_left" headers="Type">STRESS</td>
<td class="gt_row gt_left" headers="Subtype">ACID</td>
<td class="gt_row gt_left" headers="Class">NA</td>
<td class="gt_row gt_left" headers="Subclass">NA</td>
<td class="gt_row gt_left" headers="Method">EXACTP</td>
<td class="gt_row gt_right" headers="Target length">102</td>
<td class="gt_row gt_right" headers="Reference sequence length">102</td>
<td class="gt_row gt_right" headers="% Coverage of reference">100</td>
<td class="gt_row gt_right" headers="% Identity to reference">100.00</td>
<td class="gt_row gt_right" headers="Alignment length">102</td>
<td class="gt_row gt_left" headers="Closest reference accession">BAA15331.2</td>
<td class="gt_row gt_left" headers="Closest reference name">acid resistance repetitive basic protein Asr</td>
<td class="gt_row gt_left" headers="HMM accession">NF033636.1</td>
<td class="gt_row gt_left" headers="HMM description">acid resistance repetitive basic protein Asr</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Protein id">PICPPNBC_02975</td>
<td class="gt_row gt_left" headers="Element symbol">sslE</td>
<td class="gt_row gt_left" headers="Element name">lipoprotein metalloprotease SslE</td>
<td class="gt_row gt_left" headers="Scope">plus</td>
<td class="gt_row gt_left" headers="Type">VIRULENCE</td>
<td class="gt_row gt_left" headers="Subtype">VIRULENCE</td>
<td class="gt_row gt_left" headers="Class">NA</td>
<td class="gt_row gt_left" headers="Subclass">NA</td>
<td class="gt_row gt_left" headers="Method">HMM</td>
<td class="gt_row gt_right" headers="Target length">1520</td>
<td class="gt_row gt_right" headers="Reference sequence length">1520</td>
<td class="gt_row gt_right" headers="% Coverage of reference">100</td>
<td class="gt_row gt_right" headers="% Identity to reference">88.03</td>
<td class="gt_row gt_right" headers="Alignment length">1521</td>
<td class="gt_row gt_left" headers="Closest reference accession">ADE89421.1</td>
<td class="gt_row gt_left" headers="Closest reference name">lipoprotein metalloprotease SslE</td>
<td class="gt_row gt_left" headers="HMM accession">NF037973.1</td>
<td class="gt_row gt_left" headers="HMM description">lipoprotein metalloprotease SslE</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Protein id">PICPPNBC_03259</td>
<td class="gt_row gt_left" headers="Element symbol">acrF</td>
<td class="gt_row gt_left" headers="Element name">multidrug efflux RND transporter permease subunit AcrF</td>
<td class="gt_row gt_left" headers="Scope">plus</td>
<td class="gt_row gt_left" headers="Type">AMR</td>
<td class="gt_row gt_left" headers="Subtype">AMR</td>
<td class="gt_row gt_left" headers="Class">EFFLUX</td>
<td class="gt_row gt_left" headers="Subclass">EFFLUX</td>
<td class="gt_row gt_left" headers="Method">EXACTP</td>
<td class="gt_row gt_right" headers="Target length">1034</td>
<td class="gt_row gt_right" headers="Reference sequence length">1034</td>
<td class="gt_row gt_right" headers="% Coverage of reference">100</td>
<td class="gt_row gt_right" headers="% Identity to reference">100.00</td>
<td class="gt_row gt_right" headers="Alignment length">1034</td>
<td class="gt_row gt_left" headers="Closest reference accession">AAC76298.1</td>
<td class="gt_row gt_left" headers="Closest reference name">multidrug efflux RND transporter permease subunit AcrF</td>
<td class="gt_row gt_left" headers="HMM accession">NA</td>
<td class="gt_row gt_left" headers="HMM description">NA</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Protein id">PICPPNBC_03497</td>
<td class="gt_row gt_left" headers="Element symbol">arsR</td>
<td class="gt_row gt_left" headers="Element name">As(III)-sensing metalloregulatory transcriptional repressor ArsR</td>
<td class="gt_row gt_left" headers="Scope">plus</td>
<td class="gt_row gt_left" headers="Type">STRESS</td>
<td class="gt_row gt_left" headers="Subtype">METAL</td>
<td class="gt_row gt_left" headers="Class">ARSENIC</td>
<td class="gt_row gt_left" headers="Subclass">ARSENIC</td>
<td class="gt_row gt_left" headers="Method">EXACTP</td>
<td class="gt_row gt_right" headers="Target length">117</td>
<td class="gt_row gt_right" headers="Reference sequence length">117</td>
<td class="gt_row gt_right" headers="% Coverage of reference">100</td>
<td class="gt_row gt_right" headers="% Identity to reference">100.00</td>
<td class="gt_row gt_right" headers="Alignment length">117</td>
<td class="gt_row gt_left" headers="Closest reference accession">BAE77793.1</td>
<td class="gt_row gt_left" headers="Closest reference name">As(III)-sensing metalloregulatory transcriptional repressor ArsR</td>
<td class="gt_row gt_left" headers="HMM accession">NA</td>
<td class="gt_row gt_left" headers="HMM description">NA</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Protein id">PICPPNBC_03499</td>
<td class="gt_row gt_left" headers="Element symbol">arsC</td>
<td class="gt_row gt_left" headers="Element name">glutaredoxin-dependent arsenate reductase</td>
<td class="gt_row gt_left" headers="Scope">plus</td>
<td class="gt_row gt_left" headers="Type">STRESS</td>
<td class="gt_row gt_left" headers="Subtype">METAL</td>
<td class="gt_row gt_left" headers="Class">ARSENIC</td>
<td class="gt_row gt_left" headers="Subclass">ARSENATE</td>
<td class="gt_row gt_left" headers="Method">BLASTP</td>
<td class="gt_row gt_right" headers="Target length">141</td>
<td class="gt_row gt_right" headers="Reference sequence length">141</td>
<td class="gt_row gt_right" headers="% Coverage of reference">100</td>
<td class="gt_row gt_right" headers="% Identity to reference">94.33</td>
<td class="gt_row gt_right" headers="Alignment length">141</td>
<td class="gt_row gt_left" headers="Closest reference accession">AAA21096.1</td>
<td class="gt_row gt_left" headers="Closest reference name">glutaredoxin-dependent arsenate reductase</td>
<td class="gt_row gt_left" headers="HMM accession">NF007456.1</td>
<td class="gt_row gt_left" headers="HMM description">glutaredoxin-dependent arsenate reductase</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Protein id">PICPPNBC_03676</td>
<td class="gt_row gt_left" headers="Element symbol">emrD</td>
<td class="gt_row gt_left" headers="Element name">multidrug efflux MFS transporter EmrD</td>
<td class="gt_row gt_left" headers="Scope">plus</td>
<td class="gt_row gt_left" headers="Type">AMR</td>
<td class="gt_row gt_left" headers="Subtype">AMR</td>
<td class="gt_row gt_left" headers="Class">EFFLUX</td>
<td class="gt_row gt_left" headers="Subclass">EFFLUX</td>
<td class="gt_row gt_left" headers="Method">EXACTP</td>
<td class="gt_row gt_right" headers="Target length">394</td>
<td class="gt_row gt_right" headers="Reference sequence length">394</td>
<td class="gt_row gt_right" headers="% Coverage of reference">100</td>
<td class="gt_row gt_right" headers="% Identity to reference">100.00</td>
<td class="gt_row gt_right" headers="Alignment length">394</td>
<td class="gt_row gt_left" headers="Closest reference accession">BAE77620.1</td>
<td class="gt_row gt_left" headers="Closest reference name">multidrug efflux MFS transporter EmrD</td>
<td class="gt_row gt_left" headers="HMM accession">NA</td>
<td class="gt_row gt_left" headers="HMM description">NA</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Protein id">PICPPNBC_03916</td>
<td class="gt_row gt_left" headers="Element symbol">fieF</td>
<td class="gt_row gt_left" headers="Element name">CDF family cation-efflux transporter FieF</td>
<td class="gt_row gt_left" headers="Scope">plus</td>
<td class="gt_row gt_left" headers="Type">STRESS</td>
<td class="gt_row gt_left" headers="Subtype">METAL</td>
<td class="gt_row gt_left" headers="Class">NA</td>
<td class="gt_row gt_left" headers="Subclass">NA</td>
<td class="gt_row gt_left" headers="Method">EXACTP</td>
<td class="gt_row gt_right" headers="Target length">300</td>
<td class="gt_row gt_right" headers="Reference sequence length">300</td>
<td class="gt_row gt_right" headers="% Coverage of reference">100</td>
<td class="gt_row gt_right" headers="% Identity to reference">100.00</td>
<td class="gt_row gt_right" headers="Alignment length">300</td>
<td class="gt_row gt_left" headers="Closest reference accession">BAE77395.1</td>
<td class="gt_row gt_left" headers="Closest reference name">CDF family cation-efflux transporter FieF</td>
<td class="gt_row gt_left" headers="HMM accession">NA</td>
<td class="gt_row gt_left" headers="HMM description">NA</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Protein id">PICPPNBC_04160</td>
<td class="gt_row gt_left" headers="Element symbol">blaEC</td>
<td class="gt_row gt_left" headers="Element name">BlaEC family class C beta-lactamase</td>
<td class="gt_row gt_left" headers="Scope">plus</td>
<td class="gt_row gt_left" headers="Type">AMR</td>
<td class="gt_row gt_left" headers="Subtype">AMR</td>
<td class="gt_row gt_left" headers="Class">BETA-LACTAM</td>
<td class="gt_row gt_left" headers="Subclass">BETA-LACTAM</td>
<td class="gt_row gt_left" headers="Method">EXACTP</td>
<td class="gt_row gt_right" headers="Target length">377</td>
<td class="gt_row gt_right" headers="Reference sequence length">377</td>
<td class="gt_row gt_right" headers="% Coverage of reference">100</td>
<td class="gt_row gt_right" headers="% Identity to reference">100.00</td>
<td class="gt_row gt_right" headers="Alignment length">377</td>
<td class="gt_row gt_left" headers="Closest reference accession">AAC77110.1</td>
<td class="gt_row gt_left" headers="Closest reference name">BlaEC family class C beta-lactamase</td>
<td class="gt_row gt_left" headers="HMM accession">NF000185.2</td>
<td class="gt_row gt_left" headers="HMM description">BlaEC family class C beta-lactamase</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Protein id">PICPPNBC_04349</td>
<td class="gt_row gt_left" headers="Element symbol">mdtM</td>
<td class="gt_row gt_left" headers="Element name">multidrug efflux MFS transporter MdtM</td>
<td class="gt_row gt_left" headers="Scope">plus</td>
<td class="gt_row gt_left" headers="Type">AMR</td>
<td class="gt_row gt_left" headers="Subtype">AMR</td>
<td class="gt_row gt_left" headers="Class">EFFLUX</td>
<td class="gt_row gt_left" headers="Subclass">EFFLUX</td>
<td class="gt_row gt_left" headers="Method">EXACTP</td>
<td class="gt_row gt_right" headers="Target length">410</td>
<td class="gt_row gt_right" headers="Reference sequence length">410</td>
<td class="gt_row gt_right" headers="% Coverage of reference">100</td>
<td class="gt_row gt_right" headers="% Identity to reference">100.00</td>
<td class="gt_row gt_right" headers="Alignment length">410</td>
<td class="gt_row gt_left" headers="Closest reference accession">AAC77293.1</td>
<td class="gt_row gt_left" headers="Closest reference name">multidrug efflux MFS transporter MdtM</td>
<td class="gt_row gt_left" headers="HMM accession">NA</td>
<td class="gt_row gt_left" headers="HMM description">NA</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
</section>
<section id="virulence-factor-database" class="level2">
<h2 class="anchored" data-anchor-id="virulence-factor-database">Virulence Factor DataBase</h2>
<p>Primero vamos a descargar los FASTA de VFDB <a href="https://www.mgc.ac.cn/VFs/download.htm">aquí</a>. Y vamos a crear la base de datos para poder usar posteriormente blastp. <strong>NOTA</strong>: esto lo puedes aplicar a cualquier otra base de datos!!</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="ex">makeblastdb</span> <span class="dt">\</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>-in data/VFDB/VFDB_setB_pro.fas <span class="dt">\</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>-dbtype prot <span class="dt">\</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>-out data/VFDB/vfdb</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Una vez que genramos la base de datos vamos a hacer un BLASTP de nuestros genomas <code>.faa</code></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="va">BASE_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">$HOME</span><span class="st">/2025_Demo_ecoli"</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="va">samples</span><span class="op">=</span><span class="va">$(</span><span class="fu">find</span> <span class="st">"</span><span class="va">$BASE_DIR</span><span class="st">/2_annotation/prokka"</span> <span class="at">-maxdepth</span> 1 <span class="at">-type</span> d <span class="at">-not</span> <span class="at">-path</span> <span class="st">"</span><span class="va">$BASE_DIR</span><span class="st">/2_annotation/prokka"</span> <span class="at">-exec</span> basename {} <span class="dt">\;</span><span class="va">)</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="va">simult</span><span class="op">=</span>2</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a><span class="va">counter</span><span class="op">=</span>0</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> s <span class="kw">in</span> <span class="va">$samples</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> <span class="kw">[[</span> <span class="st">"</span><span class="va">$counter</span><span class="st">"</span> <span class="ot">-lt</span> <span class="st">"</span><span class="va">$simult</span><span class="st">"</span> <span class="kw">]];</span> <span class="cf">then</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>      <span class="va">protein_faa</span><span class="op">=</span><span class="va">$(</span><span class="fu">find</span> <span class="st">"</span><span class="va">$BASE_DIR</span><span class="st">/2_annotation/prokka/</span><span class="va">$s</span><span class="st">"</span> <span class="at">-maxdepth</span> 1 <span class="at">-type</span> f <span class="at">-name</span> <span class="st">"*.faa"</span><span class="va">)</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> <span class="bu">[</span> <span class="ot">-n</span> <span class="st">"</span><span class="va">$protein_faa</span><span class="st">"</span> <span class="bu">]</span><span class="kw">;</span> <span class="cf">then</span></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>        <span class="ex">blastp</span> <span class="at">-db</span> <span class="st">"</span><span class="va">$BASE_DIR</span><span class="st">/data/VFDB/vfdb"</span> <span class="dt">\</span></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>          <span class="at">-query</span> <span class="st">"</span><span class="va">$protein_faa</span><span class="st">"</span> <span class="dt">\</span></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>          <span class="at">-outfmt</span> <span class="st">"6 qseqid sseqid pident qlen slen length evalue bitscore"</span> <span class="dt">\</span></span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>          <span class="at">-max_target_seqs</span> 1 <span class="at">-num_threads</span> 4 <span class="op">&gt;</span> <span class="st">"</span><span class="va">$BASE_DIR</span><span class="st">/2_annotation/</span><span class="va">$s</span><span class="st">.out"</span> <span class="kw">&amp;</span></span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>        <span class="va">counter</span><span class="op">=</span><span class="va">$((counter</span><span class="op">+</span><span class="dv">1</span><span class="va">))</span></span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> <span class="kw">[[</span> <span class="st">"</span><span class="va">$counter</span><span class="st">"</span> <span class="ot">-eq</span> <span class="st">"</span><span class="va">$simult</span><span class="st">"</span> <span class="kw">]];</span> <span class="cf">then</span> <span class="bu">wait</span><span class="kw">;</span> <span class="va">counter</span><span class="op">=</span>0<span class="kw">;</span> <span class="cf">fi</span></span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>    <span class="cf">fi</span></span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>  <span class="cf">done</span></span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a>  <span class="bu">wait</span></span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Esto genera una salida <code>.out</code> el cual se ve así:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="ex">PICPPNBC_00001</span>  VFG039381<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">NP_819814</span><span class="kw">)</span> <span class="ex">34.545</span>  820 464 55  0.94    32.7</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="ex">PICPPNBC_00002</span>  VFG002555<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">WP_004194045</span><span class="kw">)</span>  <span class="ex">30.952</span>  310 346 84  0.031   35.8</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="ex">PICPPNBC_00003</span>  VFG014236<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">WP_011333018</span><span class="kw">)</span>  <span class="ex">30.400</span>  428 231 125 0.061   35.0</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="ex">PICPPNBC_00004</span>  VFG016425<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">WP_000780148</span><span class="kw">)</span>  <span class="ex">35.849</span>  98  459 53  2.0 26.9</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="ex">PICPPNBC_00005</span>  VFG013340<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">WP_010945509</span><span class="kw">)</span>  <span class="ex">29.213</span>  258 224 89  7.2 27.7</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a><span class="ex">PICPPNBC_00007</span>  VFG049100<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">WP_004184789</span><span class="kw">)</span>  <span class="ex">31.481</span>  317 371 54  3.6 29.3</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a><span class="ex">PICPPNBC_00008</span>  VFG038347<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">WP_005320757</span><span class="kw">)</span>  <span class="ex">45.714</span>  195 292 35  1.4 29.3</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a><span class="ex">PICPPNBC_00009</span>  VFG023006<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">WP_013827070</span><span class="kw">)</span>  <span class="ex">25.362</span>  188 383 138 0.95    30.0</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a><span class="ex">PICPPNBC_00010</span>  VFG005490<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">WP_000472995</span><span class="kw">)</span>  <span class="ex">30.556</span>  237 1906    72  0.41    32.0</span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a><span class="ex">PICPPNBC_00011</span>  VFG026864<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">WP_005058381</span><span class="kw">)</span>  <span class="ex">34.328</span>  134 738 67  0.47    30.0</span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a><span class="ex">PICPPNBC_00012</span>  VFG043573<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">NP_219906</span><span class="kw">)</span> <span class="ex">61.333</span>  638 660 600 0.0 719</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a><span class="ex">PICPPNBC_00013</span>  VFG015290<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">WP_005763320</span><span class="kw">)</span>  <span class="ex">44.444</span>  376 488 63  8.46e-09    57.0</span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a><span class="ex">PICPPNBC_00014</span>  VFG015888<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">WP_003381691</span><span class="kw">)</span>  <span class="ex">26.144</span>  370 260 153 0.37    32.3</span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a><span class="ex">PICPPNBC_00015</span>  VFG029464<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">WP_014380395</span><span class="kw">)</span>  <span class="ex">33.333</span>  46  971 30  5.0 23.9</span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a><span class="ex">PICPPNBC_00018</span>  VFG044083<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">NP_250948</span><span class="kw">)</span> <span class="ex">26.946</span>  299 312 167 1.23e-05    45.8</span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a><span class="ex">PICPPNBC_00019</span>  VFG010667<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">WP_015961409</span><span class="kw">)</span>  <span class="ex">34.783</span>  125 425 46  0.63    29.3</span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a><span class="ex">PICPPNBC_00020</span>  VFG011263<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">WP_003811903</span><span class="kw">)</span>  <span class="ex">27.848</span>  91  790 79  2.5 26.6</span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a><span class="ex">PICPPNBC_00021</span>  VFG013269<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">WP_010945211</span><span class="kw">)</span>  <span class="ex">33.333</span>  87  196 48  2.0 26.6</span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a><span class="ex">PICPPNBC_00022</span>  VFG050230<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">NP_219645</span><span class="kw">)</span> <span class="ex">31.746</span>  313 285 63  2.6 29.6</span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a><span class="ex">PICPPNBC_00023</span>  VFG007107<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">WP_047863677</span><span class="kw">)</span>  <span class="ex">21.673</span>  938 677 263 0.017   38.9</span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a><span class="ex">PICPPNBC_00024</span>  VFG006812<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">WP_003729506</span><span class="kw">)</span>  <span class="ex">32.450</span>  164 154 151 1.53e-18    77.4</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Una vez tenemos este archivo vamos a aplicarle ciertos filtros:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="va">blast_vf_i</span><span class="op">=</span>0</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="va">blast_vf_c</span><span class="op">=</span>0.9</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="va">blast_vf_e</span><span class="op">=</span><span class="st">"1e-50"</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="va">BASE_DIR</span><span class="op">=</span><span class="st">"</span><span class="va">$HOME</span><span class="st">/2025_Demo_ecoli"</span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a><span class="va">samples</span><span class="op">=</span><span class="va">$(</span><span class="fu">find</span> <span class="st">"</span><span class="va">$BASE_DIR</span><span class="st">/2_annotation/prokka"</span> <span class="at">-maxdepth</span> 1 <span class="at">-type</span> d <span class="at">-not</span> <span class="at">-path</span> <span class="st">"</span><span class="va">$BASE_DIR</span><span class="st">/2_annotation/prokka"</span> <span class="at">-exec</span> basename {} <span class="dt">\;</span><span class="va">)</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> s <span class="kw">in</span> <span class="va">$samples</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> <span class="bu">[</span> <span class="ot">-f</span> <span class="st">"</span><span class="va">$BASE_DIR</span><span class="st">/2_annotation/</span><span class="va">$s</span><span class="st">.out"</span> <span class="bu">]</span><span class="kw">;</span> <span class="cf">then</span></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>      <span class="fu">awk</span> <span class="at">-F</span> <span class="st">'\t'</span> <span class="st">'$3&gt;='</span><span class="va">$blast_vf_i</span><span class="st">' &amp;&amp; ($6/$5&gt;='</span><span class="va">$blast_vf_c</span><span class="st">') &amp;&amp; $(NF-1)&lt;='</span><span class="va">$blast_vf_e</span><span class="st">' {print}'</span> <span class="dt">\</span></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>        <span class="st">"</span><span class="va">$BASE_DIR</span><span class="st">/2_annotation/</span><span class="va">$s</span><span class="st">.out"</span> <span class="op">&gt;</span> <span class="st">"</span><span class="va">$BASE_DIR</span><span class="st">/2_annotation/</span><span class="va">$s</span><span class="st">.filtered"</span></span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>    <span class="cf">fi</span></span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Esto nos genera un versión del archivo filtrada <code>.filter</code>:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="ex">MFNFHKLG_00013</span>  VFG043573<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">NP_219906</span><span class="kw">)</span> <span class="ex">61.333</span>  638 660 600 0.0 719</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="ex">MFNFHKLG_00022</span>  VFG034856<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">WP_000181251</span><span class="kw">)</span>  <span class="ex">100.000</span> 473 473 473 0.0 970</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="ex">MFNFHKLG_00031</span>  VFG047720<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">WP_012280231</span><span class="kw">)</span>  <span class="ex">48.953</span>  382 388 382 3.83e-130   378</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="ex">MFNFHKLG_00032</span>  VFG047710<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">WP_014547360</span><span class="kw">)</span>  <span class="ex">58.116</span>  1073    1094    1072    0.0 1218</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a><span class="ex">MFNFHKLG_00037</span>  VFG030400<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">WP_081465621</span><span class="kw">)</span>  <span class="ex">32.186</span>  517 507 494 1.31e-71    236</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a><span class="ex">MFNFHKLG_00071</span>  VFG009376<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">WP_003893754</span><span class="kw">)</span>  <span class="ex">46.701</span>  201 197 197 1.04e-57    180</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a><span class="ex">MFNFHKLG_00094</span>  VFG013414<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">WP_012054478</span><span class="kw">)</span>  <span class="ex">77.303</span>  305 305 304 2.18e-180   499</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a><span class="ex">MFNFHKLG_00096</span>  VFG023546<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">YP_001703132</span><span class="kw">)</span>  <span class="ex">37.101</span>  901 772 814 6.49e-139   432</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a><span class="ex">MFNFHKLG_00102</span>  VFG046022<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">WP_000157234</span><span class="kw">)</span>  <span class="ex">100.000</span> 400 400 400 0.0 809</span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a><span class="ex">MFNFHKLG_00103</span>  VFG045984<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">WP_001025155</span><span class="kw">)</span>  <span class="ex">100.000</span> 461 461 461 0.0 937</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a><span class="ex">MFNFHKLG_00104</span>  VFG045947<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">WP_000360894</span><span class="kw">)</span>  <span class="ex">100.000</span> 146 146 146 9.84e-109   305</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a><span class="ex">MFNFHKLG_00123</span>  VFG023850<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">WP_043955033</span><span class="kw">)</span>  <span class="ex">35.922</span>  308 315 309 2.30e-56    184</span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a><span class="ex">MFNFHKLG_00131</span>  VFG004221<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">WP_000924983</span><span class="kw">)</span>  <span class="ex">32.783</span>  424 391 424 2.97e-52    179</span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a><span class="ex">MFNFHKLG_00135</span>  VFG021422<span class="er">(</span><span class="ex">gb</span><span class="kw">|</span><span class="ex">WP_071586201</span><span class="kw">)</span>  <span class="ex">62.529</span>  866 881 854 0.0 1108</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Con los datos de la base de datos <code>FASTA</code> podemos crear un archivos con el nombre de cada identificador, y asi sea más sencillo anotar:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">awk</span> <span class="st">'/^&gt;/ {if (NR&gt;1) print ""; split($0, a, " "); id=a[1]; sub(/^&gt;/, "", id); gene=""; desc=""; org=""; for(i=2; i&lt;=length(a); i++) {if (a[i] ~ /^\(/) {gene=a[i]; sub(/^\(/, "", gene); sub(/\)$/, "", gene)} else if (a[i] ~ /^\[/) {org=a[i]; sub(/^\[/, "", org); sub(/\]$/, "", org)} else {desc=desc" "a[i]}} print id"\t"gene"\t"desc"\t"org} END {if (NR&gt;0) print ""}'</span> <span class="va">$HOME</span>/2025_Demo_ecoli/data/VFDB/VFDB_setB_pro.fas <span class="op">&gt;</span> <span class="va">$HOME</span>/2025_Demo_ecoli/data/VFDB/VFDB_setB_pro.tab</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Este archivo <code>.tab</code> se vería algo así:</p>
<div class="cell">
<div class="cell-output-display">
<div id="mvktqpynqn" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#mvktqpynqn table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#mvktqpynqn thead, #mvktqpynqn tbody, #mvktqpynqn tfoot, #mvktqpynqn tr, #mvktqpynqn td, #mvktqpynqn th {
  border-style: none;
}

#mvktqpynqn p {
  margin: 0;
  padding: 0;
}

#mvktqpynqn .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#mvktqpynqn .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#mvktqpynqn .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#mvktqpynqn .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#mvktqpynqn .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#mvktqpynqn .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#mvktqpynqn .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#mvktqpynqn .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#mvktqpynqn .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#mvktqpynqn .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#mvktqpynqn .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#mvktqpynqn .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#mvktqpynqn .gt_spanner_row {
  border-bottom-style: hidden;
}

#mvktqpynqn .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#mvktqpynqn .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#mvktqpynqn .gt_from_md > :first-child {
  margin-top: 0;
}

#mvktqpynqn .gt_from_md > :last-child {
  margin-bottom: 0;
}

#mvktqpynqn .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#mvktqpynqn .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#mvktqpynqn .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#mvktqpynqn .gt_row_group_first td {
  border-top-width: 2px;
}

#mvktqpynqn .gt_row_group_first th {
  border-top-width: 2px;
}

#mvktqpynqn .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#mvktqpynqn .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#mvktqpynqn .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#mvktqpynqn .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#mvktqpynqn .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#mvktqpynqn .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#mvktqpynqn .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#mvktqpynqn .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#mvktqpynqn .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#mvktqpynqn .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#mvktqpynqn .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#mvktqpynqn .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#mvktqpynqn .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#mvktqpynqn .gt_left {
  text-align: left;
}

#mvktqpynqn .gt_center {
  text-align: center;
}

#mvktqpynqn .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#mvktqpynqn .gt_font_normal {
  font-weight: normal;
}

#mvktqpynqn .gt_font_bold {
  font-weight: bold;
}

#mvktqpynqn .gt_font_italic {
  font-style: italic;
}

#mvktqpynqn .gt_super {
  font-size: 65%;
}

#mvktqpynqn .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#mvktqpynqn .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#mvktqpynqn .gt_indent_1 {
  text-indent: 5px;
}

#mvktqpynqn .gt_indent_2 {
  text-indent: 10px;
}

#mvktqpynqn .gt_indent_3 {
  text-indent: 15px;
}

#mvktqpynqn .gt_indent_4 {
  text-indent: 20px;
}

#mvktqpynqn .gt_indent_5 {
  text-indent: 25px;
}

#mvktqpynqn .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#mvktqpynqn div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead>
<tr class="gt_col_headings header">
<th id="VFG037170(gb|WP_001081754)" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">VFG037170(gb|WP_001081754)</th>
<th id="VFC0235)]" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">VFC0235)]</th>
<th id="phospholipase-C-C---Exotoxin-baumannii-1656-2]" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">phospholipase C C - Exotoxin baumannii 1656-2]</th>
<th id="Acinetobacter" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Acinetobacter</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="VFG037170(gb|WP_001081754)">VFG037177(gb|WP_000632986)</td>
<td class="gt_row gt_left" headers="VFC0235)]">VFC0235)]</td>
<td class="gt_row gt_left" headers="phospholipase C C - Exotoxin baumannii 1656-2]">phospholipase C C - Exotoxin baumannii ACICU]</td>
<td class="gt_row gt_left" headers="Acinetobacter">Acinetobacter</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="VFG037170(gb|WP_001081754)">VFG037172(gb|WP_001081748)</td>
<td class="gt_row gt_left" headers="VFC0235)]">VFC0235)]</td>
<td class="gt_row gt_left" headers="phospholipase C C - Exotoxin baumannii 1656-2]">phospholipase C C - Exotoxin baumannii AB0057]</td>
<td class="gt_row gt_left" headers="Acinetobacter">Acinetobacter</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="VFG037170(gb|WP_001081754)">VFG037173(gb|WP_000633004)</td>
<td class="gt_row gt_left" headers="VFC0235)]">VFC0235)]</td>
<td class="gt_row gt_left" headers="phospholipase C C - Exotoxin baumannii 1656-2]">phospholipase C C - Exotoxin baumannii AB0057]</td>
<td class="gt_row gt_left" headers="Acinetobacter">Acinetobacter</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="VFG037170(gb|WP_001081754)">VFG037176(gb|WP_001081735)</td>
<td class="gt_row gt_left" headers="VFC0235)]">VFC0235)]</td>
<td class="gt_row gt_left" headers="phospholipase C C - Exotoxin baumannii 1656-2]">phospholipase C C - Exotoxin baumannii ACICU]</td>
<td class="gt_row gt_left" headers="Acinetobacter">Acinetobacter</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="VFG037170(gb|WP_001081754)">VFG037178(gb|YP_001083140)</td>
<td class="gt_row gt_left" headers="VFC0235)]">VFC0235)]</td>
<td class="gt_row gt_left" headers="phospholipase C C - Exotoxin baumannii 1656-2]">phospholipase C C - Exotoxin baumannii ATCC 17978]</td>
<td class="gt_row gt_left" headers="Acinetobacter">Acinetobacter</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="VFG037170(gb|WP_001081754)">VFG037179(gb|YP_001085084)</td>
<td class="gt_row gt_left" headers="VFC0235)]">VFC0235)]</td>
<td class="gt_row gt_left" headers="phospholipase C C - Exotoxin baumannii 1656-2]">phospholipase C C - Exotoxin baumannii ATCC 17978]</td>
<td class="gt_row gt_left" headers="Acinetobacter">Acinetobacter</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="VFG037170(gb|WP_001081754)">VFG037182(gb|WP_001081755)</td>
<td class="gt_row gt_left" headers="VFC0235)]">VFC0235)]</td>
<td class="gt_row gt_left" headers="phospholipase C C - Exotoxin baumannii 1656-2]">phospholipase C C - Exotoxin baumannii BJAB07104]</td>
<td class="gt_row gt_left" headers="Acinetobacter">Acinetobacter</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="VFG037170(gb|WP_001081754)">VFG037184(gb|WP_001081758)</td>
<td class="gt_row gt_left" headers="VFC0235)]">VFC0235)]</td>
<td class="gt_row gt_left" headers="phospholipase C C - Exotoxin baumannii 1656-2]">phospholipase C C - Exotoxin baumannii BJAB0715]</td>
<td class="gt_row gt_left" headers="Acinetobacter">Acinetobacter</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="VFG037170(gb|WP_001081754)">VFG037185(gb|WP_020753227)</td>
<td class="gt_row gt_left" headers="VFC0235)]">VFC0235)]</td>
<td class="gt_row gt_left" headers="phospholipase C C - Exotoxin baumannii 1656-2]">phospholipase C C - Exotoxin baumannii BJAB0715]</td>
<td class="gt_row gt_left" headers="Acinetobacter">Acinetobacter</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="VFG037170(gb|WP_001081754)">VFG037186(gb|WP_001081765)</td>
<td class="gt_row gt_left" headers="VFC0235)]">VFC0235)]</td>
<td class="gt_row gt_left" headers="phospholipase C C - Exotoxin baumannii 1656-2]">phospholipase C C - Exotoxin baumannii BJAB0868]</td>
<td class="gt_row gt_left" headers="Acinetobacter">Acinetobacter</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
<p>Podemos hacer un posterior parseo de datos con el lenguaje de programación qué mas prefieras para llegar a algo asi para cada genoma:</p>
<div class="cell">
<div class="cell-output-display">
<div id="cnqsljrrlf" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#cnqsljrrlf table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#cnqsljrrlf thead, #cnqsljrrlf tbody, #cnqsljrrlf tfoot, #cnqsljrrlf tr, #cnqsljrrlf td, #cnqsljrrlf th {
  border-style: none;
}

#cnqsljrrlf p {
  margin: 0;
  padding: 0;
}

#cnqsljrrlf .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#cnqsljrrlf .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#cnqsljrrlf .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#cnqsljrrlf .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#cnqsljrrlf .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#cnqsljrrlf .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#cnqsljrrlf .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#cnqsljrrlf .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#cnqsljrrlf .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#cnqsljrrlf .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#cnqsljrrlf .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#cnqsljrrlf .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#cnqsljrrlf .gt_spanner_row {
  border-bottom-style: hidden;
}

#cnqsljrrlf .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#cnqsljrrlf .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#cnqsljrrlf .gt_from_md > :first-child {
  margin-top: 0;
}

#cnqsljrrlf .gt_from_md > :last-child {
  margin-bottom: 0;
}

#cnqsljrrlf .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#cnqsljrrlf .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#cnqsljrrlf .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#cnqsljrrlf .gt_row_group_first td {
  border-top-width: 2px;
}

#cnqsljrrlf .gt_row_group_first th {
  border-top-width: 2px;
}

#cnqsljrrlf .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#cnqsljrrlf .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#cnqsljrrlf .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#cnqsljrrlf .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#cnqsljrrlf .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#cnqsljrrlf .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#cnqsljrrlf .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#cnqsljrrlf .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#cnqsljrrlf .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#cnqsljrrlf .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#cnqsljrrlf .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#cnqsljrrlf .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#cnqsljrrlf .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#cnqsljrrlf .gt_left {
  text-align: left;
}

#cnqsljrrlf .gt_center {
  text-align: center;
}

#cnqsljrrlf .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#cnqsljrrlf .gt_font_normal {
  font-weight: normal;
}

#cnqsljrrlf .gt_font_bold {
  font-weight: bold;
}

#cnqsljrrlf .gt_font_italic {
  font-style: italic;
}

#cnqsljrrlf .gt_super {
  font-size: 65%;
}

#cnqsljrrlf .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#cnqsljrrlf .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#cnqsljrrlf .gt_indent_1 {
  text-indent: 5px;
}

#cnqsljrrlf .gt_indent_2 {
  text-indent: 10px;
}

#cnqsljrrlf .gt_indent_3 {
  text-indent: 15px;
}

#cnqsljrrlf .gt_indent_4 {
  text-indent: 20px;
}

#cnqsljrrlf .gt_indent_5 {
  text-indent: 25px;
}

#cnqsljrrlf .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#cnqsljrrlf div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>

<table class="gt_table caption-top table table-sm table-striped small" data-quarto-postprocess="true" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead>
<tr class="gt_col_headings header">
<th id="Query" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Query</th>
<th id="Target" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Target</th>
<th id="Identity" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">Identity</th>
<th id="Query_len" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">Query_len</th>
<th id="Target_len" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">Target_len</th>
<th id="Align_len" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">Align_len</th>
<th id="e_value" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">e_value</th>
<th id="bitscore" class="gt_col_heading gt_columns_bottom_border gt_right" data-quarto-table-cell-role="th" scope="col">bitscore</th>
<th id="Gene" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">Gene</th>
<th id="VF_Name" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">VF_Name</th>
<th id="VF_ID" class="gt_col_heading gt_columns_bottom_border gt_left" data-quarto-table-cell-role="th" scope="col">VF_ID</th>
<th id="VF_Category" class="gt_col_heading gt_columns_bottom_border gt_center" data-quarto-table-cell-role="th" scope="col">VF_Category</th>
<th id="VF_Category_ID" class="gt_col_heading gt_columns_bottom_border gt_center" data-quarto-table-cell-role="th" scope="col">VF_Category_ID</th>
<th id="Description" class="gt_col_heading gt_columns_bottom_border gt_center" data-quarto-table-cell-role="th" scope="col">Description</th>
<th id="Species" class="gt_col_heading gt_columns_bottom_border gt_center" data-quarto-table-cell-role="th" scope="col">Species</th>
</tr>
</thead>
<tbody class="gt_table_body">
<tr class="odd">
<td class="gt_row gt_left" headers="Query">MFNFHKLG_00013</td>
<td class="gt_row gt_left" headers="Target">VFG043573(gb|NP_219906)</td>
<td class="gt_row gt_right" headers="Identity">61.333</td>
<td class="gt_row gt_right" headers="Query_len">638</td>
<td class="gt_row gt_right" headers="Target_len">660</td>
<td class="gt_row gt_right" headers="Align_len">600</td>
<td class="gt_row gt_right" headers="e_value">0.00e+00</td>
<td class="gt_row gt_right" headers="bitscore">719</td>
<td class="gt_row gt_left" headers="Gene">VFC0001)]</td>
<td class="gt_row gt_left" headers="VF_Name">chaperone protein DnaK porin - Adherence trachomatis D/UW-3/CX]</td>
<td class="gt_row gt_left" headers="VF_ID">Chlamydia</td>
<td class="gt_row gt_center" headers="VF_Category">NA</td>
<td class="gt_row gt_center" headers="VF_Category_ID">NA</td>
<td class="gt_row gt_center" headers="Description">NA</td>
<td class="gt_row gt_center" headers="Species">NA</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Query">MFNFHKLG_00022</td>
<td class="gt_row gt_left" headers="Target">VFG034856(gb|WP_000181251)</td>
<td class="gt_row gt_right" headers="Identity">100.000</td>
<td class="gt_row gt_right" headers="Query_len">473</td>
<td class="gt_row gt_right" headers="Target_len">473</td>
<td class="gt_row gt_right" headers="Align_len">473</td>
<td class="gt_row gt_right" headers="e_value">0.00e+00</td>
<td class="gt_row gt_right" headers="bitscore">970</td>
<td class="gt_row gt_left" headers="Gene">VFC0086)]</td>
<td class="gt_row gt_left" headers="VF_Name">Type III secretion system effector EspX1 secreted effectors - Effector delivery system coli O103:H2 str. 12009]</td>
<td class="gt_row gt_left" headers="VF_ID">Escherichia</td>
<td class="gt_row gt_center" headers="VF_Category">NA</td>
<td class="gt_row gt_center" headers="VF_Category_ID">NA</td>
<td class="gt_row gt_center" headers="Description">NA</td>
<td class="gt_row gt_center" headers="Species">NA</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Query">MFNFHKLG_00031</td>
<td class="gt_row gt_left" headers="Target">VFG047720(gb|WP_012280231)</td>
<td class="gt_row gt_right" headers="Identity">48.953</td>
<td class="gt_row gt_right" headers="Query_len">382</td>
<td class="gt_row gt_right" headers="Target_len">388</td>
<td class="gt_row gt_right" headers="Align_len">382</td>
<td class="gt_row gt_right" headers="e_value">3.83e-130</td>
<td class="gt_row gt_right" headers="bitscore">378</td>
<td class="gt_row gt_left" headers="Gene">VFC0272)]</td>
<td class="gt_row gt_left" headers="VF_Name">carbamoyl phosphate synthase small subunit biosynthesis - Nutritional/Metabolic factor philomiragia subsp. philomiragia ATCC 25017]</td>
<td class="gt_row gt_left" headers="VF_ID">Francisella</td>
<td class="gt_row gt_center" headers="VF_Category">NA</td>
<td class="gt_row gt_center" headers="VF_Category_ID">NA</td>
<td class="gt_row gt_center" headers="Description">NA</td>
<td class="gt_row gt_center" headers="Species">NA</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Query">MFNFHKLG_00032</td>
<td class="gt_row gt_left" headers="Target">VFG047710(gb|WP_014547360)</td>
<td class="gt_row gt_right" headers="Identity">58.116</td>
<td class="gt_row gt_right" headers="Query_len">1073</td>
<td class="gt_row gt_right" headers="Target_len">1094</td>
<td class="gt_row gt_right" headers="Align_len">1072</td>
<td class="gt_row gt_right" headers="e_value">0.00e+00</td>
<td class="gt_row gt_right" headers="bitscore">1218</td>
<td class="gt_row gt_left" headers="Gene">VFC0272)]</td>
<td class="gt_row gt_left" headers="VF_Name">carbamoyl phosphate synthase large subunit biosynthesis - Nutritional/Metabolic factor cf. tularensis subsp. novicida 3523]</td>
<td class="gt_row gt_left" headers="VF_ID">Francisella</td>
<td class="gt_row gt_center" headers="VF_Category">NA</td>
<td class="gt_row gt_center" headers="VF_Category_ID">NA</td>
<td class="gt_row gt_center" headers="Description">NA</td>
<td class="gt_row gt_center" headers="Species">NA</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Query">MFNFHKLG_00037</td>
<td class="gt_row gt_left" headers="Target">VFG030400(gb|WP_081465621)</td>
<td class="gt_row gt_right" headers="Identity">32.186</td>
<td class="gt_row gt_right" headers="Query_len">517</td>
<td class="gt_row gt_right" headers="Target_len">507</td>
<td class="gt_row gt_right" headers="Align_len">494</td>
<td class="gt_row gt_right" headers="e_value">1.31e-71</td>
<td class="gt_row gt_right" headers="bitscore">236</td>
<td class="gt_row gt_left" headers="Gene">VFC0258)]</td>
<td class="gt_row gt_left" headers="VF_Name">long-chain fatty acid--CoA ligase operon - Immune modulation sp. JDM601]</td>
<td class="gt_row gt_left" headers="VF_ID">Mycobacterium</td>
<td class="gt_row gt_center" headers="VF_Category">NA</td>
<td class="gt_row gt_center" headers="VF_Category_ID">NA</td>
<td class="gt_row gt_center" headers="Description">NA</td>
<td class="gt_row gt_center" headers="Species">NA</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Query">MFNFHKLG_00071</td>
<td class="gt_row gt_left" headers="Target">VFG009376(gb|WP_003893754)</td>
<td class="gt_row gt_right" headers="Identity">46.701</td>
<td class="gt_row gt_right" headers="Query_len">201</td>
<td class="gt_row gt_right" headers="Target_len">197</td>
<td class="gt_row gt_right" headers="Align_len">197</td>
<td class="gt_row gt_right" headers="e_value">1.04e-57</td>
<td class="gt_row gt_right" headers="bitscore">180</td>
<td class="gt_row gt_left" headers="Gene">VFC0272)]</td>
<td class="gt_row gt_left" headers="VF_Name">3-isopropylmalate dehydratase small subunit synthesis - Nutritional/Metabolic factor smegmatis str. MC2 155]</td>
<td class="gt_row gt_left" headers="VF_ID">Mycobacterium</td>
<td class="gt_row gt_center" headers="VF_Category">NA</td>
<td class="gt_row gt_center" headers="VF_Category_ID">NA</td>
<td class="gt_row gt_center" headers="Description">NA</td>
<td class="gt_row gt_center" headers="Species">NA</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Query">MFNFHKLG_00094</td>
<td class="gt_row gt_left" headers="Target">VFG013414(gb|WP_012054478)</td>
<td class="gt_row gt_right" headers="Identity">77.303</td>
<td class="gt_row gt_right" headers="Query_len">305</td>
<td class="gt_row gt_right" headers="Target_len">305</td>
<td class="gt_row gt_right" headers="Align_len">304</td>
<td class="gt_row gt_right" headers="e_value">2.18e-180</td>
<td class="gt_row gt_right" headers="bitscore">499</td>
<td class="gt_row gt_left" headers="Gene">VFC0258)]</td>
<td class="gt_row gt_left" headers="VF_Name">UDP-3-O-(R-3-hydroxymyristoyl)-N-acetylglucosamine deacetylase - Immune modulation influenzae PittEE]</td>
<td class="gt_row gt_left" headers="VF_ID">Haemophilus</td>
<td class="gt_row gt_center" headers="VF_Category">NA</td>
<td class="gt_row gt_center" headers="VF_Category_ID">NA</td>
<td class="gt_row gt_center" headers="Description">NA</td>
<td class="gt_row gt_center" headers="Species">NA</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Query">MFNFHKLG_00096</td>
<td class="gt_row gt_left" headers="Target">VFG023546(gb|YP_001703132)</td>
<td class="gt_row gt_right" headers="Identity">37.101</td>
<td class="gt_row gt_right" headers="Query_len">901</td>
<td class="gt_row gt_right" headers="Target_len">772</td>
<td class="gt_row gt_right" headers="Align_len">814</td>
<td class="gt_row gt_right" headers="e_value">6.49e-139</td>
<td class="gt_row gt_right" headers="bitscore">432</td>
<td class="gt_row gt_left" headers="Gene">VFC0346)]</td>
<td class="gt_row gt_left" headers="VF_Name">accessory Sec system translocase SecA2 secretion factor - Others abscessus ATCC 19977]</td>
<td class="gt_row gt_left" headers="VF_ID">Mycobacterium</td>
<td class="gt_row gt_center" headers="VF_Category">NA</td>
<td class="gt_row gt_center" headers="VF_Category_ID">NA</td>
<td class="gt_row gt_center" headers="Description">NA</td>
<td class="gt_row gt_center" headers="Species">NA</td>
</tr>
<tr class="odd">
<td class="gt_row gt_left" headers="Query">MFNFHKLG_00102</td>
<td class="gt_row gt_left" headers="Target">VFG046022(gb|WP_000157234)</td>
<td class="gt_row gt_right" headers="Identity">100.000</td>
<td class="gt_row gt_right" headers="Query_len">400</td>
<td class="gt_row gt_right" headers="Target_len">400</td>
<td class="gt_row gt_right" headers="Align_len">400</td>
<td class="gt_row gt_right" headers="e_value">0.00e+00</td>
<td class="gt_row gt_right" headers="bitscore">809</td>
<td class="gt_row gt_left" headers="Gene">VFC0001)]</td>
<td class="gt_row gt_left" headers="VF_Name">protein transport protein HofC E.coli pilus - Adherence coli O103:H2 str. 12009]</td>
<td class="gt_row gt_left" headers="VF_ID">Escherichia</td>
<td class="gt_row gt_center" headers="VF_Category">NA</td>
<td class="gt_row gt_center" headers="VF_Category_ID">NA</td>
<td class="gt_row gt_center" headers="Description">NA</td>
<td class="gt_row gt_center" headers="Species">NA</td>
</tr>
<tr class="even">
<td class="gt_row gt_left" headers="Query">MFNFHKLG_00103</td>
<td class="gt_row gt_left" headers="Target">VFG045984(gb|WP_001025155)</td>
<td class="gt_row gt_right" headers="Identity">100.000</td>
<td class="gt_row gt_right" headers="Query_len">461</td>
<td class="gt_row gt_right" headers="Target_len">461</td>
<td class="gt_row gt_right" headers="Align_len">461</td>
<td class="gt_row gt_right" headers="e_value">0.00e+00</td>
<td class="gt_row gt_right" headers="bitscore">937</td>
<td class="gt_row gt_left" headers="Gene">VFC0001)]</td>
<td class="gt_row gt_left" headers="VF_Name">type II secretion system protein GspE E.coli pilus - Adherence coli O103:H2 str. 12009]</td>
<td class="gt_row gt_left" headers="VF_ID">Escherichia</td>
<td class="gt_row gt_center" headers="VF_Category">NA</td>
<td class="gt_row gt_center" headers="VF_Category_ID">NA</td>
<td class="gt_row gt_center" headers="Description">NA</td>
<td class="gt_row gt_center" headers="Species">NA</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
</section>
</section>
<section id="pangenoma" class="level1">
<h1>Pangenoma</h1>
<p>Vamos a utilizar Roary, lo puedes descargar <a href="https://github.com/sanger-pathogens/Roary">aquí</a>.</p>
<p>Preparamos los archivos <code>.gff</code> obtenidos de la anotación con <code>Prokka</code> o <code>Bakta</code> para poder realizar el análisis de pangenoma.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="va">DIR_ORIGEN</span><span class="op">=</span><span class="st">"</span><span class="va">$HOME</span><span class="st">/2025_Demo_ecoli/2_annotation/prokka"</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="va">DIR_DESTINO</span><span class="op">=</span><span class="st">"</span><span class="va">$HOME</span><span class="st">/2025_Demo_ecoli/3_pangenome/data_genomes"</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> carpeta <span class="kw">in</span> <span class="st">"</span><span class="va">$DIR_ORIGEN</span><span class="st">"</span>/<span class="pp">*</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> <span class="kw">[[</span> <span class="ot">-d</span> <span class="st">"</span><span class="va">$carpeta</span><span class="st">"</span> <span class="kw">]];</span> <span class="cf">then</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>            <span class="va">nombre_carpeta</span><span class="op">=</span><span class="va">$(</span><span class="fu">basename</span> <span class="st">"</span><span class="va">$carpeta</span><span class="st">"</span><span class="va">)</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Buscar el primer archivo .gff o .gff3 en la carpeta </span></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>            <span class="va">archivo_gff</span><span class="op">=</span><span class="va">$(</span><span class="fu">find</span> <span class="st">"</span><span class="va">$carpeta</span><span class="st">"</span> <span class="at">-maxdepth</span> 1 <span class="at">-type</span> f <span class="dt">\(</span> <span class="at">-iname</span> <span class="st">"*.gff"</span> <span class="at">-o</span> <span class="at">-iname</span> <span class="st">"*.gff3"</span> <span class="dt">\)</span> <span class="at">-print</span> <span class="at">-quit</span> <span class="kw">||</span> <span class="fu">true</span><span class="va">)</span></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>            <span class="cf">if</span> <span class="kw">[[</span> <span class="ot">-n</span> <span class="st">"</span><span class="va">$archivo_gff</span><span class="st">"</span> <span class="kw">]];</span> <span class="cf">then</span></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>                <span class="va">destino</span><span class="op">=</span><span class="st">"</span><span class="va">$DIR_DESTINO</span><span class="st">/</span><span class="va">${nombre_carpeta}</span><span class="st">.gff"</span></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>                <span class="fu">cp</span> <span class="at">-f</span> <span class="at">--</span> <span class="st">"</span><span class="va">$archivo_gff</span><span class="st">"</span> <span class="st">"</span><span class="va">$destino</span><span class="st">"</span></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>                <span class="bu">echo</span> <span class="st">"Archivo </span><span class="va">$archivo_gff</span><span class="st"> copiado como </span><span class="va">$destino</span><span class="st">"</span></span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a>            <span class="cf">else</span></span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a>                <span class="bu">echo</span> <span class="st">"No se encontró archivo .gff/.gff3 en </span><span class="va">$carpeta</span><span class="st">"</span></span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a>            <span class="cf">fi</span></span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a>        <span class="cf">fi</span></span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a>    <span class="cf">done</span></span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>PRECAUCIÓN</strong> ⚠️: con 8 núcleos de procesamiento este paso demora <strong>65 horas</strong>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="va">DIR_ORIGEN</span><span class="op">=</span><span class="st">"</span><span class="va">$HOME</span><span class="st">/2025_Demo_ecoli/2_annotation/prokka"</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="va">DIR_DESTINO</span><span class="op">=</span><span class="st">"</span><span class="va">$HOME</span><span class="st">/2025_Demo_ecoli/3_pangenome/data_genomes"</span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a><span class="ex">roary</span> <span class="at">-f</span> <span class="st">"</span><span class="va">$DIR_DESTINO</span><span class="st">/pangenome_prokka_prank"</span> <span class="dt">\</span></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>-e <span class="dt">\</span></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>-p 8 <span class="dt">\</span></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>-v <span class="st">"</span><span class="va">$DIR_DESTINO</span><span class="st">"</span>/<span class="pp">*</span>.gff</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Filogénia. Generamos un archivo <code>.newick</code></p>
<p><strong>PRECAUCIÓN</strong> ⚠️: La memoría RAM se dispara hasta 23 GB para la construcción del árbol. 40 minutos de proceso.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="ex">FastTree</span> <span class="dt">\</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>-nt <span class="dt">\</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>-gtr <span class="dt">\</span></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>3_pangenome/data_genomes/pangenome_prokka_prank/core_gene_alignment.aln <span class="op">&gt;</span> my_tree.newick  </span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="visualización" class="level3">
<h3 class="anchored" data-anchor-id="visualización">Visualización</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="ex">./bin/Roary/contrib/roary_plots/roary_plots.py</span> 3_pangenome/data_genomes/pangenome_prokka_prank/my_tree.newick 3_pangenome/data_genomes/pangenome_prokka_prank/gene_presence_absence.csv</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>Matriz</strong></p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/pangenome_matrix.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<p><strong>Pie chart</strong></p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/pangenome_pie.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<p><strong>Frecuencia</strong></p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/pangenome_frequency.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
</section>
</section>
<section id="filogénia" class="level1">
<h1>Filogénia</h1>
<p>Para poder calcular filogénias podemos utilizar diversas herramientas según sea tu capacidad computacional:</p>
<ul>
<li><p><a href="https://github.com/amkozlov/raxml-ng"><strong>RAxML Next Generation</strong></a></p></li>
<li><p><a href="https://www.megasoftware.net/"><strong>MEGA</strong></a></p></li>
<li><p><a href="https://github.com/morgannprice/fasttree"><strong>FastTree</strong></a></p></li>
<li><p><a href="https://iqtree.github.io/"><strong>IQ-tree</strong></a></p></li>
</ul>
<p>Para este tutorial vamos a utilizar <code>FastTree</code>. En todas las herramientas el archivo de salida del pangenoma <code>.aln</code> te será util como <code>input</code>.</p>
<p>Este alineamiento tambien se puede lograr con las siguientes herramientas que nos permiten alinear las secuencias para identificar regiones homólogas. Esto se hace con programas como:</p>
<ul>
<li><p><a href="https://www.drive5.com/muscle/muscle.html"><strong>MUSCLE</strong></a>: Alineamiento rápido de múltiples secuencias.</p></li>
<li><p><a href="https://www.genome.jp/tools-bin/clustalw"><strong>ClustalW</strong></a>: Alineamiento progresivo.</p></li>
<li><p><a href="https://mafft.cbrc.jp/alignment/server/index.html"><strong>MAFFT</strong></a>: Alineamiento eficiente para grandes conjuntos de datos. (Este utiliza <code>Roary</code>)</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="ex">FastTree</span> <span class="dt">\</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>-nt <span class="dt">\</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>-gtr <span class="dt">\</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>3_pangenome/data_genomes/pangenome_prokka_prank/core_gene_alignment.aln <span class="op">&gt;</span> my_tree.newick  </span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Con el alineamiento, los programas infieren el árbol filogenético usando métodos como:</p>
<ul>
<li><p><strong>Distancia</strong>: Basado en diferencias entre secuencias (por ejemplo, Neighbor-Joining).</p></li>
<li><p><strong>Máxima parsimonia</strong>: Busca el árbol que requiere el menor número de cambios evolutivos.</p></li>
<li><p><strong>Máxima verosimilitud</strong>: Estima el árbol más probable según un modelo evolutivo (por ejemplo, RAxML, IQ-TREE).</p></li>
<li><p><strong>Inferencia bayesiana</strong>: Usa probabilidades para inferir el árbol (por ejemplo, MrBayes).</p></li>
</ul>
<p>El formato de árbol <code>Newick</code> (o notación Newick , o formato de árbol de Nuevo Hampshire) es una manera de representar árboles utilizando paréntesis y comas. Este lo podemos leer directamente en interfaz gráfica como <code>MEGA</code>, en <code>R</code> o en <code>Python</code>.</p>
<p>Así se ve un archivo crudo <code>.newick</code>:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="kw">(</span><span class="ex">GCF_000026265.1:0.003181433,</span><span class="er">(</span><span class="ex">GCF_029227895.1:0.000320651,GCF_029625135.1:0.000522578</span><span class="kw">)</span><span class="ex">1.000:0.002783087,</span><span class="er">((</span><span class="va">GCF_029319035</span>.<span class="dv">1</span><span class="op">:</span><span class="dv">0</span>.<span class="dv">003675648,(((GCF_000025165.1</span><span class="op">:</span><span class="dv">0</span>.<span class="dv">000348261,(GCF_029962285.1</span><span class="op">:</span><span class="dv">0</span>.<span class="dv">000050924,(GCF_013167395.1</span><span class="op">:</span><span class="dv">0</span>.<span class="dv">000080498,GCF_013168215.1</span><span class="op">:</span><span class="dv">0</span>.<span class="dv">000060193)0</span>.<span class="dv">838</span><span class="op">:</span><span class="dv">0</span>.<span class="dv">000011756)1</span>.<span class="dv">000</span><span class="op">:</span><span class="dv">0</span>.<span class="dv">000374299)1</span>.<span class="dv">000</span><span class="op">:</span><span class="dv">0</span>.<span class="dv">008120275,(GCF_000026545.1</span><span class="op">:</span><span class="dv">0</span>.<span class="dv">013347835,GCF_009931435.1</span><span class="op">:</span><span class="dv">0</span>.<span class="dv">023794307)1</span>.<span class="dv">000</span><span class="op">:</span><span class="dv">0</span>.<span class="dv">006092878)1</span>.<span class="dv">000</span><span class="op">:</span><span class="dv">0</span>.<span class="dv">003421078,(GCF_049267065.1</span><span class="op">:</span><span class="dv">0</span>.<span class="dv">003561045,(GCF_025908275.1</span><span class="op">:</span><span class="dv">0</span>.<span class="dv">001495986,(GCF_000005845.2</span><span class="op">:</span><span class="dv">0</span>.<span class="dv">000722051,GCF_004771235.1</span><span class="op">:</span><span class="dv">0</span>.<span class="dv">002881870)1</span>.<span class="dv">000</span><span class="op">:</span><span class="dv">0</span>.<span class="dv">000792421)1</span>.<span class="dv">000</span><span class="op">:</span><span class="dv">0</span>.<span class="dv">002646365)1</span>.<span class="dv">000</span><span class="op">:</span><span class="dv">0</span>.<span class="dv">002818917)1</span>.<span class="dv">000</span><span class="op">:</span><span class="dv">0</span>.<span class="dv">002683285)1</span>.<span class="dv">000</span><span class="op">:</span><span class="dv">0</span>.<span class="dv">002002277,(GCF_000010745.1</span><span class="op">:</span><span class="dv">0</span>.<span class="dv">003339024,GCF_048571305.1</span><span class="op">:</span><span class="dv">0</span>.<span class="dv">003430258)1</span>.<span class="dv">000</span><span class="op">:</span><span class="dv">0</span>.<span class="dv">000832339)1</span>.<span class="dv">000</span><span class="op">:</span><span class="dv">0</span>.<span class="dv">000795388);</span></span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Visualización Filogénia en R</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggtree)</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(treeio)</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>tree <span class="ot">&lt;-</span> <span class="fu">read.newick</span>(<span class="st">"ecoli_tree.newick"</span>)</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a><span class="co">#Cargar información de las cepas de E. coli</span></span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(<span class="st">"Ecoli_bovineHost.tsv"</span>)</span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>metadata_ordered <span class="ot">&lt;-</span> metadata <span class="sc">%&gt;%</span></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Assembly.Accession <span class="sc">%in%</span> tree<span class="sc">$</span>tip.label) <span class="sc">%&gt;%</span></span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">match</span>(Assembly.Accession, tree<span class="sc">$</span>tip.label))</span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a><span class="co">#Convert tree information to tibble object and add metadata</span></span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a>treetib <span class="ot">&lt;-</span> <span class="fu">as_tibble</span>(tree)</span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>treetib <span class="ot">&lt;-</span> <span class="fu">left_join</span>(treetib, metadata_ordered,<span class="at">by =</span> <span class="fu">c</span>(<span class="st">'label'</span> <span class="ot">=</span> <span class="st">"Assembly.Accession"</span>))</span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a>tree_cm <span class="ot">&lt;-</span> <span class="fu">as.treedata</span>(treetib)</span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Crear la estructura princiál del arbol</span></span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ggtree</span>(tree_cm, <span class="at">ladderize =</span> <span class="cn">TRUE</span>, <span class="at">layout =</span> <span class="st">"circular"</span>) <span class="sc">+</span> </span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tiplab</span>(<span class="fu">aes</span>(<span class="at">label =</span> Strain, <span class="at">angle =</span> angle),</span>
<span id="cb25-23"><a href="#cb25-23" aria-hidden="true" tabindex="-1"></a>              <span class="at">linesize =</span> .<span class="dv">07</span>, <span class="at">size =</span> <span class="dv">4</span>, <span class="at">align =</span> <span class="cn">TRUE</span>, <span class="at">offset =</span> <span class="fl">0.008</span>) <span class="sc">+</span></span>
<span id="cb25-24"><a href="#cb25-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tippoint</span>(<span class="fu">aes</span>(<span class="at">shape =</span> Source), <span class="at">size =</span> <span class="dv">5</span>, <span class="at">alpha =</span> <span class="fl">0.3</span>) <span class="sc">+</span></span>
<span id="cb25-25"><a href="#cb25-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tippoint</span>(<span class="fu">aes</span>(<span class="at">color =</span> Geographic.location), <span class="at">size =</span> <span class="dv">5</span>, <span class="at">shape =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb25-26"><a href="#cb25-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_shape_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="dv">18</span>, <span class="dv">1</span>, <span class="dv">18</span>)) <span class="sc">+</span></span>
<span id="cb25-27"><a href="#cb25-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">shape =</span> <span class="st">"Source"</span>)<span class="sc">+</span> <span class="fu">geom_rootpoint</span>(<span class="at">col=</span><span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb25-28"><a href="#cb25-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#375c5d"</span>, <span class="st">"#b360a3"</span>, <span class="st">"darkred"</span>, <span class="st">"darkcyan"</span>,</span>
<span id="cb25-29"><a href="#cb25-29" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"#a5a436"</span>, <span class="st">"#36a540"</span>, <span class="st">"#365ba5"</span>,</span>
<span id="cb25-30"><a href="#cb25-30" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"#0cf31e"</span>, <span class="st">"#e87c09"</span>, <span class="st">"#e80917"</span>, <span class="st">"#f81df1"</span>),</span>
<span id="cb25-31"><a href="#cb25-31" aria-hidden="true" tabindex="-1"></a>                     <span class="at">na.value =</span> <span class="st">"white"</span>)</span>
<span id="cb25-32"><a href="#cb25-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-33"><a href="#cb25-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-34"><a href="#cb25-34" aria-hidden="true" tabindex="-1"></a>matrix_heatmap <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(metadata_ordered <span class="sc">%&gt;%</span> </span>
<span id="cb25-35"><a href="#cb25-35" aria-hidden="true" tabindex="-1"></a>                               <span class="fu">select</span>(Patotype))</span>
<span id="cb25-36"><a href="#cb25-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-37"><a href="#cb25-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-38"><a href="#cb25-38" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(matrix_heatmap) <span class="ot">&lt;-</span> <span class="fu">make.unique</span>(metadata_ordered<span class="sc">$</span>Assembly.Accession)</span>
<span id="cb25-39"><a href="#cb25-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-40"><a href="#cb25-40" aria-hidden="true" tabindex="-1"></a><span class="co"># Agregar heatmap</span></span>
<span id="cb25-41"><a href="#cb25-41" aria-hidden="true" tabindex="-1"></a>pphy <span class="ot">&lt;-</span> <span class="fu">gheatmap</span>(p, matrix_heatmap, <span class="at">width=</span><span class="fl">0.05</span>, </span>
<span id="cb25-42"><a href="#cb25-42" aria-hidden="true" tabindex="-1"></a>                 <span class="at">colnames=</span><span class="cn">FALSE</span>, <span class="at">legend_title=</span><span class="st">"State"</span>, <span class="at">color =</span> <span class="cn">NA</span>, ) <span class="sc">+</span></span>
<span id="cb25-43"><a href="#cb25-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>),</span>
<span id="cb25-44"><a href="#cb25-44" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="fu">c</span>(<span class="fl">1.13</span>,<span class="fl">0.5</span>), </span>
<span id="cb25-45"><a href="#cb25-45" aria-hidden="true" tabindex="-1"></a>        <span class="co"># plot.margin = margin(t = 50, r = 10, b = 50, l = 10)</span></span>
<span id="cb25-46"><a href="#cb25-46" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb25-47"><a href="#cb25-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#d65c40"</span>, <span class="st">"darkcyan"</span>, <span class="st">"#e80917"</span>, <span class="st">"lightblue"</span>,</span>
<span id="cb25-48"><a href="#cb25-48" aria-hidden="true" tabindex="-1"></a>                               <span class="st">"yellow"</span>, <span class="st">"#36a540"</span>, <span class="st">"darkblue"</span>),</span>
<span id="cb25-49"><a href="#cb25-49" aria-hidden="true" tabindex="-1"></a>                    <span class="at">na.value =</span> <span class="st">"white"</span>) <span class="sc">+</span></span>
<span id="cb25-50"><a href="#cb25-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">fill =</span> <span class="st">"Patotipo"</span>)</span></code><button title="Copiar al portapapeles" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/all_root.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
</section>
<section id="apéndice" class="level1">
<h1>Apéndice</h1>
<section id="gráficos-de-r" class="level2">
<h2 class="anchored" data-anchor-id="gráficos-de-r">Gráficos de R</h2>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copiado");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copiado");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




<script src="index_files/libs/quarto-html/zenscroll-min.js"></script>
</body></html>